--- a/build/config/win/BUILD.gn
+++ b/build/config/win/BUILD.gn
@@ -1,8 +1,9 @@
-# Copyright 2013 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
 import("//build/config/c++/c++.gni")
+import("//build/config/compiler_opt.gni")
 import("//build/config/chrome_build.gni")
 import("//build/config/clang/clang.gni")
 import("//build/config/compiler/compiler.gni")
@@ -120,8 +121,85 @@
     # Chrome currently requires SSE3. Clang supports targeting any Intel
     # microarchitecture. MSVC only supports a subset of architectures, and the
     # next step after SSE2 will be AVX.
+    if (current_cpu == "x86") {
+      cflags += [ "-mmmx", "-msse", "-msse2", "/clang:-msse2", ]
+      if (use_sse2) {
+        cflags += [ "/arch:SSE2", ]
+      }
+      if (use_sse3) {
+        cflags += [ "-msse3", "/clang:-msse3", ]
+      }
+    } else if (current_cpu == "x64") {
+      cflags += [ "-msse3", "/clang:-msse3", ]
+    }
+    
     if (current_cpu == "x86" || current_cpu == "x64") {
-      cflags += [ "-msse3" ]
+      if (is_official_build) {
+        cflags += [
+          "/clang:-O3",
+          "-Xclang", "-O3",
+        ]
+      }
+      if (use_sse41) {
+        cflags += [
+          "-mssse3",
+          "-msse4.1",
+          "/clang:-mssse3",
+          "/clang:-msse4.1",
+        ]
+      }
+      if (use_sse42) {
+        cflags += [
+          "-msse4.2",
+          "/clang:-msse4.2",
+          # "/arch:SSE4.2",
+        ]
+      }
+      if (use_avx) {
+        cflags += [
+          "-mpclmul",
+          "-maes",
+          "-mavx",
+          "/clang:-mpclmul",
+          "/clang:-maes",
+          "/clang:-mavx",
+          # "/arch:AVX",
+        ]
+      }
+      if (use_avx2) {
+        cflags += [
+          "-mavx2",
+          "-mfma",
+          "-mf16c",
+          "-mlzcnt",
+          "-mbmi",
+          "-mbmi2",
+          "/clang:-mavx2",
+          "/clang:-mfma",
+          "/clang:-mf16c",
+          "/clang:-mlzcnt",
+          "/clang:-mbmi",
+          "/clang:-mbmi2",
+          "/clang:-ffp-contract=fast",
+          # "/arch:AVX2",
+          # "/fp:fast",
+        ]
+      }
+      if (use_avx512) {
+        cflags += [
+          "-mavx512f",
+          "-mavx512cd",
+          "-mavx512vl",
+          "-mavx512bw",
+          "-mavx512dq",
+          "/clang:-mavx2",
+          "/clang:-mavx512f",
+          "/clang:-mavx512cd",
+          "/clang:-mavx512vl",
+          "/clang:-mavx512bw",
+          "/clang:-mavx512dq",
+        ]
+      }
     }
 
     # Enable ANSI escape codes if something emulating them is around (cmd.exe
@@ -160,6 +238,9 @@
 
     # Don't look for libpaths in %LIB%, similar to /X in cflags above.
     ldflags += [ "/lldignoreenv" ]
+    if (is_official_build) {
+      ldflags += [ "/OPT:lldlto=3", ]
+    }
   }
 
   # Some binaries create PDBs larger than 4 GiB. Increasing the PDB page size
@@ -187,11 +268,18 @@
       "/INCREMENTAL:NO",
       "/FIXED:NO",
     ]
+    if (use_avx2) {
+      ldflags += [ "-mllvm:-march=haswell", ]
+    }
+    if (use_avx512) {
+      ldflags += [ "-mllvm:-march=skylake-avx512", ]
+    }
 
     if (use_lld) {
       # String tail merging leads to smaller binaries, but they don't compress
       # as well, leading to increased mini_installer size (crbug.com/838449).
-      ldflags += [ "/OPT:NOLLDTAILMERGE" ]
+      # ldflags += [ "/OPT:NOLLDTAILMERGE" ]
+      ldflags += [ "/OPT:LLDTAILMERGE" ]
     }
 
     # TODO(siggi): Is this of any use anymore?
@@ -199,7 +287,7 @@
     # PDB file by about 5%) but does not otherwise alter the output binary. It
     # is enabled opportunistically for builds where it is not prohibited (not
     # supported when incrementally linking, or using /debug:fastlink).
-    ldflags += [ "/PROFILE" ]
+    # ldflags += [ "/PROFILE" ]
   }
 
   # arflags apply only to static_libraries. The normal linker configs are only
@@ -289,12 +377,14 @@
 # manually override this config for their compiles.
 config("winver") {
   defines = [
+    # 0x0A000010
     "NTDDI_VERSION=NTDDI_WIN11_GE",
 
     # We can't say `=_WIN32_WINNT_WIN10` here because some files do
     # `#if WINVER < 0x0600` without including windows.h before,
     # and then _WIN32_WINNT_WIN10 isn't yet known to be 0x0A00.
     "_WIN32_WINNT=0x0A00",
+    "_WIN64_WINNT=0x0A00",
     "WINVER=0x0A00",
   ]
 }

