# Enable Manifest V2 availability
--- a/extensions/browser/extension_prefs.cc
+++ b/extensions/browser/extension_prefs.cc
@@ -2235,7 +2235,8 @@
   registry->RegisterDictionaryPref(pref_names::kExtensionManagement);
   registry->RegisterDictionaryPref(pref_names::kOAuthRedirectUrls);
   registry->RegisterListPref(pref_names::kAllowedTypes);
-  registry->RegisterIntegerPref(pref_names::kManifestV2Availability, 0);
+  // Fuck MV3!!!
+  registry->RegisterIntegerPref(pref_names::kManifestV2Availability, 2);
   registry->RegisterListPref(pref_names::kAllowedInstallSites);
   registry->RegisterStringPref(pref_names::kLastChromeVersion, std::string());
   registry->RegisterDictionaryPref(kInstallSignature);
# Allow hosted apps on non-unpacked locations
--- a/extensions/browser/ui_util.cc
+++ b/extensions/browser/ui_util.cc
@@ -33,7 +33,7 @@
   // background pages. See http://crbug.com/116134.
   if (!Manifest::IsUnpackedLocation(location) &&
       type == Manifest::TYPE_HOSTED_APP) {
-    return false;
+    return true;
   }
 
   return true;
# Double MV3 rule limits
--- a/extensions/common/api/declarative_net_request.idl
+++ b/extensions/common/api/declarative_net_request.idl
@@ -815,28 +815,29 @@
         TestMatchOutcomeCallback callback);
   };
 
+  // Increase MV3 limits in Thorium!
   interface Properties {
     // The minimum number of static rules guaranteed to an extension across its
     // enabled static rulesets. Any rules above this limit will count towards
     // the <a href="#global-static-rule-limit">global static rule limit</a>.
-    [value=30000] static long GUARANTEED_MINIMUM_STATIC_RULES();
+    [value=60000] static long GUARANTEED_MINIMUM_STATIC_RULES();
 
     // The maximum number of combined dynamic and session scoped rules an
     // extension can add.
-    [nodoc, value=5000, deprecated="There is no longer a combined limit. See $(ref:MAX_NUMBER_OF_DYNAMIC_RULES) and $(ref:MAX_NUMBER_OF_SESSION_RULES)."] static long MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES();
+    [nodoc, value=10000, deprecated="There is no longer a combined limit. See $(ref:MAX_NUMBER_OF_DYNAMIC_RULES) and $(ref:MAX_NUMBER_OF_SESSION_RULES)."] static long MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES();
 
     // The maximum number of dynamic rules that an extension can add.
-    [value=30000] static long MAX_NUMBER_OF_DYNAMIC_RULES();
+    [value=60000] static long MAX_NUMBER_OF_DYNAMIC_RULES();
 
     // The maximum number of "unsafe" dynamic rules that an extension can add.
-    [value=5000] static long MAX_NUMBER_OF_UNSAFE_DYNAMIC_RULES();
+    [value=10000] static long MAX_NUMBER_OF_UNSAFE_DYNAMIC_RULES();
 
     // The maximum number of session scoped rules that an extension can add.
-    [value=5000] static long MAX_NUMBER_OF_SESSION_RULES();
+    [value=10000] static long MAX_NUMBER_OF_SESSION_RULES();
 
     // The maximum number of "unsafe" session scoped rules that an extension can
     // add.
-    [value=5000] static long MAX_NUMBER_OF_UNSAFE_SESSION_RULES();
+    [value=10000] static long MAX_NUMBER_OF_UNSAFE_SESSION_RULES();
 
     // Time interval within which <code>MAX_GETMATCHEDRULES_CALLS_PER_INTERVAL
     // getMatchedRules</code> calls can be made, specified in minutes.
@@ -847,20 +848,20 @@
 
     // The number of times <code>getMatchedRules</code> can be called within a
     // period of <code>GETMATCHEDRULES_QUOTA_INTERVAL</code>.
-    [value=20] static long MAX_GETMATCHEDRULES_CALLS_PER_INTERVAL();
+    [value=25] static long MAX_GETMATCHEDRULES_CALLS_PER_INTERVAL();
 
     // The maximum number of regular expression rules that an extension can
     // add. This limit is evaluated separately for the set of dynamic rules and
     // those specified in the rule resources file.
-    [value=1000] static long MAX_NUMBER_OF_REGEX_RULES();
+    [value=2000] static long MAX_NUMBER_OF_REGEX_RULES();
 
     // The maximum number of static <code>Rulesets</code> an extension can
     // specify as part of the <code>"rule_resources"</code> manifest key.
-    [value=100] static long MAX_NUMBER_OF_STATIC_RULESETS();
+    [value=256] static long MAX_NUMBER_OF_STATIC_RULESETS();
 
     // The maximum number of static <code>Rulesets</code> an extension can
     // enable at any one time.
-    [value=50] static long MAX_NUMBER_OF_ENABLED_STATIC_RULESETS();
+    [value=128] static long MAX_NUMBER_OF_ENABLED_STATIC_RULESETS();
 
     // Ruleset ID for the dynamic rules added by the extension.
     [value="_dynamic"] static DOMString DYNAMIC_RULESET_ID();
# Silence deprecated MV2 warnings
--- a/extensions/common/extension.cc
+++ b/extensions/common/extension.cc
@@ -62,7 +62,7 @@
 static_assert(kMaximumSupportedManifestVersion >=
                   kMinimumSupportedManifestVersion,
               "The modern manifest version must be supported.");
-bool g_silence_deprecated_manifest_version_warnings = false;
+bool g_silence_deprecated_manifest_version_warnings = true;
 
 // KEY MARKERS
 constexpr char kKeyBeginHeaderMarker[] = "-----BEGIN";
# Whitelist Thorium Hangouts ID
--- a/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
+++ b/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
@@ -27,10 +27,14 @@
 
 namespace extensions {
 
+// The UUID of Thorium's external Hangouts component extension
+static inline constexpr char kThoriumHangoutsId[] = "inomiaajaofonadigcpnaacolkggjjpo";
+
 bool IsComponentExtensionAllowlisted(const std::string& extension_id) {
   constexpr auto kAllowed = base::MakeFixedFlatSet<std::string_view>({
       extension_misc::kInAppPaymentsSupportAppId,
       extension_misc::kPdfExtensionId,
+      kThoriumHangoutsId,
 #if BUILDFLAG(IS_CHROMEOS)
       extension_misc::kAssessmentAssistantExtensionId,
       extension_misc::kAccessibilityCommonExtensionId,
# Default to Manifest V2 enabled
--- a/chrome/browser/extensions/extension_management_internal.h
+++ b/chrome/browser/extensions/extension_management_internal.h
@@ -189,7 +189,7 @@
   std::optional<std::vector<Manifest::Type>> allowed_types;
 
   // An enum setting indicates if manifest v2 is allowed.
-  ManifestV2Setting manifest_v2_setting = ManifestV2Setting::kDefault;
+  ManifestV2Setting manifest_v2_setting = ManifestV2Setting::kEnabled;
 
   UnpublishedAvailability unpublished_availability_setting =
       UnpublishedAvailability::kAllowUnpublished;
