# Show bookmark bar by default, no tab groups
--- a/components/bookmarks/browser/bookmark_utils.cc
+++ b/components/bookmarks/browser/bookmark_utils.cc
@@ -449,14 +449,16 @@
 
 void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(
-      prefs::kShowBookmarkBar, false,
+      prefs::kShowBookmarkBar, true,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(prefs::kEditBookmarksEnabled, true);
   registry->RegisterBooleanPref(
+      // Alex313031 removed in Thorium M120
       prefs::kShowAppsShortcutInBookmarkBar, false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(
-      prefs::kShowTabGroupsInBookmarkBar, true,
+      // Alex313031 disabled in Thorium M138
+      prefs::kShowTabGroupsInBookmarkBar, false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(
       prefs::kShowManagedBookmarksInBookmarkBar, true,
# Keep all history flag
--- a/components/history/core/browser/history_backend.cc
+++ b/components/history/core/browser/history_backend.cc
@@ -14,6 +14,7 @@
 #include <utility>
 #include <vector>
 
+#include "base/command_line.h"
 #include "base/compiler_specific.h"
 #include "base/containers/flat_set.h"
 #include "base/feature_list.h"
@@ -1326,8 +1327,12 @@
   // Get the first item in our database.
   db_->GetStartDate(&first_recorded_time_);
 
-  // Start expiring old stuff.
-  expirer_.StartExpiringOldStuff(base::Days(kExpireDaysThreshold));
+  // Start expiring old stuff if flag is unset.
+  static const bool keep_all_history =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("keep-all-history");
+  if (!keep_all_history) {
+    expirer_.StartExpiringOldStuff(base::Days(kExpireDaysThreshold));
+  }
 }
 
 void HistoryBackend::OnMemoryPressure(
@@ -1549,7 +1554,13 @@
 }
 
 bool HistoryBackend::IsExpiredVisitTime(const base::Time& time) const {
-  return time < expirer_.GetCurrentExpirationTime();
+  static const bool keep_all_history =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("keep-all-history");
+  if (keep_all_history) {
+    return false;
+  } else {
+    return time < expirer_.GetCurrentExpirationTime();
+  }
 }
 
 // static
# Increase history threshold to 120 days
--- a/components/history/core/browser/history_backend.h
+++ b/components/history/core/browser/history_backend.h
@@ -208,7 +208,7 @@
 
   // The number of days old a history entry can be before it is considered "old"
   // and is deleted.
-  static constexpr int kExpireDaysThreshold = 90;
+  static constexpr int kExpireDaysThreshold = 120;
 
   // Init must be called to complete object creation. This object can be
   // constructed on any thread, but all other functions including Init() must
# Disable background mode by default
--- a/chrome/browser/background/extensions/background_mode_manager.cc
+++ b/chrome/browser/background/extensions/background_mode_manager.cc
@@ -364,7 +364,7 @@
 
 // static
 void BackgroundModeManager::RegisterPrefs(PrefRegistrySimple* registry) {
-  registry->RegisterBooleanPref(prefs::kBackgroundModeEnabled, true);
+  registry->RegisterBooleanPref(prefs::kBackgroundModeEnabled, false);
 }
 
 void BackgroundModeManager::RegisterProfile(Profile* profile) {
# Comment out Ubuntu 18.04 as obsolete
--- a/chrome/browser/obsolete_system/obsolete_system_linux.cc
+++ b/chrome/browser/obsolete_system/obsolete_system_linux.cc
@@ -30,7 +31,7 @@
     "Fedora 38",
     "Ubuntu 14.04",
     "Ubuntu 16.04",
-    "Ubuntu 18.04",
+    //"Ubuntu 18.04",
     "openSUSE Leap 15.0",
     "openSUSE Leap 15.1",
     "openSUSE Leap 15.2",
