# Allow insecure downloads flag
--- a/chrome/browser/download/chrome_download_manager_delegate.cc
+++ b/chrome/browser/download/chrome_download_manager_delegate.cc
@@ -1952,6 +1952,13 @@ bool ChromeDownloadManagerDelegate::IsOpenInBrowserPreferredForFile(
 bool ChromeDownloadManagerDelegate::ShouldBlockFile(
     download::DownloadItem* item,
     download::DownloadDangerType danger_type) const {
+  // Don't block downloads if flag is set
+  static const bool allow_insecure_downloads_ =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("allow-insecure-downloads");
+  if (allow_insecure_downloads_) {
+    return false;
+  }
+
   policy::DownloadRestriction download_restriction =
       download_prefs_->download_restriction();
 
@@ -1977,42 +1984,13 @@ bool ChromeDownloadManagerDelegate::ShouldBlockFile(
 void ChromeDownloadManagerDelegate::MaybeSendDangerousDownloadOpenedReport(
     DownloadItem* download,
     bool show_download_in_folder) {
-#if BUILDFLAG(SAFE_BROWSING_DOWNLOAD_PROTECTION)
-  safe_browsing::DownloadProtectionService* service =
-      GetDownloadProtectionService();
-  if (service) {
-    service->MaybeSendDangerousDownloadOpenedReport(download,
-                                                    show_download_in_folder);
-  }
-#endif
+  return;
 }
 
 void ChromeDownloadManagerDelegate::MaybeSendDangerousDownloadCanceledReport(
     DownloadItem* download,
     bool is_shutdown) {
-#if BUILDFLAG(SAFE_BROWSING_DOWNLOAD_PROTECTION)
-  safe_browsing::SafeBrowsingService* sb_service =
-      g_browser_process->safe_browsing_service();
-  if (!sb_service) {
-    return;
-  }
-  // Note: We cannot go through download_protection_service here, because this
-  // function may be called at shutdown. The download_protection_service
-  // object may already be deleted at this point.
-  if (is_shutdown) {
-    sb_service->PersistDownloadReportAndSendOnNextStartup(
-        download,
-        safe_browsing::ClientSafeBrowsingReportRequest::
-            DANGEROUS_DOWNLOAD_PROFILE_CLOSED,
-        /*did_proceed=*/false, std::nullopt);
-  } else {
-    sb_service->SendDownloadReport(
-        download,
-        safe_browsing::ClientSafeBrowsingReportRequest::
-            DANGEROUS_DOWNLOAD_AUTO_DELETED,
-        /*did_proceed=*/false, std::nullopt);
-  }
-#endif
+  return;
 }
 
 void ChromeDownloadManagerDelegate::CheckDownloadAllowed(

# Download target determiner fixes
--- a/chrome/browser/download/download_target_determiner.cc
+++ b/chrome/browser/download/download_target_determiner.cc
@@ -8,6 +8,7 @@
 #include <string>
 #include <vector>
 
+#include "base/command_line.h"
 #include "base/files/file_path.h"
 #include "base/functional/bind.h"
 #include "base/location.h"
@@ -992,8 +993,13 @@
   // result is available.
   danger_level_ = GetDangerLevel(NO_VISITS_TO_REFERRER);
 
-  if (danger_level_ == DownloadFileType::NOT_DANGEROUS)
+  static const bool allow_insecure_downloads_ =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("allow-insecure-downloads");
+
+  // Continue with this Thorium flag
+  if (danger_level_ == DownloadFileType::NOT_DANGEROUS || allow_insecure_downloads_) {
     return CONTINUE;
+  }
 
   if (danger_level_ == DownloadFileType::ALLOW_ON_USER_GESTURE) {
 #if BUILDFLAG(IS_ANDROID)
@@ -1295,6 +1301,13 @@
 DownloadFileType::DangerLevel DownloadTargetDeterminer::GetDangerLevel(
     PriorVisitsToReferrer visits) const {
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
+
+  static const bool allow_insecure_downloads_ =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("allow-insecure-downloads");
+  // Allow all downloads with this Thorium flag
+  if (allow_insecure_downloads_) {
+    return DownloadFileType::NOT_DANGEROUS;
+  }
 
   // User-initiated extension downloads from pref-whitelisted sources are not
   // considered dangerous.
# Download UI controller
--- a/chrome/browser/download/download_ui_controller.cc
+++ b/chrome/browser/download/download_ui_controller.cc
@@ -152,6 +152,68 @@
       });
 }

+class DownloadShelfUIControllerDelegate
+    : public DownloadUIController::Delegate {
+ public:
+  // |profile| is required to outlive DownloadShelfUIControllerDelegate.
+  explicit DownloadShelfUIControllerDelegate(Profile* profile)
+      : profile_(profile) {
+    // Match DownloadBubbleUIControllerDelegate behavior: in incognito mode,
+    // prompt for download location.
+    if (profile_->IsOffTheRecord()) {
+      profile_->GetPrefs()->SetBoolean(prefs::kPromptForDownload, true);
+    }
+  }
+  ~DownloadShelfUIControllerDelegate() override = default;
+
+ private:
+  // DownloadUIController::Delegate
+  void OnNewDownloadReady(download::DownloadItem* item) override;
+
+  raw_ptr<Profile> profile_;
+};
+
+void DownloadShelfUIControllerDelegate::OnNewDownloadReady(
+    download::DownloadItem* item) {
+  // Here the item will be surfaced to the bubble UI and should
+  // subject to the auto resumption logic.
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC)
+  if (download::features::IsBackoffInDownloadingEnabled()) {
+    auto* handler = download::DesktopAutoResumptionHandler::Get();
+    item->RemoveObserver(handler);
+    item->AddObserver(handler);
+  }
+#endif
+  content::WebContents* web_contents =
+      content::DownloadItemUtils::GetWebContents(item);
+  // For the case of DevTools web contents, we'd like to use target browser
+  // shelf although saving from the DevTools web contents.
+  if (web_contents && DevToolsWindow::IsDevToolsWindow(web_contents)) {
+    DevToolsWindow* devtools_window =
+        DevToolsWindow::AsDevToolsWindow(web_contents);
+    content::WebContents* inspected =
+        devtools_window->GetInspectedWebContents();
+    // Do not overwrite web contents for the case of remote debugging.
+    if (inspected)
+      web_contents = inspected;
+  }
+  Browser* browser =
+      web_contents ? chrome::FindBrowserWithTab(web_contents) : nullptr;
+
+  // As a last resort, use the last active browser for this profile. Not ideal,
+  // but better than not showing the download at all.
+  if (browser == nullptr)
+    browser = chrome::FindLastActiveWithProfile(profile_);
+
+  if (browser && browser->window() && browser->window()->GetDownloadShelf() &&
+      DownloadItemModel(item).ShouldShowInShelf()) {
+    DownloadUIModel::DownloadUIModelPtr model = DownloadItemModel::Wrap(item);
+
+    // GetDownloadShelf creates the download shelf if it was not yet created.
+    browser->window()->GetDownloadShelf()->AddDownload(std::move(model));
+  }
+}
+
 #endif
 } // namespace

@@ -174,8 +236,12 @@
 #if BUILDFLAG(IS_CHROMEOS)
     delegate_ = std::make_unique<DownloadNotificationManager>(profile);
 #else
-    delegate_ = std::make_unique<DownloadBubbleUIControllerDelegate>(profile);
-    InitializeDownloadBubbleUpdateService(profile, manager);
+    if (download::IsDownloadBubbleEnabled()) {
+      delegate_ = std::make_unique<DownloadBubbleUIControllerDelegate>(profile);
+      InitializeDownloadBubbleUpdateService(profile, manager);
+    } else {
+      delegate_ = std::make_unique<DownloadShelfUIControllerDelegate>(profile);
+    }
 #endif
   }
 #endif  // BUILDFLAG(IS_ANDROID)

--- a/chrome/browser/download/insecure_download_blocking.cc
+++ b/chrome/browser/download/insecure_download_blocking.cc
@@ -6,6 +6,7 @@
 
 #include <optional>
 
+#include "base/command_line.h"
 #include "base/debug/crash_logging.h"
 #include "base/debug/dump_without_crashing.h"
 #include "base/memory/raw_ptr.h"
@@ -289,6 +290,8 @@
     // used for Chrome stuff. Background fetch should probably be HTTPS-only.
     auto download_source = item->GetDownloadSource();
     auto transition_type = item->GetTransitionType();
+    static const bool allow_insecure_downloads_ =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("allow-insecure-downloads");
     if (download_source == DownloadSource::RETRY ||
         (transition_type & ui::PAGE_TRANSITION_RELOAD) ||
         (transition_type & ui::PAGE_TRANSITION_TYPED) ||
@@ -300,7 +303,8 @@
         download_source == DownloadSource::OFFLINE_PAGE ||
         download_source == DownloadSource::INTERNAL_API ||
         download_source == DownloadSource::EXTENSION_API ||
-        download_source == DownloadSource::EXTENSION_INSTALLER) {
+        download_source == DownloadSource::EXTENSION_INSTALLER ||
+        allow_insecure_downloads_) {
       base::UmaHistogramEnumeration(
           kInsecureDownloadHistogramName,
           InsecureDownloadSecurityStatus::kDownloadIgnored);
@@ -340,7 +344,8 @@
         download_source == DownloadSource::OFFLINE_PAGE ||
         download_source == DownloadSource::INTERNAL_API ||
         download_source == DownloadSource::EXTENSION_API ||
-        download_source == DownloadSource::EXTENSION_INSTALLER) {
+        download_source == DownloadSource::EXTENSION_INSTALLER ||
+        allow_insecure_downloads_) {
       is_insecure_download_ = false;
     } else {  // Not ignorable download.
       // TODO(crbug.com/40857867): Add blocking metrics.
@@ -473,8 +478,10 @@
     const download::DownloadItem* item) {
   InsecureDownloadData data(path, item);
 
-  // If the download is fully secure, early abort.
-  if (!data.is_insecure_download_) {
+  static const bool allow_insecure_downloads_ =
+    base::CommandLine::ForCurrentProcess()->HasSwitch("allow-insecure-downloads");
+  // If the download is fully secure, early abort. Don't nag.
+  if (!data.is_insecure_download_ || allow_insecure_downloads_) {
     return InsecureDownloadStatus::SAFE;
   }
 
# Download features config
--- a/components/download/public/common/download_features.cc
+++ b/components/download/public/common/download_features.cc
@@ -14,7 +14,7 @@
 #if BUILDFLAG(IS_ANDROID)
              base::FEATURE_ENABLED_BY_DEFAULT
 #else
-             base::FEATURE_DISABLED_BY_DEFAULT
+             base::FEATURE_ENABLED_BY_DEFAULT
 #endif
 );
 
