# GPU build with VAAPI config
--- a/content/gpu/BUILD.gn
+++ b/content/gpu/BUILD.gn
@@ -6,6 +6,7 @@
 import("//build/config/ui.gni")
 import("//gpu/vulkan/features.gni")
 import("//media/media_options.gni")
+import("//media/gpu/args.gni")
 
 # See //content/BUILD.gn for how this works.
 group("gpu") {
@@ -128,4 +129,8 @@
       !is_castos) {
     configs += [ "//build/config/linux/dri" ]
   }
+
+  if (is_linux && use_vaapi) {
+    public_configs = [ "//build/config/linux/libva" ]
+  }
 }
# Vulkan ICD paths in GPU sandbox
--- a/content/common/gpu_pre_sandbox_hook_linux.cc
+++ b/content/common/gpu_pre_sandbox_hook_linux.cc
@@ -262,6 +262,12 @@
       "/usr/lib64/dri/r300_dri.so",
       "/usr/lib64/dri/r600_dri.so",
       "/usr/lib64/dri/radeonsi_dri.so",
+      // GPU Log Warnings Workaround
+      "/usr/lib/libvulkan_radeon.so",
+      "/usr/share/vulkan/icd.d",
+      "/usr/share/vulkan/icd.d/radeon_icd.x86_64.json",
+      "/etc/vulkan/icd.d",
+      "/etc/vulkan/icd.d/radeon_icd.x86_64.json",
       // Allow libglvnd files and libs.
       "/usr/share/glvnd/egl_vendor.d",
       "/usr/share/glvnd/egl_vendor.d/50_mesa.json",
