# Android compiler flags and official build
--- a/build/config/android/BUILD.gn
+++ b/build/config/android/BUILD.gn
@@ -1,9 +1,10 @@
-# Copyright 2014 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
 import("//build/config/android/rules.gni")
 import("//build/config/c++/c++.gni")
+import("//build/config/compiler_opt.gni")
 import("//build/config/compiler/compiler.gni")
 import("//build/config/sanitizers/sanitizers.gni")
 
@@ -73,6 +74,16 @@
 
   cflags += [ "--target=$android_abi_target$android_ndk_api_level" ]
   ldflags += [ "--target=$android_abi_target$android_ndk_api_level" ]
+
+  if (is_official_build) {
+    cflags += [ "-O3", ]
+    ldflags += [ "-Wl,-O3", ]
+  }
+
+  if (current_cpu == "x86" || current_cpu == "x64") {
+    cflags += [ "-msse3", "-mssse3", "-msse4.1" ]
+    ldflags += [ "-msse3", "-mssse3", "-msse4.1" ]
+  }
 
   # Assign any flags set for the C compiler to asmflags so that they are sent
   # to the assembler.
# ARM FPU/NEON defaults removal
--- a/build/config/arm.gni
+++ b/build/config/arm.gni
@@ -1,4 +1,4 @@
-# Copyright 2014 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -120,15 +120,42 @@
   }
 } else if (current_cpu == "arm64" || v8_current_cpu == "arm64" ||
            current_cpu == "arm64e" || v8_current_cpu == "arm64e") {
-  # arm64 supports only "hard".
-  arm_float_abi = "hard"
-  arm_use_neon = true
   declare_args() {
     # Enables the new Armv8 branch protection features. Valid strings are:
     # - "pac": Enables Pointer Authentication Code (PAC, featured in Armv8.3)
     # - "standard": Enables both PAC and Branch Target Identification (Armv8.5).
     # - "none": No branch protection.
     arm_control_flow_integrity = "none"
+    
+    # Version of the ARM processor when compiling on ARM. Ignored on non-ARM
+    # platforms.
+    arm_version = 8
+    
+    # arm64 supports only "hard".
+    arm_float_abi = "hard"
+
+    # The ARM64 architecture. This will be a string like "armv8-a" or "armv8.5-a.
+    # An empty string means to use the default for the arm_version.
+    arm_arch = "armv8-a"
+    
+    # The ARM variant-specific tuning mode. This will be a string like "armv6"
+    # or "cortex-a15". An empty string means to use the default for the
+    # arm_version.
+    arm_tune = "generic-armv8-a"
+
+    # The ARM64 floating point hardware. This will be a string like "neon" or
+    # "vfpv3-d16". An empty string means to use the default for the arm_version.
+    arm_fpu = "neon"
+
+    # Whether to use the neon FPU instruction set or not.
+    arm_use_neon = "true"
+
+    # Whether to enable optional NEON code paths.
+    arm_optionally_use_neon = false
+
+    # Thumb is a reduced instruction set available on some ARM processors that
+    # has increased code density.
+    arm_use_thumb = true
 
     if ((is_android || is_linux) && target_cpu == "arm64") {
       # Enable PAC and BTI on AArch64 Linux/Android systems.
@@ -137,7 +164,6 @@
       # use the arm64 simulator.
       arm_control_flow_integrity = "standard"
     }
-
     if (host_os == "mac" && host_cpu == "arm64") {
       # Disable ARM integrity flow or build
       # will fail because of this issue
@@ -145,8 +171,13 @@
       arm_control_flow_integrity = "none"
     }
   }
+  # Initial values from upstream. arm64 supports only "hard".
+  arm_version = 8
+  arm_float_abi = "hard"
+  arm_use_neon = true
+  
   assert(arm_control_flow_integrity == "none" ||
              arm_control_flow_integrity == "standard" ||
              arm_control_flow_integrity == "pac",
-         "Invalid branch protection option")
+         "Invalid ARM branch protection option.")
 }
# SIMD optimization config
--- a/build/config/BUILDCONFIG.gn
+++ b/build/config/BUILDCONFIG.gn
@@ -421,6 +421,9 @@
   default_compiler_configs += [ "//build/config:release" ]
 }
 
+# Make sure Thorium SIMD flags come last
+default_compiler_configs += [ "//build/config/compiler:thorium_simd_optimization", ]
+
 # Static libraries and source sets use only the compiler ones.
 set_defaults("static_library") {
   configs = default_compiler_configs
# Compiler build config
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -7,6 +7,7 @@ import("//build/config/android/config.gni")
 import("//build/config/c++/c++.gni")
 import("//build/config/c++/modules.gni")
 import("//build/config/cast.gni")
+import("//build/config/compiler_opt.gni")
 import("//build/config/chrome_build.gni")
 import("//build/config/chromeos/args.gni")
 import("//build/config/clang/clang.gni")
@@ -467,6 +468,11 @@ config("compiler") {
       }
     }
 
+    # ADDED BY ALEX313031 FOR THORIUM
+    if (is_clang && is_android && !is_ubsan && !is_ubsan_security) {
+      cflags += [ "-fwrapv" ]
+    }
+
     # Linker warnings.
     if (fatal_linker_warnings && !is_apple && current_os != "aix" &&
         current_os != "zos") {
@@ -562,6 +568,18 @@ config("compiler") {
     cflags += [ "-fPIC" ]
     ldflags += [ "-fPIC" ]
     rustflags += [ "-Crelocation-model=pic" ]
+    #if (current_cpu == "x86" || current_cpu == "x64") {
+      #asmflags += [ "-fno-emulated-tls", "-w", ]
+      #cflags += [ "-fno-emulated-tls", "-w", ]
+      #ldflags += [ "-Wl,-plugin-opt=-emulated-tls=0", "-w", ]
+      #rustflags += [ "-Awarnings", ]
+    #}
+
+    #if (is_clang && chrome_pgo_phase == 2 && !is_android) {
+      #asmflags += [ "-fbasic-block-sections=labels" ]
+      #cflags += [ "-fbasic-block-sections=labels" ]
+      #rustflags += [ "-Cllvm-args=-basic-block-sections=labels" ]
+    #}
 
     if (!is_clang) {
       # Use pipes for communicating between sub-processes. Faster.
@@ -646,9 +664,17 @@ config("compiler") {
 
     if (!use_sized_deallocation) {
       if (is_win) {
-        cflags += [ "/Zc:sizedDealloc-" ]
+        if (is_debug) {
+          cflags += [ "/Zc:sizedDealloc-" ]
+        } else {
+          cflags += [ "/Zc:sizedDealloc" ]
+        }
       } else {
-        cflags += [ "-fno-sized-deallocation" ]
+        if (is_debug) {
+          cflags += [ "-fno-sized-deallocation" ]
+        } else {
+          cflags += [ "-fsized-deallocation" ]
+        }
       }
     }
 
@@ -715,11 +741,19 @@ config("compiler") {
       }
     }
 
-    # TODO(crbug.com/40192287): Investigate why/if this should be needed.
-    if (is_win) {
-      cflags += [ "/clang:-ffp-contract=off" ]
+    if (use_avx2 || use_avx512 || use_fma || current_cpu == "arm64" || is_mac) {
+      if (is_win) {
+        cflags += [ "/clang:-ffp-contract=fast" ]
+      } else {
+        cflags += [ "-ffp-contract=fast" ]
+      }
     } else {
-      cflags += [ "-ffp-contract=off" ]
+      # TODO(crbug.com/40192287): Investigate why/if this should be needed.
+      if (is_win) {
+        cflags += [ "/clang:-ffp-contract=off" ]
+      } else {
+        cflags += [ "-ffp-contract=off" ]
+      }
     }
 
     # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF.
@@ -874,7 +908,11 @@ config("compiler") {
     # [0]: https://pinpoint-dot-chromeperf.appspot.com/job/16984a18be0000
     # [1]: https://pinpoint-dot-chromeperf.appspot.com/job/11984a18be0000
     # [2]: https://ci.chromium.org/ui/p/chromium/builders/try/android-binary-size/1848442
-    import_instr_limit = 30
+    if (is_official_build) {
+      import_instr_limit = 100
+    } else {
+      import_instr_limit = 30
+    }
 
     if (is_win) {
       ldflags += [
@@ -900,11 +938,11 @@ config("compiler") {
         # bloat of ThinLTO to <10%, but that's potentially no longer true.
         # FIXME(inglorion): maybe tune these?
         # TODO(b/271459198): Revert limit on amd64 to 30 when fixed.
-        import_instr_limit = 20
+        import_instr_limit = 100
       } else if (is_android && optimize_for_size) {
         # TODO(crbug.com/40219076): Investigate if we can get the > 6% perf win
         # of import_instr_limit 30 with a binary size hit smaller than ~2 MiB.
-        import_instr_limit = 5
+        import_instr_limit = 100
       }
 
       ldflags += [ "-Wl,-mllvm,-import-instr-limit=$import_instr_limit" ]
@@ -935,6 +973,7 @@ config("compiler") {
     # TODO(crbug.com/40182783): investigate why this isn't effective on
     # arm32.
     if (!is_android || current_cpu == "arm64") {
+      # Add "-fstrict-vtable-pointers",
       cflags += [ "-fwhole-program-vtables" ]
 
       if (toolchain_supports_rust_thin_lto) {
@@ -955,6 +994,10 @@ config("compiler") {
       }
     }
 
+    if (is_win && target_cpu == "arm64") {
+      cflags += [ "-Wno-error", "-Wno-unreachable-code", ]
+    }
+
     # This flag causes LTO to create an .ARM.attributes section with the correct
     # architecture. This is necessary because LLD will refuse to link a program
     # unless the architecture revision in .ARM.attributes is sufficiently new.
@@ -1152,6 +1195,34 @@ config("compiler") {
     "-Zexternal-clangrt",
   ]
 
+  if (current_cpu == "arm64") {
+    rustflags += [ "-Cllvm-args=-fp-contract=fast", "-Ctarget-feature=+neon", ]
+  }
+
+  if (current_cpu == "x86" || current_cpu == "x64") {
+    if (use_sse2 && !use_sse3) {
+      rustflags += [ "-Ctarget-feature=+sse2", ]
+    }
+    if (use_sse3 && !use_sse41) {
+      rustflags += [ "-Ctarget-feature=+sse3", ]
+    }
+    if (use_sse41 && !use_sse42) {
+      rustflags += [ "-Ctarget-feature=+sse3,+ssse3,+sse4.1", ]
+    }
+    if (use_sse42 && !use_avx) {
+      rustflags += [ "-Ctarget-feature=+sse3,+ssse3,+sse4.1,+sse4.2", ]
+    }
+    if (use_avx && !use_avx2) {
+      rustflags += [ "-Ctarget-feature=+sse3,+ssse3,+sse4.1,+sse4.2,+pclmulqdq,+aes,+avx", ]
+    }
+    if (use_avx2 && !use_avx512) {
+      rustflags += [ "-Ctarget-feature=+sse3,+ssse3,+sse4.1,+sse4.2,+pclmulqdq,+aes,+avx,+avx2,+fma,+f16c,+lzcnt,+bmi2", "-Cllvm-args=-fp-contract=fast", ]
+    }
+    if (use_avx512) {
+      rustflags += [ "-Ctarget-feature=+sse3,+ssse3,+sse4.1,+sse4.2,+pclmulqdq,+aes,+avx,+avx2,+fma,+f16c,+lzcnt,+bmi2,+mavx512f", "-Cllvm-args=-fp-contract=fast", ]
+    }
+  }
+
   if (!is_win || force_rustc_color_output) {
     # Colorize error output. The analogous flag is passed for clang. This must
     # be platform-gated since rustc will unconditionally output ANSI escape
@@ -1171,7 +1242,7 @@ config("compiler") {
     rustflags += [ "-Clto=no" ]
   }
   if (is_official_build) {
-    rustflags += [ "-Ccodegen-units=1" ]
+    rustflags += [ "-Ccodegen-units=1", "-Copt-level=3", ]
   }
   if (!rust_prebuilt_stdlib) {
     # When building against the Chromium Rust stdlib (which we compile) always
@@ -1218,12 +1289,16 @@ config("disallow_unstable_features") {
 # building with ThinLTO, no optimization is performed in the link step.
 config("thinlto_optimize_default") {
   if (!is_debug && use_thin_lto && is_a_target_toolchain) {
-    lto_opt_level = 0
+    lto_opt_level = 2
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-mllvm:-enable-pre=false", ]
+      ldflags += [ "-opt:lldltocgo=" + lto_opt_level ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,-mllvm,-enable-pre=false", ]
+      ldflags += [ "-Wl,--lto-CGO" + lto_opt_level ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1251,13 +1326,18 @@ config("thinlto_optimize_max") {
     if (thin_lto_enable_optimizations) {
       lto_opt_level = 2
     } else {
-      lto_opt_level = 0
+      # Alex313031 TODO: Set back to 3 when LLVM bug is fixed
+      lto_opt_level = 2
     }
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-mllvm:-enable-pre=false", ]
+      ldflags += [ "-opt:lldltocgo=" + lto_opt_level ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,-mllvm,-enable-pre=false", ]
+      ldflags += [ "-Wl,--lto-CGO" + lto_opt_level ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1298,7 +1378,7 @@ config("compiler_cpu_abi") {
         cflags += [ "-m64" ]
         ldflags += [ "-m64" ]
       }
-      cflags += [ "-msse3" ]
+      #cflags += [ "-msse3" ]
     } else if (current_cpu == "x86") {
       if (is_clang && !is_android && !is_chromeos_device) {
         cflags += [ "--target=i386-unknown-linux-gnu" ]
@@ -1309,7 +1389,9 @@ config("compiler_cpu_abi") {
       }
       cflags += [
         "-mfpmath=sse",
-        "-msse3",
+        "-mmmx",
+        "-msse",
+        "-msse2",
       ]
     } else if (current_cpu == "arm") {
       if (is_clang && !is_android && !is_chromeos_device) {
@@ -2430,6 +2512,14 @@ config("no_chromium_code") {
   cflags_cc = []
   defines = []
 
+  if (is_official_build) {
+    if (is_win) {
+      cflags += [ "/O2" ]
+    } else {
+      cflags += [ "-O3" ]
+    }
+  }
+
   if (is_win) {
     if (is_clang) {
       cflags += [ "/W3" ]  # Warning level 3.
@@ -2455,6 +2545,9 @@ config("no_chromium_code") {
       # Disabled for similar reasons as -Wunused-variable.
       "-Wno-unused-but-set-variable",
     ]
+    if (is_official_build && is_win) {
+      cflags += [ "-Xclang", "-O3" ]
+    }
   }
 
   configs = [ ":default_warnings" ]
@@ -2642,6 +2735,120 @@ config("no_incompatible_pointer_warnings") {
   }
 }
 
+# Thorium SIMD optimizations
+config("thorium_simd_optimization") {
+  cflags = []
+  ldflags = []
+  if (current_cpu == "x86") {
+    if (is_win) {
+      if (use_sse2) {
+        cflags += [ "/arch:SSE2", "/clang:-msse2", ]
+      }
+      if (use_sse3) {
+        cflags += [ "/clang:-msse3", ]
+      }
+      if (use_sse41) {
+        cflags += [ "/clang:-mssse3", "/clang:-msse4.1", ]
+      }
+      if (use_sse42) {
+        cflags += [ "/clang:-msse4.2", ]
+      }
+    } else {
+      if (use_sse2) {
+        cflags += [ "-msse2", ]
+        ldflags += [ "-msse2", ]
+      }
+      if (use_sse3) {
+        cflags += [ "-msse3", ]
+        ldflags += [ "-msse3", ]
+      }
+      if (use_sse41) {
+        cflags += [ "-mssse3", "-msse4.1", ]
+        ldflags += [ "-mssse3", "-msse4.1", ]
+      }
+      if (use_sse42) {
+        cflags += [ "-msse4", "-msse4.2", ]
+        ldflags += [ "-msse4.2", ]
+      }
+    }
+  } else if (current_cpu == "x64") {
+    if (is_win) {
+      if (use_sse3) {
+        cflags += [ "/clang:-msse3", ]
+      }
+      if (use_sse41) {
+        cflags += [ "/clang:-mssse3", "/clang:-msse4.1", ]
+      }
+      if (use_sse42) {
+        cflags += [ "/clang:-msse4.2", ]
+      }
+      if (use_avx) {
+        cflags += [ "/clang:-mpclmul", "/clang:-maes", "/clang:-mavx", ]
+      }
+      if (use_fma) {
+        cflags += [ "/clang:-mfma", "/clang:-ffp-contract=fast", ]
+      }
+      if (use_avx2) {
+        cflags += [ "/clang:-mavx2", "/clang:-mf16c", "/clang:-mlzcnt", "/clang:-mbmi", "/clang:-mbmi2", ]
+        ldflags += [ "-mllvm:-march=haswell", ]
+      }
+      if (use_avx512) {
+        cflags += [ "/clang:-mavx512f", "/clang:-mavx512cd", "/clang:-mavx512vl", "/clang:-mavx512bw", "/clang:-mavx512dq", ]
+        ldflags += [ "-mllvm:-march=skylake-avx512", ]
+      }
+    } else if (is_mac) {
+      if (use_avx2) {
+        cflags += [ "-march=x86-64-v3", ]
+        ldflags += [ "-Wl,-mllvm,-march=x86-64-v3", ]
+      } else {
+        cflags += [ "-msse3", "-mssse3", "-msse4.1", ]
+      }
+    } else {
+      # Linux and ChromiumOS and Android
+      if (use_sse3) {
+        cflags += [ "-msse3", ]
+        ldflags += [ "-msse3", ]
+      }
+      if (use_sse41) {
+        cflags += [ "-mssse3", "-msse4.1", ]
+        ldflags += [ "-mssse3", "-msse4.1", ]
+      }
+      if (use_sse42) {
+        cflags += [ "-msse4", "-msse4.2", ]
+        ldflags += [ "-msse4.2", ]
+      }
+      if (use_avx) {
+        cflags += [ "-mpclmul", "-maes", "-mavx", ]
+        ldflags += [ "-mpclmul", "-maes", "-mavx", ]
+      }
+      if (use_fma) {
+        cflags += [ "-mfma", "-ffp-contract=fast", ]
+        ldflags += [ "-mfma", "-Wl,-mllvm,-fp-contract=fast", ]
+      }
+      if (use_avx2) {
+        cflags += [ "-mavx2", "-mf16c", "-mlzcnt", "-mbmi", "-mbmi2", ]
+        ldflags += [ "-mavx2", "-mf16c", "-mlzcnt", "-mbmi", "-mbmi2", "-Wl,-mllvm,-march=haswell", ]
+      }
+      if (use_avx512) {
+        cflags += [ "-mavx512f", "-mavx512cd", "-mavx512vl", "-mavx512bw", "-mavx512dq", ]
+        ldflags += [ "-mavx512f", "-mavx512cd", "-mavx512vl", "-mavx512bw", "-mavx512dq", "-Wl,-mllvm,-march=skylake-avx512", ]
+      }
+    }
+  } else if (current_cpu == "arm") {
+    if (is_android || is_linux) {
+    }
+  } else if (current_cpu == "arm64") {
+    if (is_win) {
+    } else if (is_mac) {
+      cflags += [ "-mcpu=apple-m1"]
+    } else if (!is_android) {
+      if (is_raspi) {
+        cflags += [ "-mtune=cortex-a72", ]
+      }
+    }
+  }
+}
+
 # Optimization -----------------------------------------------------------------
 #
 # The BUILDCONFIG file sets the "default_optimization" config on targets by
@@ -2657,7 +2865,10 @@ config("no_incompatible_pointer_warnings") {
 # Shared settings for both "optimize" and "optimize_max" configs.
 # IMPORTANT: On Windows "/O1" and "/O2" must go before the common flags.
 if (is_win) {
-  common_optimize_on_cflags = [
+  common_optimize_on_cflags = []
+  common_optimize_on_ldflags = []
+
+  common_optimize_on_cflags += [
     "/Oy-",  # Disable omitting frame pointers, must be after /O2.
     "/Zc:inline",  # Remove unreferenced COMDAT (faster links).
   ]
@@ -2671,7 +2882,23 @@ if (is_win) {
       "/Gw",
     ]
   }
-  common_optimize_on_ldflags = []
+
+  if (is_official_build) {
+    common_optimize_on_cflags += [
+      "-mllvm", "-aggressive-ext-opt",
+      "-mllvm", "-enable-gvn-hoist",
+      "/O2",
+      "/clang:-O3",
+      "-Xclang", "-O3",
+    ]
+  }
+
+  if (is_official_build) {
+    common_optimize_on_ldflags += [
+      "-mllvm:-aggressive-ext-opt",
+      "-mllvm:-enable-gvn-hoist",
+    ]
+  }
 
   # /OPT:ICF is not desirable in Debug builds, since code-folding can result in
   # misleading symbols in stack traces.
@@ -2686,6 +2913,19 @@ if (is_win) {
   common_optimize_on_cflags = []
   common_optimize_on_ldflags = []
 
+  if (is_official_build) {
+    common_optimize_on_cflags += [
+      "-mllvm", "-aggressive-ext-opt",
+      "-mllvm", "-enable-gvn-hoist",
+      "-O3",
+    ]
+    common_optimize_on_ldflags += [
+      "-Wl,-mllvm,-aggressive-ext-opt",
+      "-Wl,-mllvm,-enable-gvn-hoist",
+      "-Wl,-O3",
+    ]
+  }
+
   if (is_android) {
     # TODO(jdduke) Re-enable on mips after resolving linking
     # issues with libc++ (crbug.com/456380).
@@ -2717,9 +2957,7 @@ if (is_win) {
       "-ffunction-sections",
     ]
     if (is_clang) {
-      # We don't care about unique section names, this makes object files a bit
-      # smaller.
-      common_optimize_on_cflags += [ "-fno-unique-section-names" ]
+      common_optimize_on_cflags += [ "-funique-section-names" ]
     }
 
     if (is_official_build) {
@@ -2727,13 +2965,14 @@ if (is_win) {
         # Specifically tell the linker to perform optimizations.
         # See http://lwn.net/Articles/192624/.
         # -O2 enables string tail merge optimization in lld.
-        "-Wl,-O2",
+        "-Wl,-O3",
       ]
     }
 
     common_optimize_on_ldflags += [ "-Wl,--gc-sections" ]
   }
 
+  
   if (chrome_pgo_phase == 1 && target_os == "android") {
     # LLVM's SLP vectorizer is causing compiler hangs when run on PGO
     # instrumented code. Disable it until that's fixed. This does not affect the
@@ -2743,6 +2982,72 @@ if (is_win) {
   }
 }
 
+config("march_crc") {
+  if (!is_win || is_clang) {
+    cflags = [ "-march=armv8-a+crc" ]
+    asmflags = cflags
+  }
+}
+
+config("march_dotprod") {
+  if (!is_win || is_clang) {
+    cflags = [ "-march=armv8.2-a+dotprod" ]
+    asmflags = cflags
+  }
+}
+
+config("march_fp16") {
+  if (!is_win || is_clang) {
+    cflags = [ "-march=armv8.2-a+fp16" ]
+    asmflags = cflags
+  }
+}
+
+config("march_dotprod_fp16") {
+  if (!is_win || is_clang) {
+    cflags = [ "-march=armv8.2-a+fp16+dotprod" ]
+    asmflags = cflags
+  }
+}
+
+config("march_dotprod_i8mm") {
+  if (!is_win || is_clang) {
+    cflags = [ "-march=armv8-a+dotprod+i8mm" ]
+    asmflags = cflags
+  }
+}
+
+config("march_dotprod_i8mm_sve") {
+  if (!is_win || is_clang) {
+    cflags = [ "-march=armv8.2-a+dotprod+i8mm+sve" ]
+    asmflags = cflags
+  }
+}
+
+config("march_i8mm") {
+  if (!is_win || is_clang) {
+    cflags = [ "-march=armv8.2-a+i8mm" ]
+    asmflags = cflags
+  }
+}
+
+config("march_i8mm_f16") {
+  if (!is_win || is_clang) {
+    cflags = [ "-march=armv8.2-a+i8mm+fp16" ]
+    asmflags = cflags
+  }
+}
+
+config("march_sve2") {
+  cflags = [ "-march=armv9-a+sve2" ]
+  asmflags = cflags
+}
+
+config("march_sme") {
+  cflags = [ "-march=armv9-a+sme" ]
+  asmflags = cflags
+}
+
 config("default_stack_frames") {
   if (!is_win) {
     if (enable_frame_pointers) {
@@ -2790,7 +3095,8 @@ config("optimize") {
     # consistency with the other platforms.
     cflags = [
                "/O2",
-               "/clang:-O2",
+               "/clang:-O3",
+               "-Xclang", "-O3",
              ] + common_optimize_on_cflags
 
     # The `-O3` for clang turns on extra optimizations compared to the standard
@@ -2803,9 +3109,9 @@ config("optimize") {
     # (AKA -Oz).
 
     if (is_fuchsia || enable_cast_receiver) {
-      cflags = [ "-Oz" ] + common_optimize_on_cflags
+      cflags = [ "-O3" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-Os" ] + common_optimize_on_cflags
+      cflags = [ "-O3" ] + common_optimize_on_cflags
     }
 
     if (is_clang && use_ml_inliner && is_a_target_toolchain && !is_chromeos &&
@@ -2820,9 +3126,9 @@ config("optimize") {
 
     # Similar to clang, we optimize with `-Copt-level=s` to keep loop
     # vectorization while otherwise optimizing for size.
-    rustflags = [ "-Copt-level=s" ]
+    rustflags = [ "-Copt-level=3" ]
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = [ "-O3" ] + common_optimize_on_cflags
 
     # The `-O3` for clang turns on extra optimizations compared to the standard
     # `-O2`. But for rust, `-Copt-level=3` is the default and is thus reliable
@@ -2874,10 +3180,13 @@ config("optimize_max") {
     # Favor speed over size, /O2 must be before the common flags.
     # /O2 implies /Ot, /Oi, and /GF.
     cflags = [ "/O2" ] + common_optimize_on_cflags
+    if (is_clang) {
+        cflags += [ "/clang:-O3", "-Xclang", "-O3", ]
+    }
   } else if (optimize_for_fuzzing) {
     cflags = [ "-O1" ] + common_optimize_on_cflags
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = [ "-O3" ] + common_optimize_on_cflags
   }
   rustflags = [ "-Copt-level=3" ]
 }
@@ -2896,10 +3205,10 @@ config("optimize_speed") {
     # /O2 implies /Ot, /Oi, and /GF.
     cflags = [ "/O2" ] + common_optimize_on_cflags
     if (is_clang) {
-      cflags += [ "/clang:-O3" ]
+        cflags += [ "/clang:-O3", "-Xclang", "-O3", ]
     }
   } else if (optimize_for_fuzzing) {
-    cflags = [ "-O1" ] + common_optimize_on_cflags
+    cflags = [ "-O2" ] + common_optimize_on_cflags
   } else {
     cflags = [ "-O3" ] + common_optimize_on_cflags
   }
@@ -2932,6 +3241,41 @@ config("default_optimization") {
   }
 }
 
+# Emit relocations for BOLT
+config("emit-relocs") {
+  if (!using_sanitizer && use_bolt) {
+    if (is_win) {
+      ldflags = [ "-mllvm:--emit-relocs", ]
+    } else {
+      ldflags = [ "-Wl,--emit-relocs", ]
+    }
+  }
+}
+
+# Use LLVM's Polly optimizer
+config("polly") {
+  if (use_polly == true) {
+    ldflags = common_optimize_on_ldflags
+    if (is_win) {
+      ldflags += [
+        "-mllvm:-polly",
+        "-mllvm:-polly-detect-profitability-min-per-loop-insts=40",
+        #"-mllvm:-polly-invariant-load-hoisting",
+        "-mllvm:-polly-run-dce",
+        "-mllvm:-polly-vectorizer=stripmine",
+      ]
+    } else {
+      ldflags += [
+        "-Wl,-mllvm,-polly",
+        "-Wl,-mllvm,-polly-detect-profitability-min-per-loop-insts=40",
+        #"-Wl,-mllvm,-polly-invariant-load-hoisting",
+        "-Wl,-mllvm,-polly-run-dce",
+        "-Wl,-mllvm,-polly-vectorizer=stripmine",
+      ]
+    }
+  }
+}
+
 _clang_sample_profile = ""
 if (is_clang && is_a_target_toolchain) {
   if (clang_sample_profile_path != "") {
@@ -3268,7 +3612,7 @@ config("minimal_symbols") {
 
     asmflags = cflags
   }
-  rustflags += [ "-Cdebuginfo=1" ]
+  rustflags += [ "-Cdebuginfo=1", "-Copt-level=3", ]
 }
 
 # This configuration contains function names only. That is, the compiler is
--- a/build/config/mac/BUILD.gn
+++ b/build/config/mac/BUILD.gn
@@ -5,6 +5,7 @@
 import("//build/config/apple/arch.gni")
 import("//build/config/apple/symbols.gni")
 import("//build/config/c++/c++.gni")
+import("//build/config/compiler_opt.gni")
 import("//build/config/mac/mac_sdk.gni")
 import("//build/config/sysroot.gni")
 import("//build/toolchain/rbe.gni")
@@ -15,6 +16,8 @@
 config("compiler") {
   # These flags are shared between the C compiler and linker.
   common_mac_flags = []
+  common_mac_cflags = []
+  common_mac_ldflags = []
 
   # CPU architecture.
   _archs_mapping_os = archs_mapping[current_os]
@@ -27,15 +30,28 @@
   }
   common_mac_flags += [ "--target=$clang_arch-apple-macos" ]
 
+  if (is_official_build) {
+    common_mac_cflags += [ "-O3", ]
+    common_mac_ldflags += [ "-Wl,-O3", ]
+  }
+
   # This is here so that all files get recompiled after an Xcode update.
   # (defines are passed via the command line, and build system rebuild things
   # when their commandline changes). Nothing should ever read this define.
   defines = [ "CR_XCODE_BUILD=$xcode_build" ]
 
   asmflags = common_mac_flags
-  cflags = common_mac_flags
+  cflags = common_mac_flags + common_mac_cflags
 
-  ldflags = common_mac_flags
+  ldflags = common_mac_flags + common_mac_ldflags
+
+  if (is_official_build) {
+    cflags += [ "-ffp-contract=fast", ]
+    ldflags += [
+      "-Wl,-mllvm,-fp-contract=fast",
+      "-Wl,-mllvm,-import-instr-limit=100",
+    ]
+  }
 
   # ld64 defaults to fixup chains on macOS 12+ when targeting arm64 and on
   # macOS 13+ when targeting x86_64.
--- a/build/config/win/BUILD.gn
+++ b/build/config/win/BUILD.gn
@@ -1,8 +1,9 @@
-# Copyright 2013 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
 import("//build/config/c++/c++.gni")
+import("//build/config/compiler_opt.gni")
 import("//build/config/chrome_build.gni")
 import("//build/config/clang/clang.gni")
 import("//build/config/compiler/compiler.gni")
@@ -120,8 +121,85 @@
     # Chrome currently requires SSE3. Clang supports targeting any Intel
     # microarchitecture. MSVC only supports a subset of architectures, and the
     # next step after SSE2 will be AVX.
+    if (current_cpu == "x86") {
+      cflags += [ "-mmmx", "-msse", "-msse2", "/clang:-msse2", ]
+      if (use_sse2) {
+        cflags += [ "/arch:SSE2", ]
+      }
+      if (use_sse3) {
+        cflags += [ "-msse3", "/clang:-msse3", ]
+      }
+    } else if (current_cpu == "x64") {
+      cflags += [ "-msse3", "/clang:-msse3", ]
+    }
+    
     if (current_cpu == "x86" || current_cpu == "x64") {
-      cflags += [ "-msse3" ]
+      if (is_official_build) {
+        cflags += [
+          "/clang:-O3",
+          "-Xclang", "-O3",
+        ]
+      }
+      if (use_sse41) {
+        cflags += [
+          "-mssse3",
+          "-msse4.1",
+          "/clang:-mssse3",
+          "/clang:-msse4.1",
+        ]
+      }
+      if (use_sse42) {
+        cflags += [
+          "-msse4.2",
+          "/clang:-msse4.2",
+          # "/arch:SSE4.2",
+        ]
+      }
+      if (use_avx) {
+        cflags += [
+          "-mpclmul",
+          "-maes",
+          "-mavx",
+          "/clang:-mpclmul",
+          "/clang:-maes",
+          "/clang:-mavx",
+          # "/arch:AVX",
+        ]
+      }
+      if (use_avx2) {
+        cflags += [
+          "-mavx2",
+          "-mfma",
+          "-mf16c",
+          "-mlzcnt",
+          "-mbmi",
+          "-mbmi2",
+          "/clang:-mavx2",
+          "/clang:-mfma",
+          "/clang:-mf16c",
+          "/clang:-mlzcnt",
+          "/clang:-mbmi",
+          "/clang:-mbmi2",
+          "/clang:-ffp-contract=fast",
+          # "/arch:AVX2",
+          # "/fp:fast",
+        ]
+      }
+      if (use_avx512) {
+        cflags += [
+          "-mavx512f",
+          "-mavx512cd",
+          "-mavx512vl",
+          "-mavx512bw",
+          "-mavx512dq",
+          "/clang:-mavx2",
+          "/clang:-mavx512f",
+          "/clang:-mavx512cd",
+          "/clang:-mavx512vl",
+          "/clang:-mavx512bw",
+          "/clang:-mavx512dq",
+        ]
+      }
     }
 
     # Enable ANSI escape codes if something emulating them is around (cmd.exe
@@ -160,6 +238,9 @@
 
     # Don't look for libpaths in %LIB%, similar to /X in cflags above.
     ldflags += [ "/lldignoreenv" ]
+    if (is_official_build) {
+      ldflags += [ "/OPT:lldlto=3", ]
+    }
   }
 
   # Some binaries create PDBs larger than 4 GiB. Increasing the PDB page size
@@ -187,11 +268,18 @@
       "/INCREMENTAL:NO",
       "/FIXED:NO",
     ]
+    if (use_avx2) {
+      ldflags += [ "-mllvm:-march=haswell", ]
+    }
+    if (use_avx512) {
+      ldflags += [ "-mllvm:-march=skylake-avx512", ]
+    }
 
     if (use_lld) {
       # String tail merging leads to smaller binaries, but they don't compress
       # as well, leading to increased mini_installer size (crbug.com/838449).
-      ldflags += [ "/OPT:NOLLDTAILMERGE" ]
+      # ldflags += [ "/OPT:NOLLDTAILMERGE" ]
+      ldflags += [ "/OPT:LLDTAILMERGE" ]
     }
 
     # TODO(siggi): Is this of any use anymore?
@@ -199,7 +287,7 @@
     # PDB file by about 5%) but does not otherwise alter the output binary. It
     # is enabled opportunistically for builds where it is not prohibited (not
     # supported when incrementally linking, or using /debug:fastlink).
-    ldflags += [ "/PROFILE" ]
+    # ldflags += [ "/PROFILE" ]
   }
 
   # arflags apply only to static_libraries. The normal linker configs are only
@@ -289,12 +377,14 @@
 # manually override this config for their compiles.
 config("winver") {
   defines = [
+    # 0x0A000010
     "NTDDI_VERSION=NTDDI_WIN11_GE",
 
     # We can't say `=_WIN32_WINNT_WIN10` here because some files do
     # `#if WINVER < 0x0600` without including windows.h before,
     # and then _WIN32_WINNT_WIN10 isn't yet known to be 0x0A00.
     "_WIN32_WINNT=0x0A00",
+    "_WIN64_WINNT=0x0A00",
     "WINVER=0x0A00",
   ]
 }

# V8 SSE2/SSE3/AVX flags
--- a/v8/BUILD.gn
+++ b/v8/BUILD.gn
@@ -1,9 +1,10 @@
-# Copyright 2014 The Chromium Authors. All rights reserved.
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
 import("//build/config/android/config.gni")
 import("//build/config/arm.gni")
+import("//build/config/compiler_opt.gni")
 import("//build/config/coverage/coverage.gni")
 import("//build/config/dcheck_always_on.gni")
 import("//build/config/host_byteorder.gni")
@@ -1627,7 +1628,31 @@ config("toolchain") {
     defines += [ "V8_TARGET_ARCH_IA32" ]
     if (is_win) {
       # Ensure no surprising artifacts from 80bit double math with x86.
-      cflags += [ "/arch:SSE2" ]
+      if (use_sse2 && !use_sse3) {
+        cflags += [ "/arch:SSE2", ]
+      }
+      if (use_sse3 && !use_sse41) {
+        cflags += [ "/clang:-msse3", ]
+      }
+      if (use_sse41 && !use_sse42) {
+        cflags += [ "/clang:-mssse3", "/clang:-msse4.1", ]
+      }
+      if (use_sse42) {
+        cflags += [ "/clang:-msse4.2", ]
+      }
+    } else {
+      if (use_sse2 && !use_sse3) {
+        cflags += [ "-msse2", ]
+      }
+      if (use_sse3 && !use_sse41) {
+        cflags += [ "-msse3", ]
+      }
+      if (use_sse41 && !use_sse42) {
+        cflags += [ "-mssse3", "-msse4.1", ]
+      }
+      if (use_sse42) {
+        cflags += [ "-msse4.2", ]
+      }
     }
   }
   if (v8_current_cpu == "x64") {
@@ -1636,9 +1661,65 @@ config("toolchain") {
       # Increase the initial stack size. The default is 1MB, this is 2MB. This
       # applies only to executables and shared libraries produced by V8 since
       # ldflags are not pushed to dependants.
+      cflags += [ "/clang:-msse3", ]
       ldflags += [ "/STACK:2097152" ]
+      if (use_sse41) {
+        cflags += [ "/clang:-mssse3", "/clang:-msse4.1", ]
+      }
+      if (use_sse42) {
+        cflags += [ "/clang:-msse4.2", ]
+      }
+      if (use_avx) {
+        cflags += [ "/clang:-mpclmul", "/clang:-maes", "/clang:-mavx", ]
+      }
+      if (use_avx2) {
+        cflags += [ "/clang:-mfma", "/clang:-ffp-contract=fast", "/clang:-mavx2", "/clang:-mf16c", "/clang:-mlzcnt", "/clang:-mbmi", "/clang:-mbmi2", ]
+      }
+      if (use_fma) {
+        cflags += [ "/clang:-ffp-contract=fast", ]
+      }
+      if (use_avx512) {
+        cflags += [
+          "/clang:-mavx512f", "/clang:-mavx512cd", "/clang:-mavx512vl", "/clang:-mavx512bw", "/clang:-mavx512dq",
+        ]
+      }
+    } else {
+      if (current_cpu == "x64") {
+        # Non-Windows SIMD flags
+        cflags += [ "-msse3", ]
+        if (use_sse41) {
+          cflags += [ "-mssse3", "-msse4.1", ]
+        }
+        if (use_sse42) {
+          cflags += [ "-msse4.2", ]
+        }
+        if (use_avx) {
+          cflags += [ "-mpclmul", "-maes", "-mavx", ]
+        }
+        if (use_avx2) {
+          cflags += [ "-mavx2", "-mf16c", "-mlzcnt", "-mbmi", "-mbmi2", ]
+        }
+        if (use_fma) {
+          cflags += [ "-mfma", "-ffp-contract=fast", ]
+        }
+        if (use_avx512) {
+          cflags += [ "-mavx512f", "-mavx512cd", "-mavx512vl", "-mavx512bw", "-mavx512dq", ]
+        }
+      }
     }
   }
+  if (is_linux || is_chromeos || is_android || is_mac || is_win) {
+    # Full optimization for V8.
+    if (is_official_build) {
+      if (is_win) {
+        cflags += [ "/O2", "-Xclang", "-O3",  ]
+      } else {
+        cflags += [ "-O3", ]
+        ldflags += [ "-Wl,-O3", ]
+      }
+    }
+  }
+
   if (is_android && v8_android_log_stdout) {
     defines += [ "V8_ANDROID_LOG_STDOUT" ]
   }
@@ -2590,9 +2671,9 @@ template("run_mksnapshot") {
         "--turbo-profiling-input",
         rebase_path(v8_builtins_profiling_log_file, root_build_dir),
 
-        # Replace this with --warn-about-builtin-profile-data to see the full
-        # list of builtins with incompatible profiles.
-        "--abort-on-bad-builtin-profile-data",
+        # TODO(crbug.com/v8/13647): Do not fail for invalid profiles
+        # "--abort-on-bad-builtin-profile-data",
+        "--warn-about-builtin-profile-data",
       ]
 
       if (!v8_enable_builtins_profiling && v8_enable_builtins_reordering) {
# LLVM/clang AVX/AES/Polly flags
--- a/tools/clang/scripts/build.py
+++ b/tools/clang/scripts/build.py
@@ -73,6 +73,8 @@ BUG_REPORT_URL = ('https://crbug.com in the Tools>LLVM component,'
 LIBXML2_VERSION = 'libxml2-v2.9.12'
 ZSTD_VERSION = 'zstd-1.5.5'
 
+MIMALLOC_URL = 'https://github.com/microsoft/mimalloc/archive/refs/tags/v3.1.4.tar.gz'
+
 win_sdk_dir = None
 def GetWinSDKDir():
   """Get the location of the current SDK."""
@@ -165,7 +167,7 @@ def CheckoutGitRepo(name, git_url, commit, dir):
         ['git', 'diff-index', '--exit-code', 'HEAD'], fail_hard=False)
         and RunCommand(['git', 'fetch'], fail_hard=False)
         and RunCommand(['git', 'checkout', commit], fail_hard=False)
-        and RunCommand(['git', 'clean', '-f'], fail_hard=False)):
+        and RunCommand(['git', 'clean', '-ffd'], fail_hard=False)):
       return
 
     # If we can't use the current repo, delete it.
@@ -383,6 +385,13 @@ def BuildLibXml2():
           '-DCMAKE_INSTALL_LIBDIR=lib',
           '-DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded',  # /MT to match LLVM.
           '-DBUILD_SHARED_LIBS=OFF',
+          '-DCMAKE_C_COMPILER=' + os.path.join(PINNED_CLANG_DIR, 'bin', 'clang'),
+          '-DCMAKE_CXX_COMPILER=' + os.path.join(PINNED_CLANG_DIR, 'bin', 'clang++'),
+          '-DCMAKE_C_FLAGS_RELEASE=-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -w ',
+          '-DCMAKE_CXX_FLAGS_RELEASE=-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -w ',
+          '-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -w ',
+          '-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -w ',
+          '-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -w ',
           '-DLIBXML2_WITH_C14N=OFF',
           '-DLIBXML2_WITH_CATALOG=OFF',
           '-DLIBXML2_WITH_DEBUG=OFF',
@@ -442,6 +451,52 @@ def BuildLibXml2():
 
   return extra_cmake_flags, extra_cflags
 
+class LibMimallocDirs:
+  """
+  The set of directories where mimalloc is located.
+
+  Includes the diractories where the source is unpacked, where it is built,
+  and installed.
+  """
+  def __init__(self):
+    self.unzip_dir = LLVM_BUILD_TOOLS_DIR
+    self.src_dir = os.path.join(self.unzip_dir, 'mimalloc-3.1.4')
+    self.build_dir = os.path.join(self.src_dir, 'build')
+
+def BuildLibMimalloc():
+  """Download and build mimalloc"""
+
+  dirs = LibMimallocDirs()
+  if os.path.exists(dirs.src_dir):
+    RmTree(dirs.src_dir)
+  DownloadAndUnpack(MIMALLOC_URL, dirs.unzip_dir)
+  os.mkdir(dirs.build_dir)
+  os.chdir(dirs.build_dir)
+
+  RunCommand(
+    [
+        'cmake',
+        '-GNinja',
+        '-DCMAKE_BUILD_TYPE=Release',
+        '-DCMAKE_C_COMPILER=' + os.path.join(PINNED_CLANG_DIR, 'bin', 'clang'),
+        '-DCMAKE_CXX_COMPILER=' + os.path.join(PINNED_CLANG_DIR, 'bin', 'clang++'),
+        '-DCMAKE_C_FLAGS_RELEASE=-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -w ',
+        '-DCMAKE_CXX_FLAGS_RELEASE=-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -w ',
+        '-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -w ',
+        '-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -w ',
+        '-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -w ',
+        '-DMI_OVERRIDE=ON',
+        '-DMI_SHOW_ERRORS=OFF',
+        '..',
+    ],
+    setenv=True)
+  RunCommand(['ninja'], setenv=True)
+
+  MIMALLOC_LIB = os.path.abspath(os.path.join(LLVM_DIR, '..', 'llvm-build-tools/mimalloc-3.1.4')) + '/build/mimalloc.o'
+
+  mimalloc_ldflags = [ '' + MIMALLOC_LIB ]
+
+  return mimalloc_ldflags
 
 class ZStdDirs:
   """
@@ -491,6 +546,13 @@ def BuildZStd():
           '-DCMAKE_INSTALL_LIBDIR=lib',
           '-DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded',  # /MT to match LLVM.
           '-DZSTD_BUILD_SHARED=OFF',
+          '-DCMAKE_C_COMPILER=' + os.path.join(PINNED_CLANG_DIR, 'bin', 'clang'),
+          '-DCMAKE_CXX_COMPILER=' + os.path.join(PINNED_CLANG_DIR, 'bin', 'clang++'),
+          '-DCMAKE_C_FLAGS_RELEASE=-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -w ',
+          '-DCMAKE_CXX_FLAGS_RELEASE=-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -w ',
+          '-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -w ',
+          '-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -w ',
+          '-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -w ',
           '../build/cmake',
       ],
       setenv=True)
@@ -737,21 +799,9 @@ def main():
                       action='store_true',
                       help='use the cmake from PATH instead of downloading '
                       'and using prebuilt cmake binaries')
-  parser.add_argument('--tf-path',
-                      help='path to python tensorflow pip package. '
-                      'Used for embedding an MLGO model')
-  parser.add_argument(
-      '--with-ml-inliner-model',
-      help='path to MLGO inliner model to embed. Setting to '
-      '\'default\', will download an official model which was '
-      'trained for Chrome on Android',
-      default='default' if sys.platform.startswith('linux') else '')
   parser.add_argument('--with-android', type=gn_arg, nargs='?', const=True,
                       help='build the Android ASan runtime (linux only)',
                       default=sys.platform.startswith('linux'))
-  parser.add_argument('--pic',
-                      action='store_true',
-                      help='Uses PIC when building LLVM')
   parser.add_argument('--with-fuchsia',
                       type=gn_arg,
                       nargs='?',
@@ -768,10 +818,6 @@ def main():
   parser.add_argument('--with-ccache',
                       action='store_true',
                       help='Use ccache to build the stage 1 compiler')
-  parser.add_argument('--without-zstd',
-                      dest='with_zstd',
-                      action='store_false',
-                      help='Disable zstd in the build')
   parser.add_argument('--preserve-gcs-signature',
                       action='store_true',
                       help='By default, this script removes gcs hash files '
@@ -784,8 +830,14 @@ def main():
 
   global CLANG_REVISION, PACKAGE_VERSION, LLVM_BUILD_DIR, STAMP_FILE, FORCE_HEAD_REVISION_FILE
 
-  if (args.pgo or args.thinlto) and not args.bootstrap:
-    print('--pgo/--thinlto requires --bootstrap')
+  if not sys.platform.startswith('linux'):
+    print('Customized build script only works with Linux')
+    return 1
+  if not args.bootstrap:
+    print('Customized build script requires --bootstrap')
+    return 1
+  if args.skip_checkout:
+    print('Customized build script does not support --skip-checkout')
     return 1
   if args.with_android and not os.path.exists(ANDROID_NDK_DIR):
     print('Android NDK not found at ' + ANDROID_NDK_DIR)
@@ -810,10 +862,6 @@ def main():
     print('for general Fuchsia build instructions.')
     return 1
 
-  if args.with_ml_inliner_model and not sys.platform.startswith('linux'):
-    print('--with-ml-inliner-model only supports linux hosts')
-    return 1
-
   # Don't buffer stdout, so that print statements are immediately flushed.
   # LLVM tests print output without newlines, so with buffering they won't be
   # immediately printed.
@@ -886,12 +934,12 @@ def main():
   # LLVM_ENABLE_LLD).
   cc, cxx, lld = None, None, None
 
-  cflags = []
-  cxxflags = []
-  ldflags = []
+  cflags = ['-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -pthread -w']
+  cxxflags = ['-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -pthread -w']
+  ldflags = ['-fuse-ld=lld  -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -pthread -w']
 
   targets = 'AArch64;ARM;LoongArch;Mips;PowerPC;RISCV;SystemZ;WebAssembly;X86'
-  projects = 'clang;lld'
+  projects = 'clang;lld;polly'
   if not args.no_tools:
     projects += ';clang-tools-extra'
   if args.bolt:
@@ -904,7 +952,6 @@ def main():
     runtimes = 'compiler-rt'
 
   pic_default = sys.platform == 'win32'
-  pic_mode = 'ON' if args.pic or pic_default else 'OFF'
 
   base_cmake_args = [
       '-GNinja',
@@ -913,8 +960,8 @@ def main():
       f'-DLLVM_ENABLE_PROJECTS={projects}',
       f'-DLLVM_ENABLE_RUNTIMES={runtimes}',
       f'-DLLVM_TARGETS_TO_BUILD={targets}',
-      f'-DLLVM_ENABLE_PIC={pic_mode}',
-      '-DLLVM_ENABLE_TERMINFO=OFF',
+      f'-DLLVM_ENABLE_PIC=ON',
+      # '-DLLVM_ENABLE_TERMINFO=OFF',
       '-DLLVM_ENABLE_Z3_SOLVER=OFF',
       '-DCLANG_PLUGIN_SUPPORT=OFF',
       '-DCLANG_ENABLE_STATIC_ANALYZER=OFF',
@@ -938,7 +985,8 @@ def main():
       # `bootstrap` tool:
       # https://github.com/rust-lang/rust/blob/021861aea8de20c76c7411eb8ada7e8235e3d9b5/src/bootstrap/src/core/build_steps/llvm.rs#L348
       '-DLLVM_INSTALL_UTILS=ON',
-      '-DLLVM_ENABLE_ZSTD=%s' % ('ON' if args.with_zstd else 'OFF'),
+      '-DLLVM_ENABLE_ZSTD=ON',
+      '-DLLVM_ENABLE_BINDINGS=OFF'
   ]
 
   if sys.platform == 'darwin':
@@ -1001,6 +1049,10 @@ def main():
     base_cmake_args.append('-DLLVM_WINSYSROOT="%s"' %
                            os.path.dirname(os.path.dirname(GetWinSDKDir())))
 
+  #Build mimalloc
+  libmimalloc_ldflags = BuildLibMimalloc()
+  ldflags += libmimalloc_ldflags
+
   # Statically link libxml2 to make lld-link not require mt.exe on Windows,
   # and to make sure lld-link output on other platforms is identical to
   # lld-link on Windows (for cross-builds).
@@ -1010,13 +1062,12 @@ def main():
   cflags += libxml_cflags
   cxxflags += libxml_cflags
 
-  if args.with_zstd:
-    # Statically link zstd to make lld support zstd compression for debug info.
-    with timer.time('zstd build'):
-      zstd_cmake_args, zstd_cflags = BuildZStd()
-    base_cmake_args += zstd_cmake_args
-    cflags += zstd_cflags
-    cxxflags += zstd_cflags
+  # Statically link zstd to make lld support zstd compression for debug info.
+  with timer.time('zstd build'):
+    zstd_cmake_args, zstd_cflags = BuildZStd()
+  base_cmake_args += zstd_cmake_args
+  cflags += zstd_cflags
+  cxxflags += zstd_cflags
 
   lit_excludes = []
   if sys.platform.startswith('linux'):
@@ -1077,7 +1128,7 @@ def main():
       bootstrap_targets += ';ARM;AArch64'
     bootstrap_args = base_cmake_args + ccache_cmake_args + [
         '-DLLVM_TARGETS_TO_BUILD=' + bootstrap_targets,
-        '-DLLVM_ENABLE_PROJECTS=clang;lld',
+        '-DLLVM_ENABLE_PROJECTS=clang;lld;polly',
         '-DLLVM_ENABLE_RUNTIMES=' + ';'.join(runtimes),
         '-DCMAKE_INSTALL_PREFIX=' + LLVM_BOOTSTRAP_INSTALL_DIR,
         '-DCMAKE_C_FLAGS=' + ' '.join(cflags),
@@ -1135,6 +1186,9 @@ def main():
 
     print('Bootstrap compiler installed.')
 
+  cflags += ['-mllvm -polly']
+  cxxflags += ['-mllvm -polly']
+
   if args.pgo:
     print('Building instrumented compiler')
     if os.path.exists(LLVM_INSTRUMENTED_DIR):
@@ -1148,9 +1202,9 @@ def main():
     cxxflags += ['-Wno-backend-plugin']
 
     instrument_args = base_cmake_args + [
-        '-DLLVM_ENABLE_PROJECTS=clang',
-        '-DCMAKE_C_FLAGS=' + ' '.join(cflags),
-        '-DCMAKE_CXX_FLAGS=' + ' '.join(cxxflags),
+        '-DLLVM_ENABLE_PROJECTS=clang;polly',
+        '-DCMAKE_C_FLAGS= -mllvm -vp-counters-per-site=3 ' + ' '.join(cflags),
+        '-DCMAKE_CXX_FLAGS= -mllvm -vp-counters-per-site=3 ' + ' '.join(cxxflags),
         '-DCMAKE_EXE_LINKER_FLAGS=' + ' '.join(ldflags),
         '-DCMAKE_SHARED_LINKER_FLAGS=' + ' '.join(ldflags),
         '-DCMAKE_MODULE_LINKER_FLAGS=' + ' '.join(ldflags),
@@ -1196,8 +1250,11 @@ def main():
       with open(training_source, 'wb') as f:
         DownloadUrl(CDS_URL + '/' + training_source, f)
       train_cmd = [os.path.join(LLVM_INSTRUMENTED_DIR, 'bin', 'clang++'),
-                  '-target', 'x86_64-unknown-unknown', '-O2', '-g', '-std=c++20',
-                   '-fno-exceptions', '-fno-rtti', '-w', '-c', training_source]
+                  '-target', 'x86_64-unknown-unknown', '-O3', '-g', '-std=c++20',
+                 '-march=x86-64-v3', '-ffp-contract=fast', '-fmerge-all-constants',
+                 '-fdata-sections', '-ffunction-sections', '-fno-exceptions',
+                 '-fno-rtti', '-mllvm', '-polly', '-mllvm', '-polly-process-unprofitable',
+                 '-Rpass-analysis=polly', '-w', '-c', training_source]
       if sys.platform == 'darwin':
         train_cmd.extend(['-isysroot', isysroot])
       RunCommand(train_cmd, setenv=True)
@@ -1248,13 +1305,18 @@ def main():
   if cxx is not None: base_cmake_args.append('-DCMAKE_CXX_COMPILER=' + cxx)
   if lld is not None: base_cmake_args.append('-DCMAKE_LINKER=' + lld)
   final_install_dir = args.install_dir if args.install_dir else LLVM_BUILD_DIR
+  tf_path = subprocess.check_output(
+    ['vpython3', os.path.join(THIS_DIR, 'get_tensorflow.py')],
+    universal_newlines=True).rstrip()
+  base_cmake_args += [
+    '-DTENSORFLOW_AOT_PATH=%s' % tf_path,
+  ]
   cmake_args = base_cmake_args + [
-      '-DCMAKE_C_FLAGS=' + ' '.join(cflags),
-      '-DCMAKE_CXX_FLAGS=' + ' '.join(cxxflags),
-      '-DCMAKE_EXE_LINKER_FLAGS=' + ' '.join(ldflags),
-      '-DCMAKE_SHARED_LINKER_FLAGS=' + ' '.join(ldflags),
-      '-DCMAKE_MODULE_LINKER_FLAGS=' + ' '.join(ldflags),
       '-DCMAKE_INSTALL_PREFIX=' + final_install_dir,
+      '-DLLVM_INLINER_MODEL_CURRENT_URL=https://commondatastorage.googleapis.com/chromium-browser-clang/tools/mlgo_model3.tgz',
+      '-DLLVM_INLINER_MODEL_PATH=download',
+      '-DLLVM_RAEVICT_MODEL_CURRENT_URL=https://github.com/google/ml-compiler-opt/releases/download/regalloc-evict-v1.1/model.zip',
+      '-DLLVM_RAEVICT_MODEL_PATH=download',
   ]
   if not args.no_tools:
     cmake_args.extend([
@@ -1263,10 +1325,36 @@ def main():
         os.path.join(CHROMIUM_DIR, 'tools', 'clang'),
         '-DCHROMIUM_TOOLS=%s' % ';'.join(chrome_tools)
     ])
-  if args.pgo:
+  if args.pgo and not args.thinlto:
     cmake_args.append('-DLLVM_PROFDATA_FILE=' + LLVM_PROFDATA_FILE)
-  if args.thinlto:
+    cmake_args.append('-DCMAKE_C_FLAGS=-mllvm -enable-ext-tsp-block-placement ' + ' '.join(cflags))
+    cmake_args.append('-DCMAKE_CXX_FLAGS=-mllvm -enable-ext-tsp-block-placement ' + ' '.join(cxxflags))
+    cmake_args.append('-DCMAKE_EXE_LINKER_FLAGS=-Wl,-z,keep-text-section-prefix ' + ' '.join(ldflags))
+    cmake_args.append('-DCMAKE_SHARED_LINKER_FLAGS=-Wl,-z,keep-text-section-prefix ' + ' '.join(ldflags))
+    cmake_args.append('-DCMAKE_MODULE_LINKER_FLAGS=-Wl,-z,keep-text-section-prefix ' + ' '.join(ldflags))
+  if args.thinlto and not args.pgo:
     cmake_args.append('-DLLVM_ENABLE_LTO=Thin')
+    cmake_args.append('-DLLVM_PARALLEL_LINK_JOBS=16')
+    cmake_args.append('-DCMAKE_C_FLAGS=-flto=thin -fwhole-program-vtables -fsplit-lto-unit ' + ' '.join(cflags))
+    cmake_args.append('-DCMAKE_CXX_FLAGS=-flto=thin -fwhole-program-vtables -fsplit-lto-unit ' + ' '.join(cxxflags))
+    cmake_args.append('-DCMAKE_EXE_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-polly ' + ' '.join(ldflags))
+    cmake_args.append('-DCMAKE_SHARED_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-polly ' + ' '.join(ldflags))
+    cmake_args.append('-DCMAKE_MODULE_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-polly ' + ' '.join(ldflags))
+  if args.thinlto and args.pgo:
+    cmake_args.append('-DLLVM_PROFDATA_FILE=' + LLVM_PROFDATA_FILE)
+    cmake_args.append('-DLLVM_ENABLE_LTO=Thin')
+    cmake_args.append('-DLLVM_PARALLEL_LINK_JOBS=16')
+    cmake_args.append('-DCMAKE_C_FLAGS=-flto=thin -fwhole-program-vtables -fsplit-lto-unit ' + ' '.join(cflags))
+    cmake_args.append('-DCMAKE_CXX_FLAGS=-flto=thin -fwhole-program-vtables -fsplit-lto-unit ' + ' '.join(cxxflags))
+    cmake_args.append('-DCMAKE_EXE_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-enable-ext-tsp-block-placement -Wl,-z,keep-text-section-prefix -Wl,-mllvm,-polly ' + ' '.join(ldflags))
+    cmake_args.append('-DCMAKE_SHARED_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-enable-ext-tsp-block-placement -Wl,-z,keep-text-section-prefix -Wl,-mllvm,-polly ' + ' '.join(ldflags))
+    cmake_args.append('-DCMAKE_MODULE_LINKER_FLAGS=-flto=thin -fwhole-program-vtables -Wl,--lto-O3 -Wl,-mllvm,-import-instr-limit=5 -Wl,-mllvm,-enable-ext-tsp-block-placement -Wl,-z,keep-text-section-prefix -Wl,-mllvm,-polly ' + ' '.join(ldflags))
+  else:
+    cmake_args.append('-DCMAKE_C_FLAGS= ' + ' '.join(cflags))
+    cmake_args.append('-DCMAKE_CXX_FLAGS= ' + ' '.join(cxxflags))
+    cmake_args.append('-DCMAKE_EXE_LINKER_FLAGS= ' + ' '.join(ldflags))
+    cmake_args.append('-DCMAKE_SHARED_LINKER_FLAGS= ' + ' '.join(ldflags))
+    cmake_args.append('-DCMAKE_MODULE_LINKER_FLAGS= ' + ' '.join(ldflags))
   if sys.platform == 'win32':
     cmake_args.append('-DLLVM_ENABLE_ZLIB=FORCE_ON')
 
@@ -1308,6 +1396,11 @@ def main():
             # TODO(crbug.com/40242553): pass proper flags to i386 tests so they
             # compile correctly
             'LLVM_INCLUDE_TESTS=OFF',
+            'CMAKE_C_FLAGS=-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -mllvm -polly -w ',
+            'CMAKE_CXX_FLAGS=-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -mllvm -polly -w ',
+            'CMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly -w ',
+            'CMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly -w ',
+            'CMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly -w '
         ],
         "profile":
         True,
@@ -1317,6 +1410,12 @@ def main():
     runtimes_triples_args['x86_64-unknown-linux-gnu'] = {
         "args": [
             'CMAKE_SYSROOT=%s' % sysroot_amd64,
+            'CMAKE_C_FLAGS=-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -mllvm -polly -w ',
+            'CMAKE_CXX_FLAGS=-O3 -march=x86-64-v3 -fdata-sections -ffunction-sections -mllvm -polly -w ',
+            'CMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly -w ',
+            'CMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly -w ',
+            'CMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld -Wl,-O2 -Wl,--gc-sections -Wl,--icf=all -mllvm -polly -w ',
+
         ],
         "profile": True,
         "sanitizers": True,
@@ -1500,32 +1599,6 @@ def main():
           "profile": build_profile
       }
 
-  # Embed MLGO inliner model. If tf_path is not specified, a vpython3 env
-  # will be created which contains the necessary source files for compilation.
-  # MLGO is only officially supported on linux. This condition is checked at
-  # the top of main()
-  if args.with_ml_inliner_model:
-    if args.with_ml_inliner_model == 'default':
-      model_path = ('https://commondatastorage.googleapis.com/'
-                    'chromium-browser-clang/tools/mlgo_model3.tgz')
-    else:
-      model_path = args.with_ml_inliner_model
-    if not args.tf_path:
-      with timer.time('get tensorflow'):
-        tf_path = subprocess.check_output(
-            ['vpython3', os.path.join(THIS_DIR, 'get_tensorflow.py')],
-            universal_newlines=True).rstrip()
-    else:
-      tf_path = args.tf_path
-    print('Embedding MLGO inliner model at %s using Tensorflow at %s' %
-          (model_path, tf_path))
-    cmake_args += [
-        '-DLLVM_INLINER_MODEL_PATH=%s' % model_path,
-        '-DTENSORFLOW_AOT_PATH=%s' % tf_path,
-        # Disable Regalloc model generation since it is unused
-        '-DLLVM_RAEVICT_MODEL_PATH=none'
-    ]
-
   # Convert FOO=BAR CMake flags per triple into
   # -DBUILTINS_$triple_FOO=BAR/-DRUNTIMES_$triple_FOO=BAR and build up
   # -DLLVM_BUILTIN_TARGETS/-DLLVM_RUNTIME_TARGETS.
@@ -1584,7 +1657,7 @@ def main():
     with timer.time('bolt instrument'):
       RunCommand([
           'bin/llvm-bolt', 'bin/clang', '-o', 'bin/clang-bolt.inst',
-          '-instrument', '--instrumentation-file-append-pid',
+          '-lite=false', '-instrument', '--instrumentation-file-append-pid',
           '--instrumentation-file=' +
           os.path.join(bolt_profiles_dir, 'prof.fdata')
       ])
@@ -1597,14 +1670,18 @@ def main():
     # Train by building a part of Clang.
     os.mkdir('bolt-training')
     os.chdir('bolt-training')
+    if args.pgo:
+      cflags += ['-mllvm -enable-ext-tsp-block-placement']
+      cxxflags += ['-mllvm -enable-ext-tsp-block-placement']
+      ldflags += ['-Wl,-z,keep-text-section-prefix']
     bolt_train_cmake_args = base_cmake_args + [
         '-DLLVM_TARGETS_TO_BUILD=X86',
         '-DLLVM_ENABLE_PROJECTS=clang',
-        '-DCMAKE_C_FLAGS=' + ' '.join(cflags),
-        '-DCMAKE_CXX_FLAGS=' + ' '.join(cxxflags),
-        '-DCMAKE_EXE_LINKER_FLAGS=' + ' '.join(ldflags),
-        '-DCMAKE_SHARED_LINKER_FLAGS=' + ' '.join(ldflags),
-        '-DCMAKE_MODULE_LINKER_FLAGS=' + ' '.join(ldflags),
+        '-DCMAKE_C_FLAGS=-g -fmerge-all-constants -Rpass-analysis=polly ' + ' '.join(cflags),
+        '-DCMAKE_CXX_FLAGS=-g -fmerge-all-constants -Rpass-analysis=polly ' + ' '.join(cxxflags),
+        '-DCMAKE_EXE_LINKER_FLAGS= ' + ' '.join(ldflags),
+        '-DCMAKE_SHARED_LINKER_FLAGS= ' + ' '.join(ldflags),
+        '-DCMAKE_MODULE_LINKER_FLAGS= ' + ' '.join(ldflags),
         '-DCMAKE_C_COMPILER=' +
         os.path.join(LLVM_BUILD_DIR, 'bin/clang-bolt.inst'),
         '-DCMAKE_CXX_COMPILER=' +
@@ -1613,6 +1690,8 @@ def main():
         os.path.join(LLVM_BUILD_DIR, 'bin/clang-bolt.inst'),
         '-DCMAKE_ASM_COMPILER_ID=Clang',
     ]
+    if args.pgo:
+      bolt_train_cmake_args.append('-DLLVM_PROFDATA_FILE=' + LLVM_PROFDATA_FILE)
     with timer.time('bolt training cmake'):
       RunCommand(['cmake'] + bolt_train_cmake_args +
                  [os.path.join(LLVM_DIR, 'llvm')])
@@ -1632,9 +1711,15 @@ def main():
       ])
       RunCommand([
           'bin/llvm-bolt', 'bin/clang', '-o', 'bin/clang-bolt.opt', '-data',
-          'merged.fdata', '-reorder-blocks=ext-tsp', '-reorder-functions=hfsort+',
-          '-split-functions', '-split-all-cold', '-split-eh', '-dyno-stats',
-          '-icf=1', '-use-gnu-stack', '-use-old-text'
+          'merged.fdata', '-reorder-blocks=ext-tsp', '-reorder-functions=cdsort',
+          '-split-functions', '-split-strategy=cdsplit',
+          '-split-all-cold', '-split-eh', '-dyno-stats',
+          '-icf=all', '-use-gnu-stack', '-peepholes=all', '-elim-link-veneers',
+          '-group-stubs', '-align-blocks', '-sctc-mode=heuristic',
+          '-jump-tables=aggressive', '-simplify-rodata-loads',
+          '-tail-duplication=cache', '-indirect-call-promotion=all', '-icp-eliminate-loads',
+          '-hot-data', '-x86-strip-redundant-address-size', '-reorder-functions-use-hot-size',
+          '-reorder-data-algo=funcs', '-lite=false',
       ])
 
     # Overwrite clang, preserving its timestamp so ninja doesn't rebuild it.
@@ -1643,8 +1728,7 @@ def main():
 
   VerifyVersionOfBuiltClangMatchesVERSION()
   VerifyZlibSupport()
-  if args.with_zstd:
-    VerifyZStdSupport()
+  VerifyZStdSupport()
 
   # Run tests.
   if (chrome_tools and (args.run_tests or args.llvm_force_head_revision)):
@@ -1667,6 +1751,9 @@ def main():
                  FORCE_HEAD_REVISION_FILE,
                  preserve_hash_files=args.preserve_gcs_signature)
 
+  RunCommand(['ln', '--symbolic', '--relative', '-f', 'bin/clang', 'bin/clang++'])
+  RunCommand(['ln', '--symbolic', '--relative', '-f', 'bin/clang', 'bin/clang-cl'])
+
   print('Clang build was successful.')
 
   return 0
