# Add LOAD_MINIMAL_HEADERS flag
--- a/net/base/load_flags_list.h
+++ b/net/base/load_flags_list.h
@@ -115,3 +115,9 @@
 // This flag is used to bypass HSTS upgrades. This flag must be set for AIA,
 // CRL, and OCSP requests in order to prevent circular dependencies.
 LOAD_FLAG(SHOULD_BYPASS_HSTS, 1 << 19)
+
+// This load will not send Accept-Language or User-Agent headers, and not
+// advertise brotli or zstd encoding.
+// Used to comply with IETF (draft) DNS-over-HTTPS:
+// "Implementors SHOULD NOT set non-essential HTTP headers in DoH client requests."
+LOAD_FLAG(MINIMAL_HEADERS, 1 << 20)
# Thorium DNS config override
--- a/net/dns/dns_client.cc
+++ b/net/dns/dns_client.cc
@@ -10,6 +10,7 @@
 #include <string>
 #include <utility>
 
+#include "base/command_line.h"
 #include "base/functional/bind.h"
 #include "base/logging.h"
 #include "base/memory/raw_ptr.h"
@@ -88,6 +89,8 @@
 }
 
 class DnsClientImpl : public DnsClient {
+ const bool disable_thorium_dns_config =
+     base::CommandLine::ForCurrentProcess()->HasSwitch("disable-thorium-dns-config");
  public:
   DnsClientImpl(NetLog* net_log, const RandIntCallback& rand_int_callback)
       : net_log_(net_log), rand_int_callback_(rand_int_callback) {}
@@ -250,11 +253,14 @@
  private:
   std::optional<DnsConfig> BuildEffectiveConfig() const {
     DnsConfig config;
-    if (config_overrides_.OverridesEverything()) {
+    // In Thorium it is sufficient to have secure DoH enabled to give the overrides priority
+    if (config_overrides_.dns_over_https_config && config_overrides_.secure_dns_mode && !disable_thorium_dns_config) {
       config = config_overrides_.ApplyOverrides(DnsConfig());
     } else {
-      if (!system_config_)
+      if (!system_config_) {
+        LOG(WARNING) << "dns_client.cc->BuildEffectiveConfig(): System configuration not set: No system_config_ ";
         return std::nullopt;
+      }
 
       config = config_overrides_.ApplyOverrides(system_config_.value());
     }
@@ -269,8 +275,10 @@
     if (config.unhandled_options)
       config.nameservers.clear();
 
-    if (!config.IsValid())
+    if (!config.IsValid()) {
+      DLOG(WARNING) << "dns_client.cc->BuildEffectiveConfig(): Invalid configuration";
       return std::nullopt;
+    }
 
     return config;
   }
# LOAD_MINIMAL_HEADERS in DoH
--- a/net/dns/dns_transaction.cc
+++ b/net/dns/dns_transaction.cc
@@ -484,7 +484,7 @@
     // avoid deadlock and enable the use of preconfigured IP addresses.
     request_->SetSecureDnsPolicy(SecureDnsPolicy::kBootstrap);
     request_->SetLoadFlags(request_->load_flags() | LOAD_DISABLE_CACHE |
-                           LOAD_BYPASS_PROXY);
+                           LOAD_MINIMAL_HEADERS | LOAD_BYPASS_PROXY);
     request_->set_allow_credentials(false);
     request_->set_isolation_info(isolation_info);
   }
# Remove UA/Referer with MINIMAL_HEADERS
--- a/net/url_request/url_request_http_job.cc
+++ b/net/url_request/url_request_http_job.cc
@@ -516,18 +516,22 @@
   // fields in the referrer.
   GURL referrer(request_->referrer());
 
-  // Our consumer should have made sure that this is a safe referrer (e.g. via
-  // URLRequestJob::ComputeReferrerForPolicy).
-  if (referrer.is_valid()) {
-    std::string referer_value = referrer.spec();
-    request_info_.extra_headers.SetHeader(HttpRequestHeaders::kReferer,
-                                          referer_value);
-  }
-
-  request_info_.extra_headers.SetHeaderIfMissing(
-      HttpRequestHeaders::kUserAgent,
-      http_user_agent_settings_ ?
-          http_user_agent_settings_->GetUserAgent() : std::string());
+  if (!(request_info_.load_flags & LOAD_MINIMAL_HEADERS)) {
+    // Our consumer should have made sure that this is a safe referrer (e.g. via
+    // URLRequestJob::ComputeReferrerForPolicy).
+    if (referrer.is_valid()) {
+      std::string referer_value = referrer.spec();
+      request_info_.extra_headers.SetHeader(HttpRequestHeaders::kReferer,
+                                            referer_value);
+    }
+  }
+
+  if (!(request_info_.load_flags & LOAD_MINIMAL_HEADERS)) {
+    request_info_.extra_headers.SetHeaderIfMissing(
+        HttpRequestHeaders::kUserAgent,
+        http_user_agent_settings_ ?
+            http_user_agent_settings_->GetUserAgent() : std::string());
+  }
 
   AddExtraHeaders();
 
@@ -773,10 +777,10 @@
 void URLRequestHttpJob::AddExtraHeaders() {
   request_info_.extra_headers.SetAcceptEncodingIfMissing(
       request()->url(), request()->accepted_stream_types(),
-      request()->context()->enable_brotli(),
-      request()->context()->enable_zstd());
-
-  if (http_user_agent_settings_) {
+      !(request_info_.load_flags & LOAD_MINIMAL_HEADERS) && request()->context()->enable_brotli(),
+      !(request_info_.load_flags & LOAD_MINIMAL_HEADERS) && request()->context()->enable_zstd());
+
+  if (!(request_info_.load_flags & LOAD_MINIMAL_HEADERS) && http_user_agent_settings_) {
     // Only add default Accept-Language if the request didn't have it
     // specified.
     std::string accept_language =
# Increase RSA key length to 2048
--- a/net/cert/x509_util.cc
+++ b/net/cert/x509_util.cc
@@ -281,7 +281,7 @@
 }
 
 // RSA keys created by CreateKeyAndSelfSignedCert will be of this length.
-static const uint16_t kRSAKeyLength = 1024;
+static const uint16_t kRSAKeyLength = 2048;
 
 // Certificates made by CreateKeyAndSelfSignedCert will be signed using this
 // digest algorithm.
# Secure DoH by default
--- a/chrome/browser/net/default_dns_over_https_config_source.cc
+++ b/chrome/browser/net/default_dns_over_https_config_source.cc
@@ -25,7 +25,7 @@
   if (set_up_pref_defaults) {
     local_state->SetDefaultPrefValue(prefs::kDnsOverHttpsMode,
                                      base::Value(SecureDnsConfig::ModeToString(
-                                         net::SecureDnsMode::kAutomatic)));
+                                         net::SecureDnsMode::kSecure)));
   }
 }
 
# Remove Windows parental DoH check
--- a/chrome/browser/net/stub_resolver_config_reader.cc
+++ b/chrome/browser/net/stub_resolver_config_reader.cc
@@ -77,12 +77,6 @@
   kMaxValue = kSecureByEnterprisePolicy,
 };
 
-#if BUILDFLAG(IS_WIN)
-bool ShouldDisableDohForWindowsParentalControls() {
-  return GetWinParentalControls().web_filter;
-}
-#endif  // BUILDFLAG(IS_WIN)
-
 // Check the AsyncDns field trial and return true if it should be enabled. On
 // Android this includes checking the Android version in the field trial.
 bool ShouldEnableAsyncDns() {
@@ -168,42 +162,11 @@
 }
 
 bool StubResolverConfigReader::ShouldDisableDohForManaged() {
-// This function ignores cloud policies which are loaded on a per-profile basis.
-#if BUILDFLAG(IS_ANDROID)
-  // Check for MDM/management/owner apps. android_has_owner_ is true if either a
-  // device or policy owner app is discovered by
-  // GetAndroidEnterpriseInfoState(). If android_has_owner_ is nullopt, take a
-  // value of false so that we don't disable DoH during the async check.
-
-  // Because Android policies can only be loaded with owner apps this is
-  // sufficient to check for the prescences of policies as well.
-  if (android_has_owner_.value_or(false))
-    return true;
-#elif BUILDFLAG(IS_WIN)
-  // TODO(crbug.com/40229843): What is the correct function to use here? (This
-  // may or may not obsolete the following TODO)
-  // TODO(crbug.com/40223626): For legacy compatibility, this uses
-  // IsEnterpriseDevice() which effectively equates to a domain join check.
-  // Consider whether this should use IsManagedDevice() instead.
-  if (base::win::IsEnrolledToDomain())
-    return true;
-#endif
-#if !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_CHROMEOS)
-  if (g_browser_process->browser_policy_connector()->HasMachineLevelPolicies())
-    return true;
-#endif
   return false;
 }
 
 bool StubResolverConfigReader::ShouldDisableDohForParentalControls() {
-  if (parental_controls_testing_override_.has_value())
-    return parental_controls_testing_override_.value();
-
-#if BUILDFLAG(IS_WIN)
-  return ShouldDisableDohForWindowsParentalControls();
-#else
   return false;
-#endif
 }
 
 void StubResolverConfigReader::SetOverrideDnsOverHttpsConfigSource(
# Expand saveable URL schemes
--- a/content/common/url_schemes.cc
+++ b/content/common/url_schemes.cc
@@ -24,13 +24,22 @@
 bool g_registered_url_schemes = false;
 
 const char* const kDefaultSavableSchemes[] = {
+  url::kAboutScheme,
+  url::kBlobScheme,
+  url::kContentScheme,
   url::kHttpScheme,
   url::kHttpsScheme,
   url::kFileScheme,
   url::kFileSystemScheme,
+  url::kFtpScheme,
   kChromeDevToolsScheme,
   kChromeUIScheme,
-  url::kDataScheme
+  url::kDataScheme,
+  url::kJavaScriptScheme,
+  url::kMailToScheme,
+  url::kWsScheme,
+  url::kWssScheme,
+  kViewSourceScheme
 };
 
 // These lists are lazily initialized below and are leaked on shutdown to
# All URLs saveable
--- a/content/public/common/url_utils.cc
+++ b/content/public/common/url_utils.cc
@@ -33,11 +33,7 @@
 }
 
 bool IsSavableURL(const GURL& url) {
-  for (auto& scheme : GetSavableSchemes()) {
-    if (url.SchemeIs(scheme))
-      return true;
-  }
-  return false;
+  return true;
 }
 
 bool IsURLHandledByNetworkStack(const GURL& url) {
