--- a/chrome/browser/ui/webui/BUILD.gn
+++ b/chrome/browser/ui/webui/BUILD.gn
@@ -23,6 +23,7 @@ source_set("configs") {
     "chrome_untrusted_web_ui_configs.h",
     "chrome_web_ui_configs.cc",
     "chrome_web_ui_configs.h",
+    "thorium_easter_eggs.h",
   ]
 
   configs += [ "//build/config/compiler:wexit_time_destructors" ]
--- a/chrome/browser/ui/webui/chrome_web_ui_configs.cc
+++ b/chrome/browser/ui/webui/chrome_web_ui_configs.cc
@@ -45,6 +45,7 @@
 #include "chrome/browser/ui/webui/segmentation_internals/segmentation_internals_ui.h"
 #include "chrome/browser/ui/webui/signin_internals_ui.h"
 #include "chrome/browser/ui/webui/sync_internals/sync_internals_ui.h"
+#include "chrome/browser/ui/webui/thorium_easter_eggs.h"
 #include "chrome/browser/ui/webui/translate_internals/translate_internals_ui.h"
 #include "chrome/browser/ui/webui/usb_internals/usb_internals_ui.h"
 #include "chrome/browser/ui/webui/user_actions/user_actions_ui.h"
@@ -261,6 +262,7 @@ void RegisterChromeWebUIConfigs() {
   map.AddWebUIConfig(std::make_unique<SignInInternalsUIConfig>());
   map.AddWebUIConfig(std::make_unique<SiteEngagementUIConfig>());
   map.AddWebUIConfig(std::make_unique<SyncInternalsUIConfig>());
+  map.AddWebUIConfig(std::make_unique<ThoriumWebUILoadUIConfig>());
   map.AddWebUIConfig(std::make_unique<TranslateInternalsUIConfig>());
   map.AddWebUIConfig(std::make_unique<UsbInternalsUIConfig>());
   map.AddWebUIConfig(std::make_unique<UserActionsUIConfig>());

--- a/chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
+++ b/chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
@@ -359,7 +359,8 @@ bool ChromeWebUIControllerFactory::IsWebUIAllowedToMakeNetworkRequests(
       // https://crbug.com/859345
       origin.host() == chrome::kChromeUIDownloadsHost ||
       // https://crbug.com/376417346
-      origin.host() == chrome::kChromeUIExtensionsHost;
+      origin.host() == chrome::kChromeUIExtensionsHost ||
+      origin.host() == chrome::kChromeUIEggsHost;
 }
 
 ChromeWebUIControllerFactory::ChromeWebUIControllerFactory() = default;
--- a/chrome/common/webui_url_constants.cc
+++ b/chrome/common/webui_url_constants.cc
@@ -74,6 +74,7 @@
 // These hosts will also be suggested by BuiltinProvider.
 base::span<const base::cstring_view> ChromeURLHosts() {
   static constexpr auto kChromeURLHosts = std::to_array<base::cstring_view>({
+      kChromeUIEggsHost,
       kChromeUIAboutHost,
       kChromeUIAccessibilityHost,
       kChromeUIActorInternalsHost,
--- a/chrome/common/webui_url_constants.h
+++ b/chrome/common/webui_url_constants.h
@@ -31,6 +31,8 @@ namespace chrome {
 // Not all components have corresponding URLs and vice versa. Only add as
 // needed.
 // Please keep in alphabetical order, with OS/feature specific sections below.
+inline constexpr char kChromeUIEggsHost[] = "eggs";
+inline constexpr char kChromeUIEggsURL[] = "chrome://eggs/";
 inline constexpr char kChromeUIAboutHost[] = "about";
 inline constexpr char kChromeUIAboutURL[] = "chrome://about/";
 inline constexpr char kChromeUIAccessCodeCastHost[] = "access-code-cast";
--- a/components/webui/about/resources/about_credits_reciprocal.tmpl
+++ b/components/webui/about/resources/about_credits_reciprocal.tmpl
@@ -1,4 +1,4 @@
 <div class="open-sourced">
-    {{opensource_project}} software is made available as source code
-    <a href="{{opensource_link}}">here</a>.
+    {{thorium_project}} is based on {{opensource_project}} and is made available as source code
+    <a href="{{thorium_link}}">here</a>. Chromium source code is available <a href="{{opensource_link}}">here</a>.
 </div>
--- a/tools/licenses/licenses.py
+++ b/tools/licenses/licenses.py
@@ -1083,6 +1083,8 @@
   entries_contents = '\n'.join([entry['content'] for entry in entries])
 
   reciprocal_contents = EvaluateTemplate(reciprocal_template, {
+      'thorium_project': 'Thorium',
+      'thorium_link': 'https://github.com/Alex313031/thorium',
       'opensource_project': 'Chromium',
       'opensource_link': 'https://source.chromium.org/chromium'
   },
--- a/ui/webui/resources/BUILD.gn
+++ b/ui/webui/resources/BUILD.gn
@@ -38,6 +38,7 @@ generate_grd("build_grd") {
     "images:build_grdp",
     "js:build_grdp",
     "mojo:build_grdp",
+    "text:build_grdp",
     "//third_party/lit/v3_0:build_grdp",
   ]
   grdp_files = [
@@ -48,6 +49,7 @@ generate_grd("build_grd") {
     "$target_gen_dir/images/resources.grdp",
     "$target_gen_dir/js/resources.grdp",
     "$target_gen_dir/mojo/resources.grdp",
+    "$target_gen_dir/text/resources.grdp",
   ]
 
   if ((!is_android && !is_ios) || is_desktop_android) {

