--- a/chrome/browser/download/download_target_determiner.cc
+++ b/chrome/browser/download/download_target_determiner.cc
@@ -1,4 +1,4 @@
-// Copyright 2013 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -8,6 +8,7 @@
 #include <string>
 #include <vector>
 
+#include "base/command_line.h"
 #include "base/files/file_path.h"
 #include "base/functional/bind.h"
 #include "base/location.h"
@@ -992,8 +993,13 @@
   // result is available.
   danger_level_ = GetDangerLevel(NO_VISITS_TO_REFERRER);
 
-  if (danger_level_ == DownloadFileType::NOT_DANGEROUS)
+  static const bool allow_insecure_downloads_ =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("allow-insecure-downloads");
+
+  // Continue with this Thorium flag
+  if (danger_level_ == DownloadFileType::NOT_DANGEROUS || allow_insecure_downloads_) {
     return CONTINUE;
+  }
 
   if (danger_level_ == DownloadFileType::ALLOW_ON_USER_GESTURE) {
 #if BUILDFLAG(IS_ANDROID)
@@ -1295,6 +1301,13 @@
 DownloadFileType::DangerLevel DownloadTargetDeterminer::GetDangerLevel(
     PriorVisitsToReferrer visits) const {
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
+
+  static const bool allow_insecure_downloads_ =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("allow-insecure-downloads");
+  // Allow all downloads with this Thorium flag
+  if (allow_insecure_downloads_) {
+    return DownloadFileType::NOT_DANGEROUS;
+  }
 
   // User-initiated extension downloads from pref-whitelisted sources are not
   // considered dangerous.
