--- a/chrome/browser/flag_descriptions.cc
+++ b/chrome/browser/flag_descriptions.cc
@@ -3288,8 +3288,7 @@
 const char kPdfXfaFormsName[] = "PDF XFA support";
 const char kPdfXfaFormsDescription[] =
     "Enables support for XFA forms in PDFs. "
-    "Has no effect if Chrome was not built with XFA support, or if controlled "
-    "by an enterprise policy.";
+    "Has no effect if Chrome was not built with XFA support.";
 
 const char kAutoWebContentsDarkModeName[] = "Auto Dark Mode for Web Contents";
 const char kAutoWebContentsDarkModeDescription[] =
@@ -8266,6 +8265,12 @@
     "rather than crashing. If enabled, DCHECKs will crash the calling process.";
 #endif  // BUILDFLAG(DCHECK_IS_CONFIGURABLE)
 
+#if BUILDFLAG(ENABLE_JXL_DECODER)
+const char kEnableJXLName[] = "Enable/Disable JXL image format";
+const char kEnableJXLDescription[] =
+    "Enable/Disable image decoding support for the JPEG XL image format. (Default: Enabled) Thorium Flag.";
+#endif  // BUILDFLAG(ENABLE_JXL_DECODER)
+
 #if BUILDFLAG(ENABLE_NACL)
 const char kNaclName[] = "Native Client";
 const char kNaclDescription[] =
