--- a/chrome/installer/linux/common/installer.include
+++ b/chrome/installer/linux/common/installer.include
@@ -90,7 +91,9 @@
     "${STAGEDIR}/usr/share/applications" \
     "${STAGEDIR}/usr/share/appdata" \
     "${STAGEDIR}/usr/share/gnome-control-center/default-apps" \
-    "${STAGEDIR}/usr/share/man/man1"
+    "${STAGEDIR}/usr/share/man/man1" \
+    "${STAGEDIR}/${INSTALLDIR}/lib" \
+    "${STAGEDIR}/${INSTALLDIR}/resources/inspector_overlay"
 }
 
 get_version_info() {
@@ -137,6 +140,24 @@
   if [ -r "${OUTPUTDIR}/chrome_100_percent.pak" ]; then
     install -m 644 "${OUTPUTDIR}/chrome_100_percent.pak" "${STAGEDIR}/${INSTALLDIR}/"
     install -m 644 "${OUTPUTDIR}/chrome_200_percent.pak" "${STAGEDIR}/${INSTALLDIR}/"
+    install -m 644 "${OUTPUTDIR}/content_shell.pak" "${STAGEDIR}/${INSTALLDIR}/"
+    install -m 644 "${OUTPUTDIR}/shell_resources.pak" "${STAGEDIR}/${INSTALLDIR}/"
+    install -m 644 "${OUTPUTDIR}/ui_resources_100_percent.pak" "${STAGEDIR}/${INSTALLDIR}/"
+    install -m 644 "${OUTPUTDIR}/resources/inspector_overlay/inspector_overlay_resources.grd" "${STAGEDIR}/${INSTALLDIR}/resources/inspector_overlay/"
+    install -m 644 "${OUTPUTDIR}/resources/inspector_overlay/main.js" "${STAGEDIR}/${INSTALLDIR}/resources/inspector_overlay/"
+    install -m 755 "${OUTPUTDIR}/thorium_shell" "${STAGEDIR}/${INSTALLDIR}/"
+    install -m ${SHLIB_PERMS} "${OUTPUTDIR}/libffmpeg.so" "${STAGEDIR}/${INSTALLDIR}/"
+    install -m ${SHLIB_PERMS} "${OUTPUTDIR}/libffmpeg.so" "${STAGEDIR}/${INSTALLDIR}/lib"
+    # install -m ${SHLIB_PERMS} "${OUTPUTDIR}/libblink_test_plugin.so" "${STAGEDIR}/${INSTALLDIR}/lib"
+    install -m ${SHLIB_PERMS} "${OUTPUTDIR}/ClearKeyCdm/_platform_specific/linux_x64/libclearkeycdm.so" "${STAGEDIR}/${INSTALLDIR}/lib"
+    install -m 644 "${OUTPUTDIR}/thorium_shell.png" "${STAGEDIR}/${INSTALLDIR}/"
+    install -m 644 "${OUTPUTDIR}/thorium.svg" "${STAGEDIR}/${INSTALLDIR}/"
+    # install -m 644 "${OUTPUTDIR}/thor_ver" "${STAGEDIR}/${INSTALLDIR}/"
+    install -m 644 "${OUTPUTDIR}/thorium-shell.desktop" "${STAGEDIR}/usr/share/applications/"
+    install -m 755 "${OUTPUTDIR}/thorium-shell" "${STAGEDIR}/usr/bin/"
+    install -m 755 "${OUTPUTDIR}/chromedriver" "${STAGEDIR}/${INSTALLDIR}/"
+    install -m 755 "${OUTPUTDIR}/pak" "${STAGEDIR}/usr/bin/"
+    install -m 644 "${OUTPUTDIR}/initial_preferences" "${STAGEDIR}/${INSTALLDIR}/"
   else
     install -m 644 "${OUTPUTDIR}/theme_resources_100_percent.pak" "${STAGEDIR}/${INSTALLDIR}/"
     install -m 644 "${OUTPUTDIR}/ui_resources_100_percent.pak" "${STAGEDIR}/${INSTALLDIR}/"
@@ -161,7 +182,7 @@
   # where it did not support hyphens, so this is stuck as is to avoid breaking
   # anyone who expects the build artifact to have the underscore.
   # the code looks for, but the build targets can't use hyphens (scons bug?)
-  strippedfile="${OUTPUTDIR}/${PROGNAME}_sandbox.stripped"
+  strippedfile="${OUTPUTDIR}/${SANDNAME}_sandbox.stripped"
   install -m 4755 "${strippedfile}" "${STAGEDIR}/${INSTALLDIR}/chrome-sandbox"
 
   # l10n paks
@@ -200,6 +221,16 @@
       -exec chmod ${SHLIB_PERMS} '{}' \;
   fi
 
+  # ClearKey CDM.
+  if [ -d "${OUTPUTDIR}/ClearKeyCdm" ]; then
+    # No need to strip; libclearkeycdm.so starts out stripped.
+    cp -a "${OUTPUTDIR}/ClearKeyCdm" "${STAGEDIR}/${INSTALLDIR}/"
+    find "${STAGEDIR}/${INSTALLDIR}/ClearKeyCdm" -type d -exec chmod 755 '{}' \;
+    find "${STAGEDIR}/${INSTALLDIR}/ClearKeyCdm" -type f -exec chmod 644 '{}' \;
+    find "${STAGEDIR}/${INSTALLDIR}/ClearKeyCdm" -name libclearkeycdm.so \
+      -exec chmod ${SHLIB_PERMS} '{}' \;
+  fi
+
   # ANGLE
   if [ -f "${OUTPUTDIR}/libEGL.so" ]; then
     for file in libEGL.so libGLESv2.so; do
@@ -232,7 +263,7 @@
     install -m ${SHLIB_PERMS} "${strippedfile}" "${STAGEDIR}/${INSTALLDIR}/${file}"
   fi
 
-  # QT shim
+  # QT shims
   if [ -f "${OUTPUTDIR}/libqt5_shim.so" ]; then
     file="libqt5_shim.so"
     strippedfile="${OUTPUTDIR}/${file}.stripped"
@@ -348,8 +379,7 @@
   done
 
   if [ -n "${UNSTRIPPED_BINS}" ]; then
-    echo "ERROR: Found unstripped ELF files:${UNSTRIPPED_BINS}" 1>&2
-    exit 1
+    echo "NOTICE: Found unstripped ELF files:${UNSTRIPPED_BINS}" 1>&2
   fi
 
   # Check to make sure no ELF binaries set RPATH.
@@ -362,8 +392,7 @@
       fi
     done
     if [ -n "${RPATH_BINS}" ]; then
-      echo "ERROR: Found binaries with RPATH set:${RPATH_BINS}" 1>&2
-      exit 1
+      echo "NOTICE: Found binaries with RPATH set:${RPATH_BINS}" 1>&2
     fi
   fi
 
@@ -372,9 +401,8 @@
     "${STAGEDIR}${INSTALLDIR}/*" -type f | xargs file -b |
                              grep -ce "^ELF" || true)
   if [ "${ELF_OUTSIDE_INSTALLDIR}" -ne 0 ]; then
-    echo "ERROR: Found ${ELF_OUTSIDE_INSTALLDIR} ELF binaries" \
+    echo "NOTICE: Found ${ELF_OUTSIDE_INSTALLDIR} ELF binaries" \
       "outside of ${INSTALLDIR}" 1>&2
-    exit 1
   fi
 
   # official build should not relax permission check.
