--- a/chrome/browser/net/stub_resolver_config_reader.cc
+++ b/chrome/browser/net/stub_resolver_config_reader.cc
@@ -77,12 +77,6 @@
   kMaxValue = kSecureByEnterprisePolicy,
 };
 
-#if BUILDFLAG(IS_WIN)
-bool ShouldDisableDohForWindowsParentalControls() {
-  return GetWinParentalControls().web_filter;
-}
-#endif  // BUILDFLAG(IS_WIN)
-
 // Check the AsyncDns field trial and return true if it should be enabled. On
 // Android this includes checking the Android version in the field trial.
 bool ShouldEnableAsyncDns() {
@@ -168,42 +162,11 @@
 }
 
 bool StubResolverConfigReader::ShouldDisableDohForManaged() {
-// This function ignores cloud policies which are loaded on a per-profile basis.
-#if BUILDFLAG(IS_ANDROID)
-  // Check for MDM/management/owner apps. android_has_owner_ is true if either a
-  // device or policy owner app is discovered by
-  // GetAndroidEnterpriseInfoState(). If android_has_owner_ is nullopt, take a
-  // value of false so that we don't disable DoH during the async check.
-
-  // Because Android policies can only be loaded with owner apps this is
-  // sufficient to check for the prescences of policies as well.
-  if (android_has_owner_.value_or(false))
-    return true;
-#elif BUILDFLAG(IS_WIN)
-  // TODO(crbug.com/40229843): What is the correct function to use here? (This
-  // may or may not obsolete the following TODO)
-  // TODO(crbug.com/40223626): For legacy compatibility, this uses
-  // IsEnterpriseDevice() which effectively equates to a domain join check.
-  // Consider whether this should use IsManagedDevice() instead.
-  if (base::win::IsEnrolledToDomain())
-    return true;
-#endif
-#if !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_CHROMEOS)
-  if (g_browser_process->browser_policy_connector()->HasMachineLevelPolicies())
-    return true;
-#endif
   return false;
 }
 
 bool StubResolverConfigReader::ShouldDisableDohForParentalControls() {
-  if (parental_controls_testing_override_.has_value())
-    return parental_controls_testing_override_.value();
-
-#if BUILDFLAG(IS_WIN)
-  return ShouldDisableDohForWindowsParentalControls();
-#else
   return false;
-#endif
 }
 
 void StubResolverConfigReader::SetOverrideDnsOverHttpsConfigSource(
