--- a/chrome/installer/linux/debian/postinst
+++ b/chrome/installer/linux/debian/postinst
@@ -39,19 +39,16 @@
     ;;
   unstable )
     # Unstable, give it the "lowest" priority.
-    PRIORITY=120
+    PRIORITY=150
     ;;
   * )
-    PRIORITY=0
+    PRIORITY=150
     ;;
 esac
 
 update-alternatives --install /usr/bin/x-www-browser x-www-browser \
   /usr/bin/@@USR_BIN_SYMLINK_NAME@@ $PRIORITY
 update-alternatives --install /usr/bin/gnome-www-browser gnome-www-browser \
-  /usr/bin/@@USR_BIN_SYMLINK_NAME@@ $PRIORITY
-
-update-alternatives --install /usr/bin/@@PACKAGE_ORIG@@ @@PACKAGE_ORIG@@ \
   /usr/bin/@@USR_BIN_SYMLINK_NAME@@ $PRIORITY
 
 @@include@@../common/apt.include
@@ -62,8 +59,8 @@
 
 ## MAIN ##
 if [ ! -e "$DEFAULTS_FILE" ]; then
-  echo 'repo_add_once="true"' > "$DEFAULTS_FILE"
-  echo 'repo_reenable_on_distupgrade="true"' >> "$DEFAULTS_FILE"
+  echo 'repo_add_once="false"' > "$DEFAULTS_FILE"
+  echo 'repo_reenable_on_distupgrade="false"' >> "$DEFAULTS_FILE"
 fi
 
 # Run the cron job immediately to perform repository configuration.
