--- a/chrome/common/chrome_constants.cc
+++ b/chrome/common/chrome_constants.cc
@@ -12,7 +12,18 @@
 
 namespace chrome {
 
+#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
+#define PRODUCT_FULLNAME_STRING "Google Chrome"
+#elif BUILDFLAG(GOOGLE_CHROME_FOR_TESTING_BRANDING)
+#define PRODUCT_FULLNAME_STRING "Google Chrome for Testing"
+#elif BUILDFLAG(CHROMIUM_BRANDING)
+#define PRODUCT_FULLNAME_STRING "Thorium"
+#else
+#error Unknown branding
+#endif
+
 const char kChromeVersion[] = CHROME_VERSION_STRING;
+const char kBrandName[] = PRODUCT_FULLNAME_STRING;
 
 // The following should not be used for UI strings; they are meant
 // for system strings only. UI changes should be made in the GRD.
@@ -33,9 +44,9 @@
 
 #if BUILDFLAG(IS_WIN)
 const base::FilePath::CharType kBrowserProcessExecutableName[] =
-    FPL("chrome.exe");
+    FPL("thorium.exe");
 const base::FilePath::CharType kHelperProcessExecutableName[] =
-    FPL("chrome.exe");
+    FPL("thorium.exe");
 #elif BUILDFLAG(IS_MAC)
 const base::FilePath::CharType kBrowserProcessExecutableName[] =
     FPL(PRODUCT_FULLNAME_STRING);
@@ -43,11 +54,11 @@
     FPL(PRODUCT_FULLNAME_STRING " Helper");
 #elif BUILDFLAG(IS_ANDROID)
 // NOTE: Keep it synced with the process names defined in AndroidManifest.xml.
-const base::FilePath::CharType kBrowserProcessExecutableName[] = FPL("chrome");
+const base::FilePath::CharType kBrowserProcessExecutableName[] = FPL("thorium");
 const base::FilePath::CharType kHelperProcessExecutableName[] =
     FPL("sandboxed_process");
 #elif BUILDFLAG(IS_POSIX)
-const base::FilePath::CharType kBrowserProcessExecutableName[] = FPL("chrome");
+const base::FilePath::CharType kBrowserProcessExecutableName[] = FPL("thorium");
 // Helper processes end up with a name of "exe" due to execing via
 // /proc/self/exe.  See bug 22703.
 const base::FilePath::CharType kHelperProcessExecutableName[] = FPL("exe");
@@ -55,9 +66,9 @@
 
 #if BUILDFLAG(IS_WIN)
 const base::FilePath::CharType kBrowserProcessExecutablePath[] =
-    FPL("chrome.exe");
+    FPL("thorium.exe");
 const base::FilePath::CharType kHelperProcessExecutablePath[] =
-    FPL("chrome.exe");
+    FPL("thorium.exe");
 #elif BUILDFLAG(IS_MAC)
 const base::FilePath::CharType kBrowserProcessExecutablePath[] =
     FPL(PRODUCT_FULLNAME_STRING ".app/Contents/MacOS/" PRODUCT_FULLNAME_STRING);
@@ -68,16 +79,15 @@
 const base::FilePath::CharType kGoogleChromeBrowserProcessExecutablePath[] =
     FPL("Google Chrome.app/Contents/MacOS/Google Chrome");
 const base::FilePath::CharType kChromiumBrowserProcessExecutablePath[] =
-    FPL("Chromium.app/Contents/MacOS/Chromium");
+    FPL("Thorium.app/Contents/MacOS/Thorium");
 const base::FilePath::CharType kHelperProcessExecutablePath[] =
-    FPL(PRODUCT_FULLNAME_STRING
-        " Helper.app/Contents/MacOS/" PRODUCT_FULLNAME_STRING " Helper");
+    FPL(PRODUCT_FULLNAME_STRING " Helper.app/Contents/MacOS/" PRODUCT_FULLNAME_STRING " Helper");
 #elif BUILDFLAG(IS_ANDROID)
-const base::FilePath::CharType kBrowserProcessExecutablePath[] = FPL("chrome");
-const base::FilePath::CharType kHelperProcessExecutablePath[] = FPL("chrome");
+const base::FilePath::CharType kBrowserProcessExecutablePath[] = FPL("thorium");
+const base::FilePath::CharType kHelperProcessExecutablePath[] = FPL("thorium");
 #elif BUILDFLAG(IS_POSIX)
-const base::FilePath::CharType kBrowserProcessExecutablePath[] = FPL("chrome");
-const base::FilePath::CharType kHelperProcessExecutablePath[] = FPL("chrome");
+const base::FilePath::CharType kBrowserProcessExecutablePath[] = FPL("thorium");
+const base::FilePath::CharType kHelperProcessExecutablePath[] = FPL("thorium");
 #endif  // OS_*
 
 #if BUILDFLAG(IS_MAC)
