--- a/chrome/app/resources/manpage.1.in
+++ b/chrome/app/resources/manpage.1.in
@@ -3,16 +3,16 @@
 .TH @@PACKAGE@@ 1 "" "" "USER COMMANDS"
 
 .SH NAME
-@@PACKAGE@@ \- the web browser from Google
+@@PACKAGE@@ \- the web browser from Alex313031
 
 .SH SYNOPSIS
 .B @@PACKAGE@@
 [\fIOPTION\fR] [\fIPATH\fR|\fIURL\fR]
 
 .SH DESCRIPTION
-See the Google Chrome help center for help on using the browser.
+See the Thorium Repository for help on using the browser.
 .IP
-<https://support.google.com/chrome/>
+<https://github.com/Alex313031/thorium/>
 .PP
 This manpage only describes invocation, environment, and arguments.
 
@@ -156,10 +156,10 @@
 .SH BUGS
 Bug tracker:
 .IP
-http://code.google.com/p/chromium/issues/list
+http://code.google.com/p/chromium/issues/list & https://github.com/Alex313031/thorium/issues
 .PP
 Be sure to do your search within "All Issues" before reporting bugs,
 and be sure to pick the "Defect on Linux" template when filing a new one.
 
 .SH AUTHOR
-The Chromium team \- <http://www.chromium.org>
+The Chromium team and Alex313031 \- <http://www.chromium.org>
