--- a/chrome/browser/memory_details_linux.cc
+++ b/chrome/browser/memory_details_linux.cc
@@ -119,7 +119,7 @@
   ProcessData current_browser =
       GetProcessDataMemoryInformation(GetAllChildren(process_map, getpid()));
   current_browser.name = l10n_util::GetStringUTF16(IDS_SHORT_PRODUCT_NAME);
-  current_browser.process_name = u"chrome";
+  current_browser.process_name = u"thorium";
 
   for (auto i = current_browser.processes.begin();
        i != current_browser.processes.end(); ++i) {
