--- a/chrome/browser/download/chrome_download_manager_delegate.cc
+++ b/chrome/browser/download/chrome_download_manager_delegate.cc
@@ -1952,6 +1952,13 @@ bool ChromeDownloadManagerDelegate::IsOpenInBrowserPreferredForFile(
 bool ChromeDownloadManagerDelegate::ShouldBlockFile(
     download::DownloadItem* item,
     download::DownloadDangerType danger_type) const {
+  // Don't block downloads if flag is set
+  static const bool allow_insecure_downloads_ =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("allow-insecure-downloads");
+  if (allow_insecure_downloads_) {
+    return false;
+  }
+
   policy::DownloadRestriction download_restriction =
       download_prefs_->download_restriction();
 
@@ -1977,42 +1984,13 @@ bool ChromeDownloadManagerDelegate::ShouldBlockFile(
 void ChromeDownloadManagerDelegate::MaybeSendDangerousDownloadOpenedReport(
     DownloadItem* download,
     bool show_download_in_folder) {
-#if BUILDFLAG(SAFE_BROWSING_DOWNLOAD_PROTECTION)
-  safe_browsing::DownloadProtectionService* service =
-      GetDownloadProtectionService();
-  if (service) {
-    service->MaybeSendDangerousDownloadOpenedReport(download,
-                                                    show_download_in_folder);
-  }
-#endif
+  return;
 }
 
 void ChromeDownloadManagerDelegate::MaybeSendDangerousDownloadCanceledReport(
     DownloadItem* download,
     bool is_shutdown) {
-#if BUILDFLAG(SAFE_BROWSING_DOWNLOAD_PROTECTION)
-  safe_browsing::SafeBrowsingService* sb_service =
-      g_browser_process->safe_browsing_service();
-  if (!sb_service) {
-    return;
-  }
-  // Note: We cannot go through download_protection_service here, because this
-  // function may be called at shutdown. The download_protection_service
-  // object may already be deleted at this point.
-  if (is_shutdown) {
-    sb_service->PersistDownloadReportAndSendOnNextStartup(
-        download,
-        safe_browsing::ClientSafeBrowsingReportRequest::
-            DANGEROUS_DOWNLOAD_PROFILE_CLOSED,
-        /*did_proceed=*/false, std::nullopt);
-  } else {
-    sb_service->SendDownloadReport(
-        download,
-        safe_browsing::ClientSafeBrowsingReportRequest::
-            DANGEROUS_DOWNLOAD_AUTO_DELETED,
-        /*did_proceed=*/false, std::nullopt);
-  }
-#endif
+  return;
 }
 
 void ChromeDownloadManagerDelegate::CheckDownloadAllowed(

