--- a/chrome/common/chrome_constants.h
+++ b/chrome/common/chrome_constants.h
@@ -15,6 +15,7 @@
 namespace chrome {
 
 extern const char kChromeVersion[];
+extern const char kBrandName[];
 extern const base::FilePath::CharType kBrowserProcessExecutableName[];
 extern const base::FilePath::CharType kHelperProcessExecutableName[];
 extern const base::FilePath::CharType kBrowserProcessExecutablePath[];
