--- a/chrome/browser/ash/settings/stats_reporting_controller.cc
+++ b/chrome/browser/ash/settings/stats_reporting_controller.cc
@@ -58,9 +58,8 @@
 }
 
 bool StatsReportingController::IsEnabled() const {
-  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
-  std::optional<base::Value> value = GetValue();
-  return value.has_value() && value->is_bool() && value->GetBool();
+  // Disable Telemetry in ThoriumOS
+  return false;
 }
 
 StatsReportingController::StatsReportingController(PrefService* local_state)
