--- a/components/user_education/common/feature_promo/impl/feature_promo_controller_20.cc
+++ b/components/user_education/common/feature_promo/impl/feature_promo_controller_20.cc
@@ -227,8 +227,8 @@
     outputs->anchor_element = anchor_element;
   }
 
-  // Success - the promo can show.
-  return FeaturePromoResult::Success();
+  // Never show annoying promos in Thorium
+  return FeaturePromoResult::kBlockedByConfig;
 }
 
 void FeaturePromoController20::MaybeShowPromo(FeaturePromoParams params) {
