--- a/components/history/core/browser/history_backend.h
+++ b/components/history/core/browser/history_backend.h
@@ -208,7 +208,7 @@
 
   // The number of days old a history entry can be before it is considered "old"
   // and is deleted.
-  static constexpr int kExpireDaysThreshold = 90;
+  static constexpr int kExpireDaysThreshold = 120;
 
   // Init must be called to complete object creation. This object can be
   // constructed on any thread, but all other functions including Init() must
