--- a/components/os_crypt/async/browser/dpapi_key_provider.cc
+++ b/components/os_crypt/async/browser/dpapi_key_provider.cc
@@ -14,6 +14,7 @@
 #include <wincrypt.h>
 
 #include "base/base64.h"
+#include "base/command_line.h"
 #include "base/logging.h"
 #include "base/metrics/histogram_functions.h"
 #include "base/notreached.h"
@@ -41,6 +42,14 @@
 
 std::optional<std::vector<uint8_t>> DecryptKeyWithDPAPI(
     base::span<const uint8_t> ciphertext) {
+
+  static const bool disable_encryption =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("disable-encryption");
+
+  if (disable_encryption) {
+    return std::vector<uint8_t>(ciphertext.begin(), ciphertext.end());
+  }
+
   DATA_BLOB input = {};
   input.pbData = const_cast<BYTE*>(ciphertext.data());
   input.cbData = static_cast<DWORD>(ciphertext.size());
@@ -143,6 +152,7 @@
   }
   std::move(callback).Run(kKeyTag, base::unexpected(status));
 }
+
 bool DPAPIKeyProvider::UseForEncryption() {
   return true;
 }
