--- a/components/metrics/machine_id_provider_win.cc
+++ b/components/metrics/machine_id_provider_win.cc
@@ -1,4 +1,4 @@
-// Copyright 2014 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -10,6 +10,7 @@
 #include <winioctl.h>
 
 #include "base/base_paths.h"
+#include "base/command_line.h"
 #include "base/files/file_path.h"
 #include "base/notreached.h"
 #include "base/path_service.h"
@@ -22,13 +23,25 @@
 
 MachineIdProvider::~MachineIdProvider() = default;
 
+// static
 bool MachineIdProvider::HasId() const {
+  static const bool disable_machine_id =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("disable-machine-id");
+  if (disable_machine_id) {
+    return false;
+  }
   return true;
 }
 
 // On windows, the machine id is based on the serial number of the drive Chrome
 // is running from.
+// static
 std::string MachineIdProvider::GetMachineId() const {
+  static const bool disable_machine_id =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("disable-machine-id");
+  if (disable_machine_id) {
+    return std::string();
+  }
   base::ScopedBlockingCall scoped_blocking_call(FROM_HERE,
                                                 base::BlockingType::MAY_BLOCK);
 
