--- a/components/metrics/machine_id_provider_nonwin.cc
+++ b/components/metrics/machine_id_provider_nonwin.cc
@@ -6,6 +6,8 @@
 
 #include <stdint.h>
 
+#include "base/base_paths.h"
+#include "base/command_line.h"
 #include "base/check.h"
 #include "base/system/sys_info.h"
 
@@ -17,6 +19,11 @@
 
 // Checks if hardware model name is available.
 bool MachineIdProvider::HasId() const {
+  static const bool disable_machine_id =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("disable-machine-id");
+  if (disable_machine_id) {
+    return false;
+  }
   return !base::SysInfo::HardwareModelName().empty();
 }
 
