--- a/components/dom_distiller/core/dom_distiller_features.cc
+++ b/components/dom_distiller/core/dom_distiller_features.cc
@@ -1,4 +1,4 @@
-// Copyright 2015 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -20,13 +20,17 @@
 namespace dom_distiller {
 
 bool IsDomDistillerEnabled() {
+  static const bool kReaderModeDesktop =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("reader-mode");
   return base::CommandLine::ForCurrentProcess()->HasSwitch(
-      switches::kEnableDomDistiller);
+      switches::kEnableDomDistiller) || kReaderModeDesktop;
 }
 
 bool ShouldStartDistillabilityService() {
+  static const bool kReaderModeDesktop =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("reader-mode");
   return base::CommandLine::ForCurrentProcess()->HasSwitch(
-      switches::kEnableDistillabilityService);
+      switches::kEnableDistillabilityService) || kReaderModeDesktop;
 }
 
 #if BUILDFLAG(IS_ANDROID)
@@ -37,7 +41,7 @@
 
 BASE_FEATURE(kReaderModeImprovements,
              "ReaderModeImprovements",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 namespace android {
 static jlong JNI_DomDistillerFeatureMap_GetNativeMap(JNIEnv* env) {
