--- a/components/omnibox/browser/omnibox_view.cc
+++ b/components/omnibox/browser/omnibox_view.cc
@@ -109,19 +109,9 @@
                                      ->GetTemplateURLService()
                                      ->GetTemplateURLForKeyword(match.keyword)
                                : nullptr;
-    // For search queries, display match's search engine's favicon. If the
-    // search engine is google, return the icon instead of favicon for
-    // search queries with the chrome refresh feature.
-    if (turl && search::TemplateURLIsGoogle(turl, controller_->client()
-                                                      ->GetTemplateURLService()
-                                                      ->search_terms_data())) {
-      // For non-chrome builds this would return an empty image model. In
-      // those cases revert to using the favicon.
-      ui::ImageModel icon = model()->GetSuperGIcon(dip_size, dark_mode);
-      if (!icon.IsEmpty()) {
-        return icon;
-      }
-    } else if (turl && turl->CreatedByEnterpriseSearchAggregatorPolicy()) {
+    // For search queries, display match's search engine's favicon. Return the
+    // favicon for search queries with the chrome refresh feature.
+    if (turl && turl->CreatedByEnterpriseSearchAggregatorPolicy()) {
       // If the search engine is enterprise search aggregator, return the icon
       // from the bitmap instead of favicon.
       const SkBitmap* bitmap = model()->GetIconBitmap(turl->favicon_url());
@@ -140,9 +130,6 @@
     if (!match.keyword.empty()) {
       favicon = controller_->client()->GetFaviconForKeywordSearchProvider(
           turl, std::move(on_icon_fetched));
-    } else {
-      favicon = controller_->client()->GetFaviconForDefaultSearchProvider(
-          std::move(on_icon_fetched));
     }
   } else if (match.type != AutocompleteMatchType::HISTORY_CLUSTER) {
     // The starter pack suggestions are a unique case. These suggestions
