--- a/device/vr/BUILD.gn
+++ b/device/vr/BUILD.gn
@@ -128,7 +128,10 @@ if (enable_vr) {
   component("vr") {
     output_name = "device_vr"
 
     defines = [ "DEVICE_VR_IMPLEMENTATION" ]
-    sources = [ "vr_export.h" ]
+    sources = [
+      "vr_export.h",
+      "empty.cc",
+    ]
 
     public_deps = [ "//device/vr/public/cpp" ]
     deps = [
--- a/chrome/browser/vr/BUILD.gn
+++ b/chrome/browser/vr/BUILD.gn
@@ -485,14 +485,16 @@
         ":vr_common",
         "//chrome/browser/content_settings:content_settings_factory",
         "//components/content_settings/core/browser",
         "//device/vr:directx_helpers",
-        "//device/vr:openxr_data",
       ]
 
       if (enable_openxr) {
-        deps += [ "//third_party/openxr" ]
+        deps += [
+          "//device/vr:openxr_data",
+          "//third_party/openxr",
+        ]
         data_deps += [ "//device/vr:openxr_mock" ]
       }
     }
   }
 }
