--- a/sandbox/policy/linux/bpf_audio_policy_linux.cc
+++ b/sandbox/policy/linux/bpf_audio_policy_linux.cc
@@ -1,4 +1,4 @@
-// Copyright 2018 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -72,6 +72,9 @@
 #if defined(__NR_uname)
     case __NR_uname:
 #endif
+#if defined(__NR_sched_getaffinity)
+    case __NR_sched_getaffinity:
+#endif
       return Allow();
 #if defined(__NR_futex)
     case __NR_futex:
