--- a/services/preferences/tracked/tracked_split_preference.cc
+++ b/services/preferences/tracked/tracked_split_preference.cc
@@ -1,4 +1,4 @@
-// Copyright 2014 The Chromium Authors
+// Copyright 2026 The Chromium Authors, win32ss (Shane Fournier), and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -7,6 +7,7 @@
 #include <vector>
 
 #include "base/check.h"
+#include "base/command_line.h"
 #include "base/notreached.h"
 #include "base/values.h"
 #include "services/preferences/public/mojom/tracked_preference_validation_delegate.mojom.h"
@@ -85,6 +86,13 @@
   }
   TrackedPreferenceHelper::ResetAction reset_action =
       helper_.GetAction(value_state);
+
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch("disable-machine-id") ||
+      base::CommandLine::ForCurrentProcess()->HasSwitch("disable-encryption") ||
+      base::CommandLine::ForCurrentProcess()->HasSwitch("revert-from-portable")) {
+        reset_action = TrackedPreferenceHelper::DONT_RESET;
+  }
+
   helper_.ReportAction(reset_action);
 
   if (reset_action == TrackedPreferenceHelper::DO_RESET) {
