--- a/third_party/blink/common/mime_util/mime_util.cc
+++ b/third_party/blink/common/mime_util/mime_util.cc
@@ -16,6 +16,7 @@
 #include "media/media_buildflags.h"
 #include "net/base/mime_util.h"
 #include "third_party/blink/public/common/buildflags.h"
+#include "third_party/blink/public/common/features.h"
 
 #if !BUILDFLAG(IS_IOS)
 // iOS doesn't use and must not depend on //media
@@ -43,6 +44,9 @@
     "image/x-png",
 #if BUILDFLAG(ENABLE_AV1_DECODER)
     "image/avif",
+#endif
+#if BUILDFLAG(ENABLE_JXL_DECODER)
+    "image/jxl",
 #endif
 });
 
