--- a/third_party/blink/renderer/platform/blink_platform_unittests_bundle_data.filelist
+++ b/third_party/blink/renderer/platform/blink_platform_unittests_bundle_data.filelist
@@ -38,6 +38,7 @@
 ../../web_tests/external/wpt/fonts/ahem-extra/AHEM_monospace.TTF
 ../../web_tests/external/wpt/fonts/ahem-extra/AHEM_sans-serif.TTF
 ../../web_tests/external/wpt/fonts/ahem-extra/AHEM_serif.TTF
+../../web_tests/external/wpt/fonts/ahem-visible-zwnj.otf
 ../../web_tests/external/wpt/fonts/fail.woff
 ../../web_tests/external/wpt/fonts/hasubi-mono/HasubiMono-Regular.woff2
 ../../web_tests/external/wpt/fonts/math/axisheight5000-verticalarrow14000.woff
@@ -52,6 +53,7 @@
 ../../web_tests/external/wpt/fonts/math/fraction-numeratorgapmin9000-rulethickness1000.woff
 ../../web_tests/external/wpt/fonts/math/fraction-numeratorshiftup11000-axisheight1000-rulethickness1000.woff
 ../../web_tests/external/wpt/fonts/math/fraction-rulethickness10000.woff
+../../web_tests/external/wpt/fonts/math/largeop-displayoperatorminheight1250.woff
 ../../web_tests/external/wpt/fonts/math/largeop-displayoperatorminheight2000-2AFF-italiccorrection3000.woff
 ../../web_tests/external/wpt/fonts/math/largeop-displayoperatorminheight3000-2AFF-axisheight1000.woff
 ../../web_tests/external/wpt/fonts/math/largeop-displayoperatorminheight5000.woff
@@ -86,6 +88,7 @@
 ../../web_tests/external/wpt/fonts/math/radical-kernafterdegreeminus5000-rulethickness1000.woff
 ../../web_tests/external/wpt/fonts/math/radical-kernbeforedegree4000-rulethickness1000.woff
 ../../web_tests/external/wpt/fonts/math/radical-negativekernbeforedegree1000-rulethickness1000.woff
+../../web_tests/external/wpt/fonts/math/radical-rtlm.woff
 ../../web_tests/external/wpt/fonts/math/radical-rulethickness8000.woff
 ../../web_tests/external/wpt/fonts/math/radical-verticalgap6000-rulethickness1000.woff
 ../../web_tests/external/wpt/fonts/math/scriptpercentscaledown0-scriptscriptpercentscaledown40.woff
@@ -112,6 +115,7 @@
 ../../web_tests/external/wpt/fonts/math/stretchstack-gapbelowmin11000.woff
 ../../web_tests/external/wpt/fonts/math/stretchstack-topshiftup5000.woff
 ../../web_tests/external/wpt/fonts/math/stretchy-centered-on-baseline.woff
+../../web_tests/external/wpt/fonts/math/stretchy-text-direction-asymetrical.woff
 ../../web_tests/external/wpt/fonts/math/stretchy.woff
 ../../web_tests/external/wpt/fonts/math/underover-accentbaseheight4000-overbarextraascender3000.woff
 ../../web_tests/external/wpt/fonts/math/underover-accentbaseheight4000-overbarverticalgap11000.woff
@@ -127,6 +131,9 @@
 ../../web_tests/external/wpt/fonts/noto/NotoSansMongolian-regular.woff2
 ../../web_tests/external/wpt/fonts/noto/NotoSansNko-regular-webfont.woff2
 ../../web_tests/external/wpt/fonts/noto/cjk/NotoSansCJKjp-Regular-subset-chws.otf
+../../web_tests/external/wpt/fonts/noto/cjk/NotoSansCJKjp-Regular-subset-halt-3002.otf
+../../web_tests/external/wpt/fonts/noto/cjk/NotoSansCJKjp-Regular-subset-halt-300C.otf
+../../web_tests/external/wpt/fonts/noto/cjk/NotoSansCJKjp-Regular-subset-halt-FF1A.otf
 ../../web_tests/external/wpt/fonts/noto/cjk/NotoSansCJKjp-Regular-subset-halt-min.otf
 ../../web_tests/external/wpt/fonts/noto/cjk/NotoSansCJKjp-Regular-subset-halt.otf
 ../../web_tests/external/wpt/fonts/noto/noto-sans-v8-latin-regular.woff
@@ -241,6 +248,7 @@
 ../../web_tests/images/resources/animated-10color.gif
 ../../web_tests/images/resources/animated-gif-with-offsets.gif
 ../../web_tests/images/resources/animated.gif
+../../web_tests/images/resources/animated.png
 ../../web_tests/images/resources/animated2.gif
 ../../web_tests/images/resources/apng-test-suite-dispose-op-none-basic.png
 ../../web_tests/images/resources/apng00.png
@@ -363,6 +371,9 @@
 ../../web_tests/images/resources/count-down-color-test.png
 ../../web_tests/images/resources/count-down-color-test.webp
 ../../web_tests/images/resources/crbug.364830.webp
+../../web_tests/images/resources/crbug423310049.png
+../../web_tests/images/resources/crbug432516335-big-height.png
+../../web_tests/images/resources/crbug432516335-i32-overflow.png
 ../../web_tests/images/resources/crbug702934.png
 ../../web_tests/images/resources/crbug752898.bmp
 ../../web_tests/images/resources/crbug779261.gif
@@ -433,6 +444,9 @@
 ../../web_tests/images/resources/icofuzz/png-in-ico.ico
 ../../web_tests/images/resources/icofuzz/twitter_favicon.ico
 ../../web_tests/images/resources/icon-without-and-bitmap.ico
+../../web_tests/images/resources/ihdr50_actl_fctl30_idat.png
+../../web_tests/images/resources/ihdr50_actl_idat_fctl30_fdat.png
+../../web_tests/images/resources/ihdr50_fctl30_idat.png
 ../../web_tests/images/resources/invalid-animated-webp.webp
 ../../web_tests/images/resources/invalid-animated-webp2.webp
 ../../web_tests/images/resources/invalid-animated-webp3.webp
@@ -465,6 +479,42 @@
 ../../web_tests/images/resources/jpegfuzz/rgb-jpeg-green.jpg
 ../../web_tests/images/resources/jpegfuzz/rgb-jpeg-red.jpg
 ../../web_tests/images/resources/jpegfuzz/test-load.jpg
+../../web_tests/images/resources/jxl/3x3.png
+../../web_tests/images/resources/jxl/3x3_gbr_lossless.jxl
+../../web_tests/images/resources/jxl/3x3_gbr_lossy.jxl
+../../web_tests/images/resources/jxl/3x3_hlg_lossless.jxl
+../../web_tests/images/resources/jxl/3x3_hlg_lossy.jxl
+../../web_tests/images/resources/jxl/3x3_pq_lossless.jxl
+../../web_tests/images/resources/jxl/3x3_pq_lossy.jxl
+../../web_tests/images/resources/jxl/3x3_srgb_lossless.jxl
+../../web_tests/images/resources/jxl/3x3_srgb_lossy.jxl
+../../web_tests/images/resources/jxl/3x3a.png
+../../web_tests/images/resources/jxl/3x3a_gbr_lossless.jxl
+../../web_tests/images/resources/jxl/3x3a_gbr_lossy.jxl
+../../web_tests/images/resources/jxl/3x3a_hlg_lossless.jxl
+../../web_tests/images/resources/jxl/3x3a_hlg_lossy.jxl
+../../web_tests/images/resources/jxl/3x3a_pq_lossless.jxl
+../../web_tests/images/resources/jxl/3x3a_pq_lossy.jxl
+../../web_tests/images/resources/jxl/3x3a_srgb_lossless.jxl
+../../web_tests/images/resources/jxl/3x3a_srgb_lossy.jxl
+../../web_tests/images/resources/jxl/README.md
+../../web_tests/images/resources/jxl/alpha-large-dice.jxl
+../../web_tests/images/resources/jxl/alpha-lossless.jxl
+../../web_tests/images/resources/jxl/animated.jxl
+../../web_tests/images/resources/jxl/blue-10-lossless.jxl
+../../web_tests/images/resources/jxl/count.jxl
+../../web_tests/images/resources/jxl/green-10-lossless.jxl
+../../web_tests/images/resources/jxl/icc-grb.jxl
+../../web_tests/images/resources/jxl/icc-grb2.jxl
+../../web_tests/images/resources/jxl/icc-v2-gbr.jxl
+../../web_tests/images/resources/jxl/partial_black.jxl
+../../web_tests/images/resources/jxl/pq_gradient_icc_lossless.jxl
+../../web_tests/images/resources/jxl/pq_gradient_icc_lossy.jxl
+../../web_tests/images/resources/jxl/pq_gradient_lossless.jxl
+../../web_tests/images/resources/jxl/pq_gradient_lossy.jxl
+../../web_tests/images/resources/jxl/red-10-container.jxl
+../../web_tests/images/resources/jxl/red-10-default.jxl
+../../web_tests/images/resources/jxl/red-10-lossless.jxl
 ../../web_tests/images/resources/large-gif-checkerboard.gif
 ../../web_tests/images/resources/large-size-image-crash.jpeg
 ../../web_tests/images/resources/large.webp
@@ -630,6 +680,8 @@
 ../../web_tests/third_party/adobe-fonts/CSSFWOrientationTest.otf
 ../../web_tests/third_party/adobe-fonts/CSSHWOrientationTest.otf
 ../../web_tests/third_party/gsubtest/gsubtest-lookup3.otf
+image-decoders/testing/actl-num-frames-0.png
+image-decoders/testing/apng-with-malformed-2nd-frame.png
 image-decoders/testing/bad-code.gif
 image-decoders/testing/bad-initial-code.gif
 image-decoders/testing/basn3p01-based-long-plte.png
@@ -641,8 +693,10 @@
 image-decoders/testing/first-frame-has-greater-size-than-screen-size.gif
 image-decoders/testing/green-exif-ifd-last.jpg
 image-decoders/testing/incomplete-prev-frame-reusing-clusterfuzz-repro.gif
+image-decoders/testing/interlaced-multiframe-with-blending.png
 image-decoders/testing/invalid-disposal-method.gif
 image-decoders/testing/many-progressive-scans.jpg
+image-decoders/testing/plte-weirdness.png
 image-decoders/testing/private-critical-chunk-before-ihdr.png
 image-decoders/testing/radient-bad-terminator.gif
 image-decoders/testing/radient.gif
@@ -703,6 +757,8 @@
 testing/data/icc-v2-gbr.jpg
 testing/data/original-cat-420-629x473.jpg
 testing/data/palatted-color-png-gamma-one-color-profile.png
+testing/data/png-zero-gamma-color-profile-ref.png
+testing/data/png-zero-gamma-color-profile.png
 testing/data/red-full-ranged-8bpc.avif
 testing/data/rgb-jpeg-red.jpg
 testing/data/roboto-a-different-cmap.ttf
