--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -1,4 +1,4 @@
-// Copyright 2018 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -32,9 +32,10 @@
 
 // Controls the capturing of the Ad-Auction-Signals header, and the maximum
 // allowed Ad-Auction-Signals header value.
+// Disabled by Alex313031
 BASE_FEATURE(kAdAuctionSignals,
              "AdAuctionSignals",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 BASE_FEATURE_PARAM(int,
                    kAdAuctionSignalsMaxSizeBytes,
                    &kAdAuctionSignals,
@@ -47,9 +48,10 @@
              base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Block all MIDI access with the MIDI_SYSEX permission
+// Allow MIDI for Thorium Legacy
 BASE_FEATURE(kBlockMidiByDefault,
              "BlockMidiByDefault",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kComputePressureRateObfuscationMitigation,
              "ComputePressureRateObfuscationMitigation",
@@ -244,15 +246,17 @@
 // feature has to be enabled first (i.e. `kBrowsingTopics` is enabled, and,
 // either a valid Origin Trial token exists or `kPrivacySandboxAdsAPIsOverride`
 // is enabled.)
+// Disabled by Alex313031
 BASE_FEATURE(kBrowsingTopicsDocumentAPI,
              "BrowsingTopicsDocumentAPI",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Decoupled with the main `kBrowsingTopics` feature, so it allows us to
 // decouple the server side configs.
+// Disabled by Alex313031
 BASE_FEATURE(kBrowsingTopicsParameters,
              "BrowsingTopicsParameters",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 // The periodic topics calculation interval.
 BASE_FEATURE_PARAM(base::TimeDelta,
                    kBrowsingTopicsTimePeriodPerEpoch,
@@ -834,7 +838,7 @@
 
 BASE_FEATURE(kFileSystemUrlNavigation,
              "FileSystemUrlNavigation",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kFileSystemUrlNavigationForChromeAppsOnly,
              "FileSystemUrlNavigationForChromeAppsOnly",
@@ -858,7 +862,8 @@
 
 // See https://github.com/WICG/turtledove/blob/main/FLEDGE.md
 // Enables FLEDGE implementation. See https://crbug.com/1186444.
-BASE_FEATURE(kFledge, "Fledge", base::FEATURE_ENABLED_BY_DEFAULT);
+// Disabled in Thorium
+BASE_FEATURE(kFledge, "Fledge", base::FEATURE_DISABLED_BY_DEFAULT);
 
 // See
 // https://github.com/WICG/turtledove/blob/main/FLEDGE_browser_bidding_and_auction_API.md
@@ -2067,6 +2072,7 @@
              "UseAncestorRenderFrameForWorker",
              base::FEATURE_ENABLED_BY_DEFAULT);
 
+// TODO: Alex313031. Maybe re-enable?
 BASE_FEATURE(kPrecompileInlineScripts,
              "PrecompileInlineScripts",
              base::FEATURE_DISABLED_BY_DEFAULT);
@@ -2097,9 +2103,10 @@
 
 // Prefetch request properties are updated to be privacy-preserving. See
 // crbug.com/988956.
+// Enabled by Alex313031
 BASE_FEATURE(kPrefetchPrivacyChanges,
              "PrefetchPrivacyChanges",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kPreloadingHeuristicsMLModel,
              "PreloadingHeuristicsMLModel",
@@ -2203,9 +2210,10 @@
 #endif
 
 // Enables the Private Aggregation API.
+// Disabled by Alex313031
 BASE_FEATURE(kPrivateAggregationApi,
              "PrivateAggregationApi",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Selectively allows the JavaScript API to be disabled in just one of the
 // contexts. The Protected Audience param's name has not been updated (from
@@ -2490,9 +2501,10 @@
 
 // If enabled, force renderer process foregrounded from CommitNavigation to
 // DOMContentLoad (crbug/351953350).
+// Enabled by Alex313031
 BASE_FEATURE(kBoostRenderProcessForLoading,
              "BoostRenderProcessForLoading",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // An empty json array means that this feature is applied unconditionally. If
 // specified, it means that the specified URLs will be the target of the new
