--- a/third_party/blink/renderer/platform/graphics/bitmap_image_metrics.cc
+++ b/third_party/blink/renderer/platform/graphics/bitmap_image_metrics.cc
@@ -37,6 +37,11 @@
 #if BUILDFLAG(ENABLE_AV1_DECODER)
   if (type == "avif")
     return BitmapImageMetrics::DecodedImageType::kAVIF;
+#endif
+#if BUILDFLAG(ENABLE_JXL_DECODER)
+  if (type == "jxl") {
+    return BitmapImageMetrics::DecodedImageType::kJXL;
+  }
 #endif
   return BitmapImageMetrics::DecodedImageType::kUnknown;
 }
