--- a/chrome/browser/ui/views/toolbar/browser_app_menu_button.cc
+++ b/chrome/browser/ui/views/toolbar/browser_app_menu_button.cc
@@ -6,6 +6,7 @@
 
 #include <set>
 
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "base/functional/bind.h"
 #include "base/location.h"
@@ -150,12 +151,18 @@
 }
 
 void BrowserAppMenuButton::UpdateIcon() {
+  static const bool disable_thorium_icons =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("disable-thorium-icons");
   const gfx::VectorIcon& icon = ui::TouchUiController::Get()->touch_ui()
                                     ? kBrowserToolsTouchIcon
-                                    : kBrowserToolsChromeRefreshIcon;
+                                    : disable_thorium_icons
+                                          ? kBrowserToolsChromeRefreshIcon
+                                          : kBrowserToolsChromeRefreshThoriumIcon;
+  // Fix Thorium hamburger menu size
+  constexpr int icon_size = 22;
   for (auto state : kButtonStates) {
     SkColor icon_color = GetForegroundColor(state);
-    SetImageModel(state, ui::ImageModel::FromVectorIcon(icon, icon_color));
+    SetImageModel(state, ui::ImageModel::FromVectorIcon(icon, icon_color, icon_size));
   }
 }
 
