--- a/chrome/browser/ui/startup/default_browser_prompt/default_browser_prompt.cc
+++ b/chrome/browser/ui/startup/default_browser_prompt/default_browser_prompt.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -37,66 +37,13 @@
 namespace {
 
 void ShowPrompt() {
-  // Show the default browser request prompt in the most recently active,
-  // visible, tabbed browser. Do not show the prompt if no such browser exists.
-  for (Browser* browser : BrowserList::GetInstance()->OrderedByActivation()) {
-    // |browser| may be null in UI tests. Also, don't show the prompt in an app
-    // window, which is not meant to be treated as a Chrome window. Only show in
-    // a normal, tabbed browser.
-    if (browser && !browser->is_type_normal()) {
-      continue;
-    }
-
-    // In ChromeBot tests, there might be a race. This line appears to get
-    // called during shutdown and the active web contents can be nullptr.
-    content::WebContents* web_contents =
-        browser->tab_strip_model()->GetActiveWebContents();
-    if (!web_contents ||
-        web_contents->GetVisibility() != content::Visibility::VISIBLE) {
-      continue;
-    }
-
-    DefaultBrowserInfoBarDelegate::Create(
-        infobars::ContentInfoBarManager::FromWebContents(web_contents),
-        browser->profile(),
-        /*can_pin_to_taskbar=*/false);
-    break;
-  }
+  return;
 }
 
 // Do not show the prompt if "suppress_default_browser_prompt_for_version" in
 // the initial preferences is set to the current version.
-bool ShouldShowDefaultBrowserPromptForCurrentVersion() {
-  const std::string disable_version_string =
-      g_browser_process->local_state()->GetString(
-          prefs::kBrowserSuppressDefaultBrowserPrompt);
-  const base::Version disable_version(disable_version_string);
-  DCHECK(disable_version_string.empty() || disable_version.IsValid());
-  return !(disable_version.IsValid() &&
-           disable_version == version_info::GetVersion());
-}
-
-void OnCheckIsDefaultBrowserFinished(
-    Profile* profile,
-    base::OnceCallback<void(bool)> done_callback,
-    shell_integration::DefaultWebClientState state) {
-  bool did_show_prompt = false;
-  if (state == shell_integration::IS_DEFAULT) {
-    // Notify the user in the future if Chrome ceases to be the user's chosen
-    // default browser.
-    chrome::startup::default_prompt::ResetPromptPrefs(profile);
-  } else if (state == shell_integration::NOT_DEFAULT &&
-             shell_integration::CanSetAsDefaultBrowser() &&
-             ShouldShowDefaultBrowserPromptForCurrentVersion()) {
-    chrome::startup::default_prompt::MaybeResetAppMenuPromptPrefs(profile);
-
-    // Only show the prompt if some other program is the user's default browser.
-    // In particular, don't show it if another install mode is default (e.g.,
-    // don't prompt for Chrome Beta if stable Chrome is the default).
-    did_show_prompt =
-        DefaultBrowserPromptManager::GetInstance()->MaybeShowPrompt();
-  }
-  std::move(done_callback).Run(did_show_prompt);
+[[maybe_unused]] bool ShouldShowDefaultBrowserPromptForCurrentVersion() {
+  return false;
 }
 
 }  // namespace
@@ -159,11 +106,7 @@
     // taken care of in BrowserProcessImpl.
     return;
   }
-
-  scoped_refptr<shell_integration::DefaultBrowserWorker>(
-      new shell_integration::DefaultBrowserWorker())
-      ->StartCheckIsDefault(base::BindOnce(&OnCheckIsDefaultBrowserFinished,
-                                           profile, std::move(done_callback)));
+  return;
 }
 
 void ShowPromptForTesting() {
