--- a/chrome/browser/ui/ui_features.cc
+++ b/chrome/browser/ui/ui_features.cc
@@ -18,6 +18,7 @@
 
 // Enables the tab dragging fallback when full window dragging is not supported
 // by the platform (e.g. Wayland). See https://crbug.com/896640
+// TODO: Alex313031 possibly re-disable? Causes issues on newer Wayland
 BASE_FEATURE(kAllowWindowDragUsingSystemDragDrop,
              "AllowWindowDragUsingSystemDragDrop",
              base::FEATURE_ENABLED_BY_DEFAULT);
@@ -333,6 +334,7 @@
 #if BUILDFLAG(IS_CHROMEOS)
              base::FEATURE_ENABLED_BY_DEFAULT
 #else
+             // TODO: Alex313031 Test setting to enabled
              base::FEATURE_DISABLED_BY_DEFAULT
 #endif
 );
