--- a/chrome/browser/ui/tabs/features.cc
+++ b/chrome/browser/ui/tabs/features.cc
@@ -1,4 +1,4 @@
-// Copyright 2024 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -12,6 +12,7 @@
 
 // Splits pinned and unpinned tabs into separate TabStrips.
 // https://crbug.com/1346019
+// TODO: Alex313031 Possibly Re-Enable after feedback
 BASE_FEATURE(kSplitTabStrip,
              "SplitTabStrip",
              base::FEATURE_DISABLED_BY_DEFAULT);
@@ -26,7 +27,7 @@
 // https://crbug.com/1145747
 BASE_FEATURE(kScrollableTabStripWithDragging,
              "kScrollableTabStripWithDragging",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 const char kTabScrollingWithDraggingModeName[] = "tabScrollWithDragMode";
 
 // Enables different methods of overflow when scrolling tabs in tabstrip
@@ -42,6 +43,7 @@
              "TabSearchPositionSetting",
              base::FEATURE_DISABLED_BY_DEFAULT);
 
+// TODO: Alex313031 Maybe disable?
 BASE_FEATURE(kTabGroupShortcuts,
              "TabGroupShortcuts",
              base::FEATURE_ENABLED_BY_DEFAULT);
