--- a/chrome/browser/ui/views/toolbar/reload_button.cc
+++ b/chrome/browser/ui/views/toolbar/reload_button.cc
@@ -1,4 +1,4 @@
-// Copyright 2013 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -41,7 +41,9 @@
                     CreateMenuModel(),
                     nullptr),
       command_updater_(command_updater),
-      reload_icon_(vector_icons::kReloadChromeRefreshIcon),
+      reload_icon_(disable_thorium_icons
+                   ? vector_icons::kReloadChromeRefreshIcon
+                   : vector_icons::kReloadChromeRefreshThoriumIcon),
       reload_touch_icon_(kReloadTouchIcon),
       stop_icon_(kNavigateStopChromeRefreshIcon),
       stop_touch_icon_(kNavigateStopTouchIcon),
@@ -115,7 +117,7 @@
 }
 
 void ReloadButton::SetMenuEnabled(bool enable) {
-  menu_enabled_ = enable;
+  menu_enabled_ = true;
   UpdateAccessibleHasPopup();
   UpdateCachedTooltipText();
 }
