--- a/chrome/browser/ui/startup/bad_flags_prompt.cc
+++ b/chrome/browser/ui/startup/bad_flags_prompt.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -92,7 +92,7 @@
 
 #if BUILDFLAG(ENABLE_EXTENSIONS)
     // This flag gives extensions more powers.
-    extensions::switches::kExtensionsOnChromeURLs,
+    // extensions::switches::kExtensionsOnChromeURLs, (Disabled by Alex313031)
 #endif
 
 #if BUILDFLAG(IS_LINUX)
@@ -103,8 +103,8 @@
 
     // These flags control Blink feature state, which is not supported and is
     // intended only for use by Chromium developers.
-    switches::kDisableBlinkFeatures,
-    switches::kEnableBlinkFeatures,
+    // switches::kDisableBlinkFeatures, (Disabled by Alex313031)
+    // switches::kEnableBlinkFeatures, (Disabled by Alex313031)
 
     // This flag allows people to allowlist certain origins as secure, even
     // if they are not.
@@ -126,9 +126,9 @@
     // updating components won't be performed until shutdown.
     switches::kDisableBestEffortTasks,
 
-    // GPU sanboxing isn't implemented for the Web GPU API yet meaning it would
+    // GPU sandboxing isn't implemented for the Web GPU API yet meaning it would
     // be possible to read GPU data for other Chromium processes.
-    switches::kEnableUnsafeWebGPU,
+    // switches::kEnableUnsafeWebGPU, (Disabled by Alex313031)
 
     // A flag to bypass the WebHID blocklist for testing purposes.
     switches::kDisableHidBlocklist,
