--- a/chrome/browser/ui/toolbar/app_menu_model.cc
+++ b/chrome/browser/ui/toolbar/app_menu_model.cc
@@ -834,7 +834,7 @@
     int help_string_id = IDS_HELP_PAGE;
 #endif
     AddItemWithStringIdAndVectorIcon(this, IDC_ABOUT, IDS_ABOUT,
-                                     vector_icons::kInfoRefreshIcon);
+                                     vector_icons::kInfoRegularIcon);
 #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
     if (whats_new::IsEnabled()) {
       AddItemWithStringIdAndVectorIcon(this, IDC_CHROME_WHATS_NEW,
@@ -1912,7 +1912,7 @@
 
   AddItemWithStringIdAndVectorIcon(this, IDC_CLEAR_BROWSING_DATA,
                                    IDS_CLEAR_BROWSING_DATA,
-                                   kTrashCanRefreshIcon);
+                                   kTrashCanRegularIcon);
 
   AddSeparator(ui::NORMAL_SEPARATOR);
   CreateZoomMenu();
@@ -1996,10 +1996,12 @@
                                       sub_menus_.back().get(), kHelpMenuIcon);
 #else
 #if BUILDFLAG(IS_CHROMEOS)
-  AddItem(IDC_ABOUT, l10n_util::GetStringUTF16(IDS_ABOUT));
+  //AddItem(IDC_ABOUT, l10n_util::GetStringUTF16(IDS_ABOUT));
+  AddItemWithStringIdAndVectorIcon(this, IDC_ABOUT, IDS_ABOUT,
+                                   vector_icons::kInfoRegularIcon);
 #else
   AddItemWithStringIdAndVectorIcon(this, IDC_ABOUT, IDS_ABOUT,
-                                   vector_icons::kInfoRefreshIcon);
+                                   vector_icons::kInfoRegularIcon);
 #endif
 #endif
 
