--- a/chrome/browser/ui/browser_commands.cc
+++ b/chrome/browser/ui/browser_commands.cc
@@ -1,4 +1,4 @@
-// Copyright 2012 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -600,11 +600,6 @@
     CoreTabHelper* core_tab_helper =
         CoreTabHelper::FromWebContents(current_tab);
     content_restrictions = core_tab_helper->content_restrictions();
-    NavigationEntry* last_committed_entry =
-        current_tab->GetController().GetLastCommittedEntry();
-    if (!content::IsSavableURL(last_committed_entry->GetURL())) {
-      content_restrictions |= CONTENT_RESTRICTION_SAVE;
-    }
   }
   return content_restrictions;
 }
@@ -1917,8 +1912,7 @@
       policy::DownloadRestriction::ALL_FILES) {
     return false;
   }
-  return !browser->is_type_devtools() &&
-         !(GetContentRestrictions(browser) & CONTENT_RESTRICTION_SAVE);
+  return true;
 }
 
 void Print(Browser* browser) {
