--- a/chrome/browser/ui/startup/infobar_utils.cc
+++ b/chrome/browser/ui/startup/infobar_utils.cc
@@ -168,6 +168,7 @@
     ShowBadFlagsPrompt(web_contents);
   }
 
+  /*
   infobars::ContentInfoBarManager* infobar_manager =
       infobars::ContentInfoBarManager::FromWebContents(web_contents);
 
@@ -181,7 +182,7 @@
         !local_state->GetBoolean(prefs::kSuppressUnsupportedOSWarning)) {
       ObsoleteSystemInfoBarDelegate::Create(infobar_manager);
     }
-  }
+  } */
 
 #if BUILDFLAG(IS_WIN) && BUILDFLAG(GOOGLE_CHROME_BRANDING)
     if (auto* controller = g_browser_process->GetFeatures()
