--- a/ui/webui/resources/images/BUILD.gn
+++ b/ui/webui/resources/images/BUILD.gn
@@ -1,7 +1,8 @@
-# Copyright 2020 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+import("//build/config/chromeos/ui_mode.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
 generate_grd("build_grdp") {
@@ -24,6 +25,7 @@
     "error.svg",
     "error_yellow900.svg",
     "extension.svg",
+    "hazard.svg",
     "icon_arrow_dropdown.svg",
     "icon_cancel.svg",
     "icon_copy_content.svg",
@@ -33,6 +35,7 @@
     "icon_tab.svg",
     "icon_refresh.svg",
     "icon_search.svg",
+    "infra.png",
     "open_in_new.svg",
     "select.png",
     "throbber_medium.svg",
@@ -47,14 +50,6 @@
     "2x/select.png",
   ]
 
-  if (!is_ios) {
-    input_files += [
-      "chrome_logo_dark.svg",
-      "icon_arrow_back.svg",
-      "icon_edit.svg",
-    ]
-  }
-
   if (!is_android && !is_ios) {
     input_files += [
       "arrow_down.svg",
@@ -62,14 +57,17 @@
       "business.svg",
       "colorize.svg",
       "chevron_down.svg",
+      "chrome_logo_dark.svg",
       "dark/arrow_down.svg",
       "dark/chevron_down.svg",
+      "icon_arrow_back.svg",
       "icon_arrow_drop_down_cr23.svg",
       "icon_arrow_drop_up_cr23.svg",
       "icon_bookmark.svg",
       "icon_clear.svg",
       "icon_clock.svg",
       "icon_delete_gray.svg",
+      "icon_edit.svg",
       "icon_filetype_generic.svg",
       "icon_folder_open.svg",
       "icon_history.svg",
@@ -79,9 +77,6 @@
       "icon_settings.svg",
       "icon_visibility_off.svg",
       "icon_visibility.svg",
-      "promotion_banner_light.svg",
-      "dark/promotion_banner_dark.svg",
-      "promotion_policy_banner_close.svg",
     ]
   }
 
