--- a/ui/views/examples/BUILD.gn
+++ b/ui/views/examples/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2014 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -258,7 +258,7 @@
   public_deps = [ ":views_examples_lib" ]
 }
 
-executable("views_examples_with_content") {
+executable("thorium_ui_debug_shell") {
   testonly = true
 
   sources = [ "examples_with_content_main.cc" ]
