--- a/build/toolchain/apple/linker_driver.py
+++ b/build/toolchain/apple/linker_driver.py
@@ -1,6 +1,6 @@
 #!/usr/bin/env python3
 
-# Copyright 2016 The Chromium Authors
+# Copyright 2026 The Chromium Authors, midzer, and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -150,19 +150,23 @@
         self._linker_driver_pre_actions = {}
         self._compiler_driver_args = []
         for index, arg in enumerate(self._args[1:]):
+            BAD_RUSTC_ARGS = '-Wl,-plugin-opt=O[0-9],-plugin-opt=mcpu=.*'
             if arg.startswith(LINKER_DRIVER_COMPILER_ARG_PREFIX):
                 assert not self._driver_path
                 self._driver_path = arg[len(LINKER_DRIVER_COMPILER_ARG_PREFIX
                                             ):]
+                if not re.match(BAD_RUSTC_ARGS, arg):
+                    self._compiler_driver_args.append(arg)
             elif arg.startswith(LINKER_DRIVER_ARG_PREFIX):
                 # Convert driver actions into a map of name => lambda to invoke.
                 self._process_driver_arg(arg)
+                if not re.match(BAD_RUSTC_ARGS, arg):
+                    self._compiler_driver_args.append(arg)
             else:
                 # TODO(crbug.com/40268754): On Apple, the linker command line
                 # produced by rustc for LTO includes these arguments, but the
                 # Apple linker doesn't accept them.
                 # Upstream bug: https://github.com/rust-lang/rust/issues/60059
-                BAD_RUSTC_ARGS = '-Wl,-plugin-opt=O[0-9],-plugin-opt=mcpu=.*'
                 if not re.match(BAD_RUSTC_ARGS, arg):
                     self._compiler_driver_args.append(arg)
 
