--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2014 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -164,8 +164,6 @@
   android_resources("chrome_base_module_resources") {
     sources = [
       "java/res_base/anim/no_anim.xml",
-      "java/res_base/drawable-v26/ic_launcher.xml",
-      "java/res_base/drawable-v26/ic_launcher_round.xml",
       "java/res_base/font/chrome_google_sans.xml",
       "java/res_base/font/chrome_google_sans_bold.xml",
       "java/res_base/font/chrome_google_sans_medium.xml",
@@ -178,20 +176,14 @@
       "java/res_chromium_base/drawable/themed_app_icon.xml",
       "java/res_chromium_base/mipmap-hdpi/app_icon.png",
       "java/res_chromium_base/mipmap-hdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-hdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-mdpi/app_icon.png",
       "java/res_chromium_base/mipmap-mdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-mdpi/layered_app_icon_background.png",
-      "java/res_chromium_base/mipmap-nodpi/layered_app_icon_foreground.xml",
       "java/res_chromium_base/mipmap-xhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-xxhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xxhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xxhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-xxxhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xxxhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xxxhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/values/channel_constants.xml",
     ]
   }
@@ -1618,20 +1610,14 @@
       "java/res_chromium_base/drawable/themed_app_icon.xml",
       "java/res_chromium_base/mipmap-hdpi/app_icon.png",
       "java/res_chromium_base/mipmap-hdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-hdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-mdpi/app_icon.png",
       "java/res_chromium_base/mipmap-mdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-mdpi/layered_app_icon_background.png",
-      "java/res_chromium_base/mipmap-nodpi/layered_app_icon_foreground.xml",
       "java/res_chromium_base/mipmap-xhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-xxhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xxhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xxhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/mipmap-xxxhdpi/app_icon.png",
       "java/res_chromium_base/mipmap-xxxhdpi/layered_app_icon.png",
-      "java/res_chromium_base/mipmap-xxxhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/values/channel_constants.xml",
     ]
 
@@ -1984,7 +1970,7 @@
 
   chrome_public_apk_or_module_tmpl("chrome_public_apk") {
     target_type = "android_apk"
-    apk_name = "ChromePublic"
+    apk_name = "Thorium_Public"
     if (defined(android_app_secondary_abi)) {
       # Ensure 64-bit chrome does not depend on 32-bit things.
       assert_no_deps =
@@ -1994,7 +1980,7 @@
   }
 
   chrome_public_bundle("chrome_public_bundle") {
-    bundle_name = "ChromePublic"
+    bundle_name = "Thorium_Public_Bundle"
   }
 
   android_library("monochrome_java") {
@@ -2263,7 +2249,7 @@
   # scoped, and do not leave lingering state after execution.
   # TODO(crbug.com/40193330): Set clear rules for what tests can be added here.
   chrome_public_test_apk_tmpl("chrome_public_unit_test_apk") {
-    apk_name = "ChromePublicUnitTest"
+    apk_name = "ThoriumPublicUnitTest"
     deps = [
       ":chrome_unit_test_java",
       "//build/config/android/test/resource_overlay:unit_device_javatests",
@@ -2338,7 +2324,7 @@
     # TODO(wnwen): Re-enable when new lint failures are disabled for test targets.
     #enable_lint = true
 
-    apk_name = "ChromePublicTest"
+    apk_name = "ThoriumPublicTest"
     deps = [
       ":chrome_test_apk_template_resources",
       ":chrome_test_java",
