--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1601,6 +1601,8 @@
     "tab_group_sync/tab_group_sync_utils.h",
     "tab_group_sync/tab_group_trial.cc",
     "tab_group_sync/tab_group_trial.h",
+    "thorium_flag_choices.h",
+    "thorium_flag_entries.h",
     "tpcd/experiment/eligibility_service.cc",
     "tpcd/experiment/eligibility_service.h",
     "tpcd/experiment/eligibility_service_factory.cc",
