--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -27,15 +27,16 @@
   # WebView providers which declare one of a handful of package names. See
   # https://chromium.googlesource.com/chromium/src/+/HEAD/android_webview/docs/build-instructions.md#Changing-package-name
   # for details.
-  chrome_public_manifest_package = "org.chromium.chrome"
+  chrome_public_manifest_package = "org.chromium.thorium"
   if (use_stable_package_name_for_trichrome) {
     chrome_public_manifest_package += ".stable"
-  } else if (android_channel != "default") {
+    # Changed by Alex313031 for Thorium
+  } else if (android_channel != "default" && android_channel != "stable") {
     chrome_public_manifest_package += "." + android_channel
   }
 }
 
-chrome_public_test_manifest_package = "org.chromium.chrome.tests"
+chrome_public_test_manifest_package = "org.chromium.thorium.tests"
 
 default_chrome_public_jinja_variables = [
   "channel=$android_channel",
