--- a/components/BUILD.gn
+++ b/components/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2014 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -1095,8 +1095,8 @@
     ]
 
     if (is_mac) {
-      data_deps += [ "//content/shell:content_shell" ]
-      data += [ "$root_out_dir/Content Shell.app/" ]
+      data_deps += [ "//content/shell:thorium_shell" ]
+      data += [ "$root_out_dir/Thorium Shell.app/" ]
     }
 
     if (toolkit_views && !is_mac) {
