--- a/components/vector_icons/BUILD.gn
+++ b/components/vector_icons/BUILD.gn
@@ -1,8 +1,9 @@
-# Copyright 2017 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
 import("//components/vector_icons/vector_icons.gni")
+import("//third_party/blink/public/public_features.gni")
 
 _icons_gen_dir = "$root_build_dir/gen/components/vector_icons"
 
@@ -39,6 +40,7 @@
       "bluetooth_off_chrome_refresh.icon",
       "bluetooth_scanning.icon",
       "bluetooth_scanning_chrome_refresh.icon",
+      "bookmark_folder.icon",
       "business.icon",
       "business_chrome_refresh.icon",
       "call.icon",
@@ -103,6 +105,7 @@
       "fingerprint_off.icon",
       "folder.icon",
       "folder_chrome_refresh.icon",
+      "folder_managed.icon",
       "folder_managed_refresh.icon",
       "folder_open.icon",
       "font_download.icon",
@@ -129,6 +132,7 @@
       "image_search.icon",
       "info_outline.icon",
       "info_refresh.icon",
+      "info_regular.icon",
       "insert_drive_file_outline.icon",
       "install_desktop.icon",
       "install_desktop_off.icon",
@@ -198,6 +202,7 @@
       "receipt_long.icon",
       "reload.icon",
       "reload_chrome_refresh.icon",
+      "restore.icon",
       "replay.icon",
       "replay_10.icon",
       "router.icon",
@@ -278,6 +283,14 @@
       "work.icon",
     ]
 
+  if (is_thorium_build) {
+    sources += [
+      "thorium/reload_thorium.icon",
+      "thorium/reload_chrome_refresh_thorium.icon",
+      "thorium/restore_tab.icon"
+    ]
+  }
+
     if (is_chromeos) {
       sources += [
         "notification_download.icon",
