--- a/third_party/widevine/cdm/BUILD.gn
+++ b/third_party/widevine/cdm/BUILD.gn
@@ -40,8 +40,13 @@
   ]
 
   if (enable_widevine_cdm_host_verification) {
-    assert(is_win || is_mac)
+    assert(target_os == "win" || is_mac)
     widevine_cdm_binary_files += [ "${widevine_cdm_root}/${cdm_file_name}.sig" ]
+  }
+
+  if (target_os == "win") {
+    assert(target_os == "win")
+    widevine_cdm_binary_files += [ "${widevine_cdm_root}/${cdm_file_name}.lib" ]
   }
 }
 
