--- a/third_party/blink/renderer/platform/BUILD.gn
+++ b/third_party/blink/renderer/platform/BUILD.gn
@@ -1935,6 +1935,15 @@
     sources += [ "text/hyphenation/hyphenation_null.cc" ]
   }
 
+  if (enable_jxl_decoder && !is_android) {
+    sources += [
+      "image-decoders/jxl/jxl_image_decoder.cc",
+      "image-decoders/jxl/jxl_image_decoder.h",
+    ]
+
+    deps += [ "//third_party/libjxl:libjxl" ]
+  }
+
   if (current_cpu == "x86" || current_cpu == "x64") {
     deps += [ ":blink_x86_avx" ]
     sources += [ "audio/cpu/x86/delay_sse2.cc" ]
@@ -2429,6 +2438,10 @@
     sources += [ "text/locale_icu_test.cc" ]
   }
 
+  if (enable_jxl_decoder && !is_android) {
+    sources += [ "image-decoders/jxl/jxl_image_decoder_test.cc" ]
+  }
+
   sources += [ "testing/run_all_tests.cc" ]
 
   configs += [
@@ -2491,6 +2504,10 @@
     "//url/mojom:test_url_mojom_gurl_blink",
   ]
 
+  if (!is_android) {
+    deps += [ "//third_party/libjxl:libjxl" ]
+  }
+
   deps += [ ":bridge_ice_controller_tests" ]
 
   data_deps = [ ":blink_platform_unittests_data" ]
