--- a/third_party/widevine/cdm/widevine.gni
+++ b/third_party/widevine/cdm/widevine.gni
@@ -12,13 +12,12 @@
   # Google Chrome for Testing and Android. For Fuchsia, see crbug.com/1354347.
   # Can be optionally enabled in Chromium on non-Android platforms. Please see
   # //src/third_party/widevine/LICENSE file for details.
-  enable_widevine = ((is_chrome_branded || is_chrome_for_testing_branded) &&
-                     !is_fuchsia) || is_android
+  enable_widevine = !is_fuchsia
 }
 
 # Disable Widevine when running linux-chromeos.
 if (is_chromeos && !is_chromeos_device) {
-  enable_widevine = false
+  enable_widevine = true
 }
 
 # Widevine CDM is available as a library CDM on the following platforms and
@@ -27,7 +26,7 @@
 library_widevine_cdm_available =
     (is_chromeos &&
      (target_cpu == "x64" || target_cpu == "arm" || target_cpu == "arm64")) ||
-    (target_os == "linux" && target_cpu == "x64") ||
+    (target_os == "linux") ||
     (target_os == "mac" && (target_cpu == "x64" || target_cpu == "arm64")) ||
     (target_os == "win" &&
      (target_cpu == "x86" || target_cpu == "x64" || target_cpu == "arm64"))
@@ -64,19 +63,19 @@
   widevine_root = "."
 }
 
-# Enable Widevine CDM host verification, which will sign additional binaries
-# with Widevine certs and copy signature files as needed.
-enable_widevine_cdm_host_verification =
-    enable_library_widevine_cdm && enable_cdm_host_verification
+declare_args() {
+  # Enable Widevine CDM host verification, which will sign additional binaries
+  # with Widevine certs and copy signature files as needed.
+  enable_widevine_cdm_host_verification =
+      enable_library_widevine_cdm && enable_cdm_host_verification
 
-declare_args() {
   # If set, and Widevine CDM host verification signing failed due to no signing
   # cert, the failure will be ignored. Otherwise the build process will fail.
   # Set to false by default for official build to catch missing cert error.
   # For developers building with "is_official_build" locally without Widevine
   # signing certs, please manually set `ignore_missing_widevine_signing_cert`
   # to true to suppress the error.
-  ignore_missing_widevine_signing_cert = !is_official_build
+  ignore_missing_widevine_signing_cert = true
 }
 
 template("widevine_sign_file") {
