--- a/tools/v8_context_snapshot/BUILD.gn
+++ b/tools/v8_context_snapshot/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2017 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -101,6 +101,10 @@
         "//v8",
       ]
 
+      if (is_linux || is_chromeos) {
+        configs += [ "//build/config/gcc:rpath_for_built_shared_libraries" ]
+      }
+
       configs += [
         "//v8:external_startup_data",
         ":disable_icf",
