--- a/sandbox/win/src/sandbox_policy_base.cc
+++ b/sandbox/win/src/sandbox_policy_base.cc
@@ -275,15 +275,6 @@
   // Signed intercept rules only supported on Windows 10 TH2 and above. This
   // must match the version checks in process_mitigations.cc for
   // consistency.
-  if (base::win::GetVersion() >= base::win::Version::WIN10_TH2) {
-    DCHECK_EQ(MITIGATION_FORCE_MS_SIGNED_BINS,
-              mitigations_ & MITIGATION_FORCE_MS_SIGNED_BINS)
-        << "Enable MITIGATION_FORCE_MS_SIGNED_BINS before adding signed "
-           "policy rules.";
-    if (!SignedPolicy::GenerateRules(base::FilePath(path), PolicyMaker())) {
-      return SBOX_ERROR_BAD_PARAMS;
-    }
-  }
   return SBOX_ALL_OK;
 }
 
