--- a/content/browser/launch_as_mojo_client_browsertest.cc
+++ b/content/browser/launch_as_mojo_client_browsertest.cc
@@ -37,9 +37,9 @@
 namespace {
 
 #if BUILDFLAG(IS_WIN)
-const char kShellExecutableName[] = "content_shell.exe";
+const char kShellExecutableName[] = "thorium_shell.exe";
 #else
-const char kShellExecutableName[] = "content_shell";
+const char kShellExecutableName[] = "thorium_shell";
 #endif
 
 base::FilePath GetCurrentDirectory() {
