--- a/content/shell/android/shell_apk/AndroidManifest.xml.jinja2
+++ b/content/shell/android/shell_apk/AndroidManifest.xml.jinja2
@@ -25,7 +25,7 @@
     <application android:name="org.chromium.content_shell_apk.ContentShellApplication"
             android:icon="@mipmap/app_icon"
             android:zygotePreloadName="org.chromium.content_public.app.ZygotePreload"
-            android:label="{% block application_label %}Content Shell{% endblock %}">
+            android:label="{% block application_label %}Thorium Shell{% endblock %}">
         <activity android:name="org.chromium.content_shell_apk.ContentShellActivity"
                   android:launchMode="singleTask"
                   android:theme="@android:style/Theme.Holo.Light.NoActionBar"
