--- a/content/shell/BUILD.gn
+++ b/content/shell/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2014 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -43,8 +43,8 @@
 shell_use_toolkit_views = toolkit_views && !is_castos
 
 declare_args() {
-  content_shell_product_name = "Content Shell"
-  content_shell_version = "999.77.34.5"
+  content_shell_product_name = "Thorium Shell"
+  content_shell_version = "999.7.3434.555"
   content_shell_major_version = "999"
 }
 
@@ -603,7 +603,7 @@
 }
 
 if (is_android) {
-  group("content_shell") {
+  group("thorium_shell") {
     testonly = true
     deps = [ "//content/shell/android:content_shell_apk" ]
   }
@@ -617,7 +617,7 @@
       content_shell_version,
     ]
   }
-  mac_app_bundle("content_shell") {
+  mac_app_bundle("thorium_shell") {
     testonly = true
     output_name = content_shell_product_name
     sources = [ "app/shell_main_mac.cc" ]
@@ -728,7 +728,7 @@
     bundle_identifier = ios_content_process_bundle_identifier
   }
 } else {
-  executable("content_shell") {
+  executable("thorium_shell") {
     testonly = true
 
     sources = [ "app/shell_main.cc" ]
@@ -791,7 +791,7 @@
     fuchsia_component("content_shell_component") {
       testonly = true
       manifest = "fuchsia/content_shell.cml"
-      data_deps = [ ":content_shell" ]
+      data_deps = [ ":thorium_shell" ]
       visibility = [ ":*" ]
     }
 
@@ -1083,7 +1083,7 @@
 group("content_shell_crash_test") {
   testonly = true
   data_deps = [
-    ":content_shell",
+    ":thorium_shell",
     "//testing:test_scripts_shared",
     "//third_party/angle:includes",
   ]
