--- a/extensions/common/api/declarative_net_request.idl
+++ b/extensions/common/api/declarative_net_request.idl
@@ -815,28 +815,29 @@
         TestMatchOutcomeCallback callback);
   };
 
+  // Increase MV3 limits in Thorium!
   interface Properties {
     // The minimum number of static rules guaranteed to an extension across its
     // enabled static rulesets. Any rules above this limit will count towards
     // the <a href="#global-static-rule-limit">global static rule limit</a>.
-    [value=30000] static long GUARANTEED_MINIMUM_STATIC_RULES();
+    [value=60000] static long GUARANTEED_MINIMUM_STATIC_RULES();
 
     // The maximum number of combined dynamic and session scoped rules an
     // extension can add.
-    [nodoc, value=5000, deprecated="There is no longer a combined limit. See $(ref:MAX_NUMBER_OF_DYNAMIC_RULES) and $(ref:MAX_NUMBER_OF_SESSION_RULES)."] static long MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES();
+    [nodoc, value=10000, deprecated="There is no longer a combined limit. See $(ref:MAX_NUMBER_OF_DYNAMIC_RULES) and $(ref:MAX_NUMBER_OF_SESSION_RULES)."] static long MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES();
 
     // The maximum number of dynamic rules that an extension can add.
-    [value=30000] static long MAX_NUMBER_OF_DYNAMIC_RULES();
+    [value=60000] static long MAX_NUMBER_OF_DYNAMIC_RULES();
 
     // The maximum number of "unsafe" dynamic rules that an extension can add.
-    [value=5000] static long MAX_NUMBER_OF_UNSAFE_DYNAMIC_RULES();
+    [value=10000] static long MAX_NUMBER_OF_UNSAFE_DYNAMIC_RULES();
 
     // The maximum number of session scoped rules that an extension can add.
-    [value=5000] static long MAX_NUMBER_OF_SESSION_RULES();
+    [value=10000] static long MAX_NUMBER_OF_SESSION_RULES();
 
     // The maximum number of "unsafe" session scoped rules that an extension can
     // add.
-    [value=5000] static long MAX_NUMBER_OF_UNSAFE_SESSION_RULES();
+    [value=10000] static long MAX_NUMBER_OF_UNSAFE_SESSION_RULES();
 
     // Time interval within which <code>MAX_GETMATCHEDRULES_CALLS_PER_INTERVAL
     // getMatchedRules</code> calls can be made, specified in minutes.
@@ -847,20 +848,20 @@
 
     // The number of times <code>getMatchedRules</code> can be called within a
     // period of <code>GETMATCHEDRULES_QUOTA_INTERVAL</code>.
-    [value=20] static long MAX_GETMATCHEDRULES_CALLS_PER_INTERVAL();
+    [value=25] static long MAX_GETMATCHEDRULES_CALLS_PER_INTERVAL();
 
     // The maximum number of regular expression rules that an extension can
     // add. This limit is evaluated separately for the set of dynamic rules and
     // those specified in the rule resources file.
-    [value=1000] static long MAX_NUMBER_OF_REGEX_RULES();
+    [value=2000] static long MAX_NUMBER_OF_REGEX_RULES();
 
     // The maximum number of static <code>Rulesets</code> an extension can
     // specify as part of the <code>"rule_resources"</code> manifest key.
-    [value=100] static long MAX_NUMBER_OF_STATIC_RULESETS();
+    [value=256] static long MAX_NUMBER_OF_STATIC_RULESETS();
 
     // The maximum number of static <code>Rulesets</code> an extension can
     // enable at any one time.
-    [value=50] static long MAX_NUMBER_OF_ENABLED_STATIC_RULESETS();
+    [value=128] static long MAX_NUMBER_OF_ENABLED_STATIC_RULESETS();
 
     // Ruleset ID for the dynamic rules added by the extension.
     [value="_dynamic"] static DOMString DYNAMIC_RULESET_ID();
