--- a/net/base/mime_util_unittest.cc
+++ b/net/base/mime_util_unittest.cc
@@ -39,6 +39,7 @@
       {FILE_PATH_LITERAL("webm"), "video/webm"},
       {FILE_PATH_LITERAL("weba"), "audio/webm"},
       {FILE_PATH_LITERAL("avif"), "image/avif"},
+      {FILE_PATH_LITERAL("jxl"), "image/jxl"},
       {FILE_PATH_LITERAL("epub"), "application/epub+zip"},
       {FILE_PATH_LITERAL("apk"), "application/vnd.android.package-archive"},
       {FILE_PATH_LITERAL("cer"), "application/x-x509-ca-cert"},
@@ -80,6 +81,7 @@
     {FILE_PATH_LITERAL("webm"), {"video/webm"}},
     {FILE_PATH_LITERAL("weba"), {"audio/webm"}},
     {FILE_PATH_LITERAL("avif"), {"image/avif"}},
+    {FILE_PATH_LITERAL("jxl"), {"image/jxl"}},
 #if BUILDFLAG(IS_CHROMEOS)
     // These are test cases for testing platform mime types on ChromeOS.
     {FILE_PATH_LITERAL("epub"), {"application/epub+zip"}},
@@ -504,6 +506,7 @@
       {"MeSsAge/*", 1, "eml"},
       {"message/", 0, nullptr, true},
       {"image/avif", 1, "avif"},
+      {"image/jxl", 1, "jxl"},
       {"image/bmp", 1, "bmp"},
       {"video/*", 6, "mp4"},
       {"video/*", 6, "mpeg"},
