--- a/net/dns/dns_client.cc
+++ b/net/dns/dns_client.cc
@@ -10,6 +10,7 @@
 #include <string>
 #include <utility>
 
+#include "base/command_line.h"
 #include "base/functional/bind.h"
 #include "base/logging.h"
 #include "base/memory/raw_ptr.h"
@@ -88,6 +89,8 @@
 }
 
 class DnsClientImpl : public DnsClient {
+ const bool disable_thorium_dns_config =
+     base::CommandLine::ForCurrentProcess()->HasSwitch("disable-thorium-dns-config");
  public:
   DnsClientImpl(NetLog* net_log, const RandIntCallback& rand_int_callback)
       : net_log_(net_log), rand_int_callback_(rand_int_callback) {}
@@ -250,11 +253,14 @@
  private:
   std::optional<DnsConfig> BuildEffectiveConfig() const {
     DnsConfig config;
-    if (config_overrides_.OverridesEverything()) {
+    // In Thorium it is sufficient to have secure DoH enabled to give the overrides priority
+    if (config_overrides_.dns_over_https_config && config_overrides_.secure_dns_mode && !disable_thorium_dns_config) {
       config = config_overrides_.ApplyOverrides(DnsConfig());
     } else {
-      if (!system_config_)
+      if (!system_config_) {
+        LOG(WARNING) << "dns_client.cc->BuildEffectiveConfig(): System configuration not set: No system_config_ ";
         return std::nullopt;
+      }
 
       config = config_overrides_.ApplyOverrides(system_config_.value());
     }
@@ -269,8 +275,10 @@
     if (config.unhandled_options)
       config.nameservers.clear();
 
-    if (!config.IsValid())
+    if (!config.IsValid()) {
+      DLOG(WARNING) << "dns_client.cc->BuildEffectiveConfig(): Invalid configuration";
       return std::nullopt;
+    }
 
     return config;
   }
