--- a/net/url_request/url_request_http_job.cc
+++ b/net/url_request/url_request_http_job.cc
@@ -516,18 +516,22 @@
   // fields in the referrer.
   GURL referrer(request_->referrer());
 
-  // Our consumer should have made sure that this is a safe referrer (e.g. via
-  // URLRequestJob::ComputeReferrerForPolicy).
-  if (referrer.is_valid()) {
-    std::string referer_value = referrer.spec();
-    request_info_.extra_headers.SetHeader(HttpRequestHeaders::kReferer,
-                                          referer_value);
-  }
-
-  request_info_.extra_headers.SetHeaderIfMissing(
-      HttpRequestHeaders::kUserAgent,
-      http_user_agent_settings_ ?
-          http_user_agent_settings_->GetUserAgent() : std::string());
+  if (!(request_info_.load_flags & LOAD_MINIMAL_HEADERS)) {
+    // Our consumer should have made sure that this is a safe referrer (e.g. via
+    // URLRequestJob::ComputeReferrerForPolicy).
+    if (referrer.is_valid()) {
+      std::string referer_value = referrer.spec();
+      request_info_.extra_headers.SetHeader(HttpRequestHeaders::kReferer,
+                                            referer_value);
+    }
+  }
+
+  if (!(request_info_.load_flags & LOAD_MINIMAL_HEADERS)) {
+    request_info_.extra_headers.SetHeaderIfMissing(
+        HttpRequestHeaders::kUserAgent,
+        http_user_agent_settings_ ?
+            http_user_agent_settings_->GetUserAgent() : std::string());
+  }
 
   AddExtraHeaders();
 
@@ -773,10 +777,10 @@
 void URLRequestHttpJob::AddExtraHeaders() {
   request_info_.extra_headers.SetAcceptEncodingIfMissing(
       request()->url(), request()->accepted_stream_types(),
-      request()->context()->enable_brotli(),
-      request()->context()->enable_zstd());
-
-  if (http_user_agent_settings_) {
+      !(request_info_.load_flags & LOAD_MINIMAL_HEADERS) && request()->context()->enable_brotli(),
+      !(request_info_.load_flags & LOAD_MINIMAL_HEADERS) && request()->context()->enable_zstd());
+
+  if (!(request_info_.load_flags & LOAD_MINIMAL_HEADERS) && http_user_agent_settings_) {
     // Only add default Accept-Language if the request didn't have it
     // specified.
     std::string accept_language =
