--- a/net/dns/dns_transaction.cc
+++ b/net/dns/dns_transaction.cc
@@ -484,7 +484,7 @@
     // avoid deadlock and enable the use of preconfigured IP addresses.
     request_->SetSecureDnsPolicy(SecureDnsPolicy::kBootstrap);
     request_->SetLoadFlags(request_->load_flags() | LOAD_DISABLE_CACHE |
-                           LOAD_BYPASS_PROXY);
+                           LOAD_MINIMAL_HEADERS | LOAD_BYPASS_PROXY);
     request_->set_allow_credentials(false);
     request_->set_isolation_info(isolation_info);
   }
