--- a/chrome/app/vector_icons/BUILD.gn
+++ b/chrome/app/vector_icons/BUILD.gn
@@ -1,10 +1,11 @@
-# Copyright 2017 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
 import("//build/config/chromeos/ui_mode.gni")
 import("//components/vector_icons/vector_icons.gni")
 import("//device/vr/buildflags/buildflags.gni")
+import("//third_party/blink/public/public_features.gni")
 import("//ui/webui/webui_features.gni")
 
 aggregate_vector_icons("chrome_vector_icons") {
@@ -156,6 +157,7 @@
     "privacy_tip.icon",
     "qr_code_chrome_refresh.icon",
     "qrcode_generator.icon",
+    "quick_commands.icon",
     "read_later_add.icon",
     "reading_list.icon",
     "release_alert.icon",
@@ -209,6 +211,7 @@
     "translate.icon",
     "trash_can.icon",
     "trash_can_refresh.icon",
+    "trash_can_regular.icon",
     "tv.icon",
     "ungroup_refresh.icon",
     "user_account_avatar.icon",
@@ -245,6 +248,24 @@
 
     # go/keep-sorted end
   ]
+
+  if (is_thorium_build) {
+    sources += [
+      "thorium/browser_tools_thorium.icon",
+      "thorium/browser_tools_chrome_refresh_thorium.icon",
+      "thorium/chrome_labs_thorium.icon",
+      "thorium/chrome_labs_chrome_refresh_thorium.icon",
+      "thorium/navigate_home_thorium.icon",
+      "thorium/navigate_home_chrome_refresh_thorium.icon",
+      "thorium/science_thorium.icon",
+      "thorium/side_panel_left_thorium.icon",
+      "thorium/side_panel_left_chrome_refresh_thorium.icon",
+      "thorium/side_panel_left_touch_thorium.icon",
+      "thorium/side_panel_thorium.icon",
+      "thorium/side_panel_chrome_refresh_thorium.icon",
+      "thorium/side_panel_touch_thorium.icon",
+    ]
+  }
 
   if (is_mac) {
     sources += [ "new_tab_mac_touchbar.icon" ]
