--- a/chrome/app/theme/chromium/BRANDING
+++ b/chrome/app/theme/chromium/BRANDING
@@ -1,10 +1,10 @@
-COMPANY_FULLNAME=The Chromium Authors
-COMPANY_SHORTNAME=The Chromium Authors
-PRODUCT_FULLNAME=Chromium
-PRODUCT_SHORTNAME=Chromium
-PRODUCT_INSTALLER_FULLNAME=Chromium Installer
-PRODUCT_INSTALLER_SHORTNAME=Chromium Installer
-COPYRIGHT=Copyright @LASTCHANGE_YEAR@ The Chromium Authors. All rights reserved.
-MAC_BUNDLE_ID=org.chromium.Chromium
+COMPANY_FULLNAME=Alex313031
+COMPANY_SHORTNAME=Alex313031
+PRODUCT_FULLNAME=Thorium
+PRODUCT_SHORTNAME=Thorium
+PRODUCT_INSTALLER_FULLNAME=Thorium Installer
+PRODUCT_INSTALLER_SHORTNAME=Thorium Installer
+COPYRIGHT=Copyright @LASTCHANGE_YEAR@ The Chromium Authors and Alex313031.
+MAC_BUNDLE_ID=org.chromium.Thorium
 MAC_CREATOR_CODE=Cr24
 MAC_TEAM_ID=
