--- a/components/components_settings_strings.grdp
+++ b/components/components_settings_strings.grdp
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <grit-part>
   <message name="IDS_SETTINGS_TITLE" desc="Title for the settings tab.">
-    Settings
+    Thorium Settings
   </message>
   <message name="IDS_SETTINGS_HIDE_ADVANCED_SETTINGS" desc="Title for the link to hide advanced settings.">
     Hide advanced settings...
