--- a/components/new_or_sad_tab_strings.grdp
+++ b/components/new_or_sad_tab_strings.grdp
@@ -10,7 +10,7 @@
 
       <!-- Sad Tab Strings -->
       <message name="IDS_SAD_TAB_TITLE" desc="The title of the sad tab page that is shown when a tab crashes. This is intended to be a humorous exclamation of dismay." formatter_data="android_java">
-        Aw, Snap!
+        Aw, #@%!, this tab's process has gone bye bye...
       </message>
       <message name="IDS_SAD_TAB_MESSAGE" desc="The message displayed on the sad tab page." formatter_data="android_java">
         Something went wrong while displaying this webpage.
@@ -30,44 +30,44 @@
         </message>
       </if>
       <if expr="is_win">
-        <message name="IDS_SAD_TAB_OOM_TITLE" desc="The title of the web page displayed if content in Chrome browser does not load due to the browser being out of memory, and a reload failed to fix the issue.">
+        <message name="IDS_SAD_TAB_OOM_TITLE" desc="The title of the web page displayed if content in Thorium browser does not load due to the browser being out of memory, and a reload failed to fix the issue.">
           Not enough memory to open this page
         </message>
       </if>
-      <message name="IDS_SAD_TAB_RELOAD_TITLE" desc="The title of the web page displayed if content in Chrome browser does not load for an unknown reason, and a reload failed to fix the issue." formatter_data="android_java">
+      <message name="IDS_SAD_TAB_RELOAD_TITLE" desc="The title of the web page displayed if content in Thorium browser does not load for an unknown reason, and a reload failed to fix the issue." formatter_data="android_java">
         Can't open this page
       </message>
-      <message name="IDS_SAD_TAB_OOM_MESSAGE_TABS" desc="The message displayed on the web page if content in Chrome browser does not load due to the browser being out of memory, a reload failed to fix the issue and other tabs are open.">
+      <message name="IDS_SAD_TAB_OOM_MESSAGE_TABS" desc="The message displayed on the web page if content in Thorium browser does not load due to the browser being out of memory, a reload failed to fix the issue and other tabs are open.">
         Try closing other tabs or programs to free up memory.
       </message>
-      <message name="IDS_SAD_TAB_OOM_MESSAGE_NOTABS" desc="The message displayed on the web page if content in Chrome browser does not load due to the browser being out of memory, a reload failed to fix the issue and no other tabs are open.">
+      <message name="IDS_SAD_TAB_OOM_MESSAGE_NOTABS" desc="The message displayed on the web page if content in Thorium browser does not load due to the browser being out of memory, a reload failed to fix the issue and no other tabs are open.">
         Try exiting other programs to free up memory.
       </message>
-      <message name="IDS_SAD_TAB_RELOAD_TRY" desc="The message displayed on the web page if content in Chrome browser does not load and a reload failed to fix the issue. Introduces a bulleted list of solutions/tips for the user to try." formatter_data="android_java">
+      <message name="IDS_SAD_TAB_RELOAD_TRY" desc="The message displayed on the web page if content in Thorium browser does not load and a reload failed to fix the issue. Introduces a bulleted list of solutions/tips for the user to try." formatter_data="android_java">
         Try the following tips:
       </message>
       <if expr="is_macosx">
-        <message name="IDS_SAD_TAB_RELOAD_INCOGNITO" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to open the web page in Chrome's Incognito mode.">
+        <message name="IDS_SAD_TAB_RELOAD_INCOGNITO" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to open the web page in Thorium's Incognito mode.">
           Open page in a new Incognito window (⇧⌘N)
         </message>
       </if>
       <if expr="not is_macosx and toolkit_views">
-        <message name="IDS_SAD_TAB_RELOAD_INCOGNITO" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to open the web page in Chrome's Incognito mode.">
+        <message name="IDS_SAD_TAB_RELOAD_INCOGNITO" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to open the web page in Thorium's Incognito mode.">
           Open page in a new Incognito window (Ctrl-Shift-N)
         </message>
       </if>
       <if expr="is_android or is_ios">
-        <message name="IDS_SAD_TAB_RELOAD_INCOGNITO" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to open the web page in Chrome's Incognito mode." formatter_data="android_java">
+        <message name="IDS_SAD_TAB_RELOAD_INCOGNITO" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to open the web page in Thorium's Incognito mode." formatter_data="android_java">
           Open page in a new Incognito tab
         </message>
       </if>
       <if expr="is_macosx or is_chromeos">
-        <message name="IDS_SAD_TAB_RELOAD_CLOSE_TABS" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to close other Chrome tabs or apps running on their computer (Mac, ChromeOS).">
+        <message name="IDS_SAD_TAB_RELOAD_CLOSE_TABS" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to close other Thorium tabs or apps running on their computer (Mac, ThoriumOS).">
           Close other tabs or apps
         </message>
       </if>
       <if expr="is_linux">
-        <message name="IDS_SAD_TAB_RELOAD_CLOSE_TABS" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to close other Chrome tabs or programs running on their computer.">
+        <message name="IDS_SAD_TAB_RELOAD_CLOSE_TABS" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to close other Thorium tabs or programs running on their computer.">
           Close other tabs or programs
         </message>
       </if>
@@ -82,13 +82,13 @@
         </message>
       </if>
       <if expr="_google_chrome">
-        <message name="IDS_SAD_TAB_RELOAD_RESTART_BROWSER" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to restart Chrome." formatter_data="android_java">
-          Restart Chrome
+        <message name="IDS_SAD_TAB_RELOAD_RESTART_BROWSER" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to restart Thorium." formatter_data="android_java">
+          Restart Thorium
         </message>
       </if>
       <if expr="not _google_chrome">
-        <message name="IDS_SAD_TAB_RELOAD_RESTART_BROWSER" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to restart Chromium." formatter_data="android_java">
-          Restart Chromium
+        <message name="IDS_SAD_TAB_RELOAD_RESTART_BROWSER" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to restart Thorium." formatter_data="android_java">
+          Restart Thorium
         </message>
       </if>
       <if expr="is_android or is_ios">
@@ -163,7 +163,7 @@
       </message>
        <if expr="_google_chrome">
         <message name="IDS_NEW_TAB_OTR_NOT_SAVED" desc="Bullet points listing data that are not saved in the Incognito mode. 'Browsing history' means a history of visited websites. 'Cookies and site data' refers to data saved by websites on the user's device (e.g. sign-in state, preferences, etc.). 'Information entered in forms' refers to names, addresses, passwords etc. that users enter into forms on the web." formatter_data="android_java">
-          Chrome <ph name="BEGIN_EMPHASIS">&lt;em&gt;</ph>won’t save<ph name="END_EMPHASIS">&lt;/em&gt;</ph>:
+          Thorium <ph name="BEGIN_EMPHASIS">&lt;em&gt;</ph>won’t save<ph name="END_EMPHASIS">&lt;/em&gt;</ph>:
           <ph name="BEGIN_LIST">&lt;ul&gt;</ph>
             <ph name="LIST_ITEM">&lt;li&gt;</ph>Your browsing history
             <ph name="LIST_ITEM">&lt;li&gt;</ph>Cookies and site data
@@ -173,7 +173,7 @@
       </if>
       <if expr="not _google_chrome">
         <message name="IDS_NEW_TAB_OTR_NOT_SAVED" desc="Bullet points listing data that are not saved in the Incognito mode. 'Browsing history' means a history of visited websites. 'Cookies and site data' refers to data saved by websites on the user's device (e.g. sign-in state, preferences, etc.). 'Information entered in forms' refers to names, addresses, passwords etc. that users enter into forms on the web." formatter_data="android_java">
-          Chromium <ph name="BEGIN_EMPHASIS">&lt;em&gt;</ph>won’t save<ph name="END_EMPHASIS">&lt;/em&gt;</ph>:
+          Thorium <ph name="BEGIN_EMPHASIS">&lt;em&gt;</ph>won’t save<ph name="END_EMPHASIS">&lt;/em&gt;</ph>:
           <ph name="BEGIN_LIST">&lt;ul&gt;</ph>
             <ph name="LIST_ITEM">&lt;li&gt;</ph>Your browsing history
             <ph name="LIST_ITEM">&lt;li&gt;</ph>Cookies and site data
@@ -211,13 +211,13 @@
       <if expr="not is_android">
         <if expr="_google_chrome">
           <message name="IDS_NEW_TAB_OTR_THIRD_PARTY_BLOCKED_COOKIE_SUBLABEL" desc="Sub-text explaining 3rd-party cookie blocking with a help center link to re-enabling 3rd-party cookies for specific sites." formatter_data="android_java">
-            Chrome blocks sites from using third-party cookies to track you as you browse.
+            Thorium blocks sites from using third-party cookies to track you as you browse.
             <ph name="NEW_LINE">&lt;br/&gt;</ph>If site features aren’t working <ph name="START_LINK">&lt;a target="_blank" href="$1" aria-description="$2"&gt;</ph>try temporarily allowing third-party cookies<ph name="END_LINK">&lt;/a&gt;</ph>.
           </message>
         </if>
         <if expr="not _google_chrome">
           <message name="IDS_NEW_TAB_OTR_THIRD_PARTY_BLOCKED_COOKIE_SUBLABEL" desc="Sub-text explaining 3rd-party cookie blocking with a help center link to re-enabling 3rd-party cookies for specific sites." formatter_data="android_java">
-            Chromium blocks sites from using third-party cookies to track you as you browse.
+            Thorium blocks sites from using third-party cookies to track you as you browse.
             <ph name="NEW_LINE">&lt;br/&gt;</ph>If site features aren’t working <ph name="START_LINK">&lt;a target="_blank" href="$1" aria-description="$2"&gt;</ph>try temporarily allowing third-party cookies<ph name="END_LINK">&lt;/a&gt;</ph>.
           </message>
         </if>
@@ -225,12 +225,12 @@
       <if expr="is_android">
         <if expr="_google_chrome">
           <message name="IDS_NEW_TAB_OTR_THIRD_PARTY_BLOCKED_COOKIE_PART_ONE" desc="Sub-text explaining 3rd-party cookie blocking with a help center link to re-enabling 3rd-party cookies for specific sites. Part 1/2." formatter_data="android_java">
-            Chrome blocks sites from using third-party cookies to track you as you browse.
+            Thorium blocks sites from using third-party cookies to track you as you browse.
           </message>
         </if>
         <if expr="not _google_chrome">
           <message name="IDS_NEW_TAB_OTR_THIRD_PARTY_BLOCKED_COOKIE_PART_ONE" desc="Sub-text explaining 3rd-party cookie blocking with a help center link to re-enabling 3rd-party cookies for specific sites. Part 1/2." formatter_data="android_java">
-            Chromium blocks sites from using third-party cookies to track you as you browse.
+            Thorium blocks sites from using third-party cookies to track you as you browse.
           </message>
         </if>
         <message name="IDS_NEW_TAB_OTR_THIRD_PARTY_BLOCKED_COOKIE_PART_TWO" desc="Sub-text explaining 3rd-party cookie blocking with a help center link to re-enabling 3rd-party cookies for specific sites. Part 2/2." formatter_data="android_java">
