--- a/components/heavy_ad_intervention_strings.grdp
+++ b/components/heavy_ad_intervention_strings.grdp
@@ -7,7 +7,7 @@
     Ad removed.
   </message>
   <message name="IDS_HEAVY_AD_INTERVENTION_SUMMARY" desc="Summary in heavy ad intervention page when ad was removed for using too many system resources.">
-    This ad used too many resources for your device, so Chrome removed it.
+    This ad used too many resources for your device, so Thorium removed it.
   </message>
   <message name="IDS_HEAVY_AD_INTERVENTION_BUTTON_RELOAD" desc="Label for the button that reloads the page and disables the intervention">
     Reload and show anyway
